---
title: "HoloFoodR R package â€” case study"
date: "`r Sys.Date()`"
package: HoloFoodR
bibliography: HoloFoodR.bib
output:
   BiocStyle::html_document:
       fig_height: 7
       fig_width: 10
       toc: yes
       toc_depth: 2
       number_sections: true
vignette: >
  %\VignetteIndexEntry{Case study}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

## Introduction

[HoloFood database](https://www.holofooddata.org) is a large collection of
holo-omic and multi-omic data from two animal systems, chicken and salmon. It
was created by the HoloFood consortium. One of its objectives is to study the
interactions between animal systems, their microbiomes, and feed additives
to optimize the diet strategies of these farm animals.

For make it easily accessible, the creators provide an Application Programming
Interface (API) that permits interaction with programming languages, such as R
or Python.

We improve the accessibility by creating the HoloFoodR package that simplifies
API interaction and assists translating raw database data into R/Bioconductor
data containers, connecting to a vast ecosystem of bioinformatics R packages.

We do not aim to demonstrate HoloFoodR in isolation from the rest of the
ecosystem, but to showcase the possibility of data integration from other
databases, such as MGnify, which holds metagenomic data. Furthermore, we
provide a workflow from data exploration to advanced machine learning and
multi-omics, offering a practical example for readers.

Our main study questions are:

    - Do treatment associate with salmons' gut flora?
    - Do larger salmons have different gut microbiota compared to smaller ones?
    - Do gut microbiota have association with muscle tissue fatty acids?

```{r}
#| label: start
#| include: false

library(knitr)
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    cache = TRUE
)
```

```{r}
#| label: setup
#| output: false

# List of packages that we need
packages <- c(
    "dplyr", "DT", "ggsignif", "HoloFoodR", "MGnifyR", "mia", "miaViz", "MOFA2",
    "patchwork", "reticulate", "scater"
)

# Load all packages into session. Stop if there are packages that were not
# successfully loaded
pkgs_not_loaded <- !sapply(packages, require, character.only = TRUE)
pkgs_not_loaded <- names(pkgs_not_loaded)[pkgs_not_loaded]
if (length(pkgs_not_loaded) > 0) {
    stop("Error in loading the following packages into the session: '",
        paste0(pkgs_not_loaded, collapse = "', '"), "'")
}
```

## Import data

We start the workflow from data retrieval. We will use
[salmon data](https://www.holofooddata.org/samples/?animal__system=salmon)
and its associated fatty acid and metagenomic amplicon as an example.

### Retrieve HoloFood data

First of all, we have to query the HoloFood database to retrieve the salmon
accession numbers.

```{r}
#| label: get_animals
#| eval: false

# Get salmon samples
salmons <- HoloFoodR::doQuery("animals", system = "salmon", use.cache = TRUE)

# Get only the data that has both metagenomic amplicon and fatty acid
# data
salmons <- salmons |>
    filter(fatty_acids == TRUE & metagenomic_amplicon == TRUE)

colnames(salmons)
```

Next, we can retrieve the data associated with each salmon.

```{r}
#| label: get_animal_data
#| eval: false

# Get salmon data
salmon_data <- HoloFoodR::getData(
    accession.type = "animals",
    accession = salmons[["accession"]],
    use.cache = TRUE
)

# Get salmon samples
salmon_samples <- salmon_data[["samples"]]

# Get sample IDs
salmon_sample_ids <- unique(salmon_samples[["accession"]])

head(salmon_sample_ids)
```

The data returned above is a list of all sample accession numbers that are
associated with all salmons. For example, metagenomic amplicon samples, such as
[SAMEA112750580](https://www.holofooddata.org/sample/SAMEA112750580)
or genomic sample, such as
[SAMEA9068691](https://www.holofooddata.org/sample/SAMEA9068691).

We can use these accession numbers to fetch the data associated with each
sample type and store them as `experiments` in a `MultiAssayExperiment` (`MAE`)
object.

```{r}
#| label: get_salmon_mae
#| eval: false

# Get salmon <- experiments as MAE object
mae <- HoloFoodR::getResult(
    salmon_sample_ids,
    use.cache = TRUE
)
```

```{r}
#| label: save_salmon_mae
#| echo: false
#| eval: false

# Save salmon MAE This must be named differently
path <- system.file(
  "extdata", "salmon_mae_without_mgnify.RDS", package = "HoloFoodR")
saveRDS(object = mae, file = path)
```

```{r}
#| label: read_salmon_mae
#| echo: false

# Read MAE to avoid retrieval
path <- system.file(
  "extdata", "salmon_mae_without_mgnify.RDS", package = "HoloFoodR")
mae <- readRDS(file = path)
```

### Fetch metagenomic data from MGnify

HoloFood database does not include the data for metagenomic data. This data
can be retrieved from the [MGnify portal](https://www.ebi.ac.uk/metagenomics).
For this purpose, we will use MGnifyR package, which in a similar fashion to
HoloFoodR, allows simple interaction with MGnify API.

```{r}
#| label: get_metagenomic_samples
#| eval: false

# Create MGnify object
mg <- MgnifyClient(
    useCache = TRUE,
    cacheDir = ".MGnifyR_cache"
)

# Select only metagenomic_amplicon sample type
metagenomic_salmon_samples <- salmon_samples |>
    filter(sample_type == "metagenomic_amplicon")

# Search for sample IDs in MGnify database
salmon_analysis_ids <- searchAnalysis(
    mg,
    type = "samples",
    metagenomic_salmon_samples[["accession"]]
)
```

`salmon_analysis_ids` character vector holds associations of HoloFood
metagenomic amplicon accession numbers (SAMEAxxxxxx) to their counterparts in
MGnify database (MGYAxxxxxx).

```{r}
#| label: get_metagenomic
#| eval: false

# Get metagenomic taxonomic data for salmon from MGnify
tse <- MGnifyR::getResult(
    mg,
    accession = salmon_analysis_ids,
    get.func = FALSE
)
```

```{r}
#| label: save_metagenomic
#| eval: false
#| echo: false

# Save salmon metagenomic TreeSE
path <- system.file(
  "extdata", "salmon_metagenomic_tse.RDS", package = "HoloFoodR")
saveRDS(object = mae, file = path)
```


```{r}
#| label: read_metagenomic
#| echo: false

# Read in salmon metagenomic TreeSE object
path <- system.file(
  "extdata", "salmon_metagenomic_tse.RDS", package = "HoloFoodR")
tse <- readRDS(file = path)
```

Data fetched from MGnify has MGnify-specific identifiers. We have to first
rename samples with HoloFood specific ID and then add the data to
`MultiAssayExperiment` combining all the data.

```{r}
#| label: add_metagenomic_data

# Add MGnify results to HoloFood data
mae <- addMGnify(tse, mae)
```

Now we have retrieved all the data that we are interested in this workflow.

## Data preprocess

Data cleaning is one of the most time-consuming and most important steps in
data analysis. For instance, we need to handle missing data, transform data
assays, and agglomerate the data.

In the next steps, we will:

1. Filter data
2. Ensure that the data is in correct format for analysis
3. Agglomerate data
4. Transform data

### Wrangle the data

Below we see upset plot that summarizes the available experiments and how
samples overlap between them.

```{r}
#| label: upsetplot

upset_plot <- upsetSamples(mae)
upset_plot
```

For demonstration purposes, we will focus on investigating fatty acids and
metagenomic data (`fatty_acids_mg` and `metagenomic_amplicon`) within the trial
A performed by the HoloFood consortium. This trial the health effects of
fermented seaweed added to the diet of salmons.

```{r}
#| label: preprocess1

# Harmonize experiment names
names(mae) <- names(mae) |>
    tolower() |>
    gsub(pattern = " ", replacement = "_")

# Fetch only experiments that we need
mae <- mae[, , c("fatty_acids_mg", "metagenomic")]
names(mae) <- c("fatty_acids", "metagenomic")
# Filter MAE object to include only Trial A
mae <- mae[, colData(mae)[["Trial code"]] == "SA", ]
```

Some values of fatty acids are under detection thresholds. We assume them to be
zeroes. Moreover, the data includes feature that just states from where the
fatty acids where collected. We remove this feature so that the assay contains
only numeric values.

```{r}
#| label: preprocess2

# From metabolomic data, remove organ-fatty acids row because it only contains a
# string value "muscle" which denotes where the sample was drawn.
tse <- mae[[1]]
tse <- tse[!(rowData(tse)[["marker.name"]] %in% c("Organ-fatty acids")), ]
mae[[1]] <- tse

# Transform matrix to numeric. Some values are "< 0.01"
# If a number is < 0.01, assume it to be 0
assay <- assay(mae[[1]], "counts")
assay[assay == "<0.01"] <- 0
assay <- apply(assay, c(1, 2), function(x) as.numeric(gsub(",", ".", x)))

# Reassign assay back to MAE
assay(mae[[1]], "counts") <- assay
```

Moreover, we wrangle the sample metadata so that it includes all necessary
information.

```{r}
#| label: preprocess3

# Add treatment groups
colData(mae)[["treatment_group"]] <- ifelse(
  colData(mae)[["Treatment concentration"]]>0, "treatment", "control")

# Add salmon host mass to MAE colData
colData(mae[[2]])[["host.gutted.mass"]] <- as.numeric(
    colData(mae[[2]])[["host.gutted.mass"]])
colData(mae[[2]])[["mass_scaled"]] <- scale(
  colData(mae[[2]])[["host.gutted.mass"]])
merged_data <- merge(
    x = colData(mae),
    y = colData(mae[[2]])[, c("animal", "host.gutted.mass", "mass_scaled")],
    all.x = TRUE, by = "animal"
)
rownames(merged_data) <- merged_data[["animal"]]
# Order and add back
colData(mae) <- merged_data[match(
    colData(mae)[["animal"]], merged_data[["animal"]]), ]

# Add animal metadata to separate experiments
mae[[1]] <- getWithColData(mae, 1)
mae[[2]] <- getWithColData(mae, 2)
```

As we want to analyze the effect of treatment, we must analyze
salmons at day 60 because these salmons were sampled after experimental period.
In contrast, day-0 salmons that were sampled immediately in the beginning of
the trial.

We select those samples that are drawn in day 60.

```{r}
#| label: preprocess4

# Subset by taking only samples from day 60
samples <- colnames(mae[[2]])[mae[[2]]$trial.timepoint == 60]
mae[[2]] <- mae[[2]][, samples]
# Make sure that we have same set of samples from each experiment
mae <- intersectColumns(mae)
```

Now the data includes samples that are taken in day 60. Each animal have both
microbiota data and fatty acid data.

### Filtering and agglomeration

Next, we can agglomerate features by prevalence to reduce the number of
low-abundant taxa and contaminants.

First, we visualize prevalence distribution of taxa with a histogram to decide
the prevalence threshold to use. We use 0.1% detection level to filter out
extremely low-abundant genera.

```{r}
#| label: prevalence_histgoram

# Add relative transforrmed data
mae[[2]] <- transformAssay(mae[[2]], method = "relabundance")

prevalence <- getPrevalence(
    mae[[2]],
    rank = "Genus",
    assay.type = "relabundance",
    na.rm = TRUE,
    sort = TRUE,
    detection = 0.1 / 100
)

# Exclude microbes with 0 prevalence
prevalence <- prevalence[prevalence != 0]

hist(prevalence, main = "Prevalence of microbial taxa",
     xlab = "Prevalence")
```

We can also look at the raw prevalence numbers

```{r}
#| label: display_prevalence_numbers

# Sort prevalence in decreasing order
sort(prevalence, decreasing = TRUE) |> head(10)
```

_Mycoplasma_ is present in all samples, which is not surprising as this genus
was found to be one of the most common in salmon intestine (see @zarkasiPyrosequencingbasedCharacterizationGastrointestinal2014).

We use thresholds 20% and 0.1% for prevalence and detection, respectively.

```{r}
#| label: agglomerate_prevalence

# Agglomerate by prevalence by genus
altExp(mae[[2]], "prev_genus") <- agglomerateByPrevalence(
    mae[[2]],
    assay.type = "relabundance",
    rank = "Genus",
    prevalence = 0.2 / 100,
    detection = 0.1 / 100
)
```

Moreover, we agglomerate the data to include only 19 most abundant genera.

```{r}
#| label: select_core

tse <- agglomerateByRank(mae[[2]], rank = "Genus")
core <- getTop(tse, top = 19)
# Get vector that tells whether taxa belongs to core taxa or not
core_names <- rownames(tse)
core_names[ !core_names %in% core ] <- "other"
# Agglomerate based on core taxa
rowData(tse)[["core"]] <- core_names
altExp(mae[[2]], "core_genus") <- agglomerateByVariable(
  tse, by = "rows", group = "core")
```

Due to the limited number of samples, we also filter the fatty acid data to
include only those fatty acids that show variation within the dataset. The
rationale is that if a fatty acid does not vary, it cannot exhibit differences
between groups. We can find out a good threshold for the cut-off with a
histogram of standard deviations of fatty acid abundances.

```{r}
#| label: filter_fatty1

rowData(mae[[1]])[["sd"]] <- rowSds(assay(mae[[1]], "counts"), na.rm = TRUE)
hist(rowData(mae[[1]])[["sd"]],
    main = "Standard deviation distribution of fatty acid abundances",
    xlab = "Standard deviation",
    breaks = 20)
```

We apply a filtering threshold of 0.5 to exclude fatty acids that do not
exhibit sufficient variation in the dataset. Since most standard deviations
are below 1, it is a reasonable number if we do not want to exclude too many
fatty acids.

```{r}
#| label: filter_fatty2

mae[[1]] <- mae[[1]][ rowData(mae[[1]])[["sd"]] > 0.5, ]
```

For more detailed analysis, we pick certain fatty acids that have
well-established biological relevance. These include 22:6n-3, 20:5n-3, 18:3n-3,
20:4n-6, 18:2n-6, 18:1n-9, 16:0, and 18:0.

```{r}
#| label: filter_fatty3

relevant_fatty_acids <- c(
  "Docosahexaenoic acid 22:6n-3 (DHA)",
  "Eicosapentaenoic acid 20:5n-3 (EPA)",
  "Alpha-Linolenic acid 18:3n-3",
  "Arachidonic acid 20:4n-6 (ARA)",
  "Linoleic acid 18:2n-6",
  "Oleic acid 18:1n-9",
  "Palmitic acid 16:0",
  "Stearic acid 18:0"
)
altExp(mae[[1]], "relevant") <- mae[[1]][
    rownames(mae[[1]]) %in% relevant_fatty_acids, ]
```

### Transformation

We transform metagenomic counts with relative transformation and
centered log-ratio method to tackle the compositional data
(see @quinnFieldGuideCompositional2019).

```{r}
#| label: transformation_metagenomics

# Transform microbiome with centered log-ratio method
mae[[2]] <- transformAssay(
    mae[[2]],
    assay.type = "counts",
    method = "relabundance",
    MARGIN = "cols"
)
mae[[2]] <- transformAssay(
    mae[[2]],
    assay.type = "counts",
    method = "clr",
    pseudocount = TRUE,
    MARGIN = "cols"
)
```

Fatty acid data is already compositional as it is measured as concentration
(mg/g). We apply a log10 transformation to address
skewness in the data. Finally, the data is standardized to ensure all features
are on a comparable scale.

```{r}
#| label: transform_fatty

mae[[1]] <- transformAssay(
    mae[[1]],
    assay.type = "counts",
    method = "log10",
    MARGIN = "cols"
)
mae[[1]] <- transformAssay(
    mae[[1]],
    assay.type = "log10",
    method = "standardize",
    MARGIN = "rows"
)
```

## Fatty acids, mass and treatment

### Does treatment affect fatty acids?

We aim to investigate whether the treatment is associated with changes in fatty
acid concentrations. To begin, we assess the correlation between treatment
levels and the concentrations of specific fatty acids

```{r}
#| label: fatty_and_treatment

# Get fatty acids that we are going to test
tse <- altExp(mae[[1]], "relevant")

# For each fatty acid, test if the groups are different
res <- lapply(rownames(tse), function(feat){
    res <- pairwise.wilcox.test(
      assay(tse, "counts")[feat, ], tse[["treatment_group"]],
      p.adjust.method = "none")
    return(res)
})
# Combine results and adjust p -values
res <- do.call(rbind, res) |> as.data.frame()
res[["p.value"]] <- p.adjust(res[["p.value"]], method = "fdr")
res |> datatable()
```

As observed, the treatment appears to have no significant effect on fatty acid
concentrations.

### Is there association between salmon size and fatty acids?

Additionally, it is worth exploring whether the size of the
salmon is associated with variations in fatty acid levels.

```{r}
#| label: fatty_and_weight1

# Calculate correlation
res <- getCrossAssociation(
    mae, experiment1 = 1, experiment2 = 1, altexp1 = "relevant",
    assay.type1 = "standardize", col.var2 = "mass_scaled",
    test.signif = TRUE, method = "pearson"
)
res
```

There appears to be a notable effect: larger salmon seem to have slightly higher
concentrations of fatty acids. This trend can be further illustrated through
plots.

```{r}
#| label: fatty_and_weight2

p <- plotExpression(
    altExp(mae[[1]], "relevant"),
    features = rownames(altExp(mae[[1]], "relevant")),
    x = "mass_scaled",
    assay.type = "standardize",
    scales = "free"
    ) +
  geom_smooth(method = "lm")
p
```

As a result of this initial analysis of fatty acids, we discovered that the
size of the salmon appears to have some association with fatty acid
concentrations, although this association is not statistically significant.

## Microbiota, mass and treatment

### Microbial composition

As a first step in analysing microbiota data, we summarize the microbial
composition with relative abundance barplot.

```{r}
#| label: abundance_plot

p <- plotAbundance(
    altExp(mae[[2]], "core_genus"),
    assay.type = "relabundance",
    col.var = "treatment_group",
    facet.cols = TRUE, scales = "free_x"
    )
p
```

Salmon gut seems to be dominated by either genus _Mycoplasma_ or
_Photobacterium_.

### Is alpha diversity associated with treatment or mass of salmon?

Next, let's calculate the Shannon alpha diversity index and
test whether diversity is associated with both size and treatment.

```{r}
#| label: alpha

# Calculate alpha diversity
mae[[2]] <- addAlpha(mae[[2]])

# Plot with metadata
p1 <- plotColData(mae[[2]], x = "mass_scaled", y = "shannon_diversity") +
  geom_smooth(method = "lm") +
  ggtitle("Shannon diversity association\nwith body mass") +
  xlab("Scaled body mass") + ylab("Shannon diversity index")
p2 <- plotColData(mae[[2]], x = "treatment_group", y = "shannon_diversity") +
    geom_signif(comparisons = list(c("control", "treatment")), ) +
  ggtitle("Shannon diversity association\nwith treatment") +
  xlab("Treatment group") + ylab("Shannon diversity index")

p1 + p2
```

Once again, we observe a potential effect, this time between treatment and
microbial diversity, although it is not statistically significant. Additionally,
there is no association between diversity and mass.

Let's analyze whether we can find similar effect with beta diversity. Here we
perform Principal Coordinate Analysis (PCoA) with Bray-Curtis dissimilarity.

```{r}
#| label: pcoa_beta_diversity

# Run PCoA
mae[[2]] <- runMDS(
    mae[[2]],
    FUN = getDissimilarity,
    method = "bray",
    assay.type = "relabundance"
)

# Display dissimilarity on a plot
p <- plotReducedDim(mae[[2]], "MDS", colour_by = "treatment_group") +
  ggtitle("PCoA plot of Bray-Curtis dissimilarity by treatment group")
p
```

It seems that there are handful of samples that differ from others. However,
this effect seems not to be associated with the treatment group.

## Multi-omics integration

Multi-omic factor analysis (MOFA) (see @Argelaguet2020).  allows us to discover latent factors that
underlie the biological differences by taking in consideration 2 or more omic
assays. To cite the original authors, "MOFA can be viewed as a statistically
rigorous generalization of (sparse) principal component analysis (PCA) to
multi-omics data".

By applying MOFA analysis, our goal is to determine whether metagenomics and
fatty acids exhibit shared variability, ultimately assessing whether the
microbial community is associated with fatty acid composition.

```{r}
#| label: fit_mofa
#| output: false

mae_temp <- mae
mae_temp[[2]] <- altExp(mae_temp[[2]], "prev_genus")

# Extract only transformed metagenomic assays for MOFA analysis
assays(mae_temp[[1]]) <- assays(mae_temp[[1]])[
    names(assays(mae_temp[[1]])) %in% c("standardize")
]
assays(mae_temp[[2]]) <- assays(mae_temp[[2]])[
    names(assays(mae_temp[[2]])) %in% c("counts")
]

# Transform MAE object to MOFA model
model <- create_mofa_from_MultiAssayExperiment(mae_temp)

# Set model's options
model_opts <- get_default_model_options(model)
model_opts$num_factors <- 5
model_opts$likelihoods[[2]] <- "poisson"
train_opts <- get_default_training_options(model)
train_opts$maxiter <- 20000

# Change convergence mode to slightly improve accuracy
train_opts$convergence_mode <- "slow"

# Prepare MOFA model
model <- prepare_mofa(
    object = model,
    model_options = model_opts,
    training_options = train_opts
)

# Train model
model <- run_mofa(model, use_basilisk = TRUE)
```

Next, we will plot the variances explained by each factor.

```{r}
#| label: var_factor1

p <- plot_variance_explained(model) +
    labs(title = "Captured variance of fatty acid and metagenomic\n",
        "datasets across 5 factors") +
    theme(plot.title.position = "plot")
p
```

Factor 1 captures mostly the variance in the metagenomics data.
Factor 2 captures the shared variability between the metagenomics and fatty acid
data. While the first factor captures the highest amount of variability in
metagenomics data, almost as much variability remains to be captured by the
second factor.

Before exploring the shared variability, we first examine which taxa's
variability is captured by Factor 1.

```{r}
#| label: plot_factor1
#| fig-width: 10
#| fig-height: 8

p1 <- plot_top_weights(model, view = 1, factors = 1, nfeatures = 25) +
  labs(title = "Fatty acids")
p2 <- plot_top_weights(model, view = 2, factors = 1, nfeatures = 25) +
  labs(title = "Microbiota")

p1 + p2
```

From the plot above, we can see that the first factor captures mostly the
variability in _Mycoplasma_.

```{r}
#| label: plot_factor2
#| fig-width: 10
#| fig-height: 8

p1 <- plot_top_weights(model, view = 1, factors = 2, nfeatures = 25) +
  labs(title = "Fatty acids")
p2 <- plot_top_weights(model, view = 2, factors = 2, nfeatures = 25) +
  labs(title = "Microbiota")

p1 + p2
```

From the microbes, especially _Photobacterium_, has a positive association with
Factor 2. Many fatty acids also show significant weights, with Linoleic acid
18:2n-6 and sum of omega-6 fatty acids having
positive weights, among others. This indicates that as the abundance of
Photobacterium increases, there is a corresponding increase in levels of
these aforementioned fatty acids.

## Conclusions

The present case study has demonstrated how easy and fast it is to
download large dataset and transform the data into a `MultiAssayExperiment`,
which in turn gives the researchers access to an extensive plethora of
downstream tools, such `mia` and `MOFA2` that can be used to pre-process and
visualize the multi-omics data.

```{r session_info}
sessionInfo()
```
