<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-07-11">

<title>HoloFoodR R package — case study</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="case_study_files/libs/clipboard/clipboard.min.js"></script>
<script src="case_study_files/libs/quarto-html/quarto.js"></script>
<script src="case_study_files/libs/quarto-html/popper.min.js"></script>
<script src="case_study_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="case_study_files/libs/quarto-html/anchor.min.js"></script>
<link href="case_study_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="case_study_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="case_study_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="case_study_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="case_study_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">HoloFoodR R package — case study</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 11, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p><a href="https://www.holofooddata.org">HoloFood database</a> is a large collection of holo-omic and multi-omic data from two animal systems, chicken and salmon. It was created by the HoloFood consortium. One of its objectives is to study the interactions between animal systems, their microbiomes, and feed additives to optimize the diet strategies of these farm animals. For make it easily accessible, the creators provide an Application Programming Interface (API) that permits interaction with programming languages, such as R or Python.</p>
<p>We improve the accessibility by creating the HoloFoodR package that simplifies API interaction and assists translating raw database data into R/Bioconductor data containers, connecting to a vast ecosystem of bioinformatics R packages.</p>
<p>We do not aim to demonstrate HoloFoodR in isolation from the rest of the ecosystem, but to showcase the possibility of data integration from other databases, such as MGnify, which holds metagenomic data. Furthermore, we provide a workflow from data exploration to advanced machine learning and multi-omics, offering a practical example for readers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of packages that we need</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>packages <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ggplot2"</span>, <span class="st">"knitr"</span>, <span class="st">"mia"</span>, <span class="st">"dplyr"</span>, <span class="st">"miaViz"</span>, <span class="st">"MOFA2"</span>,  <span class="st">"patchwork"</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"reticulate"</span>, <span class="st">"MGnifyR"</span>, <span class="st">"IntegratedLearner"</span>, <span class="st">"SuperLearner"</span>, <span class="st">"bartMachine"</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"mcmcplots"</span>, <span class="st">"tidyverse"</span>, <span class="st">"shadowtext"</span>, <span class="st">"ComplexHeatmap"</span>, <span class="st">"stringr"</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cowplot"</span>, <span class="st">"caret"</span>, <span class="st">"scater"</span>, <span class="st">"ggsignif"</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get packages that are already installed installed</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>packages_already_installed <span class="ot">&lt;-</span> packages[ packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>() ]</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get packages that need to be installed</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>packages_need_to_install <span class="ot">&lt;-</span> <span class="fu">setdiff</span>( packages, packages_already_installed )</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Loads BiocManager into the session. Install it if it not already installed.</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>) ){</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: BiocManager</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co"># If there are packages that need to be installed, installs them with BiocManager</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Updates old packages</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="fu">length</span>(packages_need_to_install) <span class="sc">&gt;</span> <span class="dv">0</span> ) {</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>   <span class="fu">install</span>(packages_need_to_install, <span class="at">ask =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Load all packages into session. Stop if there are packages that were not</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co"># successfully loaded</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>pkgs_not_loaded <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">sapply</span>(packages, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: mia</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: stats4</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     findMatches</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: IRanges</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: Biobase</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     rowMedians</span></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: TreeSummarizedExperiment</span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: XVector</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     strsplit</span></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: MultiAssayExperiment</span></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: dplyr</span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:mia':</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     full_join, inner_join, left_join, right_join</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:Biostrings':</span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     collapse, intersect, setdiff, setequal, union</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:XVector':</span></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     slice</span></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:Biobase':</span></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     combine</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:GenomicRanges':</span></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     intersect, setdiff, union</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:GenomeInfoDb':</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     intersect</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:IRanges':</span></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     collapse, desc, intersect, setdiff, slice, union</span></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     combine, intersect, setdiff, union</span></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:matrixStats':</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     count</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     filter, lag</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: miaViz</span></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: ggraph</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: MOFA2</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'MOFA2'</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:stats':</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     predict</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: patchwork</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: reticulate</span></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: MGnifyR</span></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: IntegratedLearner</span></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: SuperLearner</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: nnls</span></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: gam</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: splines</span></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: foreach</span></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'foreach'</span></span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:MGnifyR':</span></span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     getResult</span></span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loaded gam 1.22-3</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Super Learner</span></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Version: 2.0-29</span></span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Package created on 2024-02-06</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: bartMachine</span></span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: rJava</span></span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: bartMachineJARs</span></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: randomForest</span></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; randomForest 4.7-1.1</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Type rfNews() to see new features/changes/bug fixes.</span></span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'randomForest'</span></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     combine</span></span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:Biobase':</span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     combine</span></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     combine</span></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:ggplot2':</span></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     margin</span></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: missForest</span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Welcome to bartMachine v1.3.4.1! You have 0.54GB memory available.</span></span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; If you run out of memory, restart R, and use e.g.</span></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'options(java.parameters = "-Xmx5g")' for 5GB of RAM before you call</span></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'library(bartMachine)'.</span></span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: mcmcplots</span></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: coda</span></span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: tidyverse</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──</span></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✔ forcats   1.0.0     ✔ stringr   1.5.1</span></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✔ lubridate 1.9.3     ✔ tibble    3.2.1</span></span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✔ purrr     1.0.2     ✔ tidyr     1.3.1</span></span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✔ readr     2.1.5     </span></span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ lubridate::%within%()    masks IRanges::%within%()</span></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ purrr::accumulate()      masks foreach::accumulate()</span></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::collapse()        masks Biostrings::collapse(), IRanges::collapse()</span></span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ randomForest::combine()  masks dplyr::combine(), Biobase::combine(), BiocGenerics::combine()</span></span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ purrr::compact()         masks XVector::compact()</span></span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::count()           masks matrixStats::count()</span></span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::desc()            masks IRanges::desc()</span></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ tidyr::expand()          masks S4Vectors::expand()</span></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::filter()          masks stats::filter()</span></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::first()           masks S4Vectors::first()</span></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::full_join()       masks mia::full_join()</span></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::inner_join()      masks mia::inner_join()</span></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::lag()             masks stats::lag()</span></span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::left_join()       masks mia::left_join()</span></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ randomForest::margin()   masks ggplot2::margin()</span></span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ BiocGenerics::Position() masks ggplot2::Position(), base::Position()</span></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ purrr::reduce()          masks GenomicRanges::reduce(), IRanges::reduce()</span></span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::rename()          masks S4Vectors::rename()</span></span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::right_join()      masks mia::right_join()</span></span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ lubridate::second()      masks S4Vectors::second()</span></span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ lubridate::second&lt;-()    masks S4Vectors::second&lt;-()</span></span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ dplyr::slice()           masks XVector::slice(), IRanges::slice()</span></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ✖ purrr::when()            masks foreach::when()</span></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</span></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: shadowtext</span></span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: ComplexHeatmap</span></span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: grid</span></span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'grid'</span></span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:Biostrings':</span></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     pattern</span></span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ComplexHeatmap version 2.20.0</span></span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; If you use it in published research, please cite either one:</span></span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     genomic data. Bioinformatics 2016.</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The new InteractiveComplexHeatmap package can directly export static </span></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; This message can be suppressed by:</span></span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: cowplot</span></span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-268"><a href="#cb1-268" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'cowplot'</span></span>
<span id="cb1-269"><a href="#cb1-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:lubridate':</span></span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     stamp</span></span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:patchwork':</span></span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     align_plots</span></span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-281"><a href="#cb1-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: caret</span></span>
<span id="cb1-282"><a href="#cb1-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-283"><a href="#cb1-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: lattice</span></span>
<span id="cb1-284"><a href="#cb1-284" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-285"><a href="#cb1-285" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-286"><a href="#cb1-286" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'caret'</span></span>
<span id="cb1-287"><a href="#cb1-287" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-288"><a href="#cb1-288" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-289"><a href="#cb1-289" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:purrr':</span></span>
<span id="cb1-290"><a href="#cb1-290" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-291"><a href="#cb1-291" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     lift</span></span>
<span id="cb1-292"><a href="#cb1-292" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-293"><a href="#cb1-293" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-294"><a href="#cb1-294" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:mia':</span></span>
<span id="cb1-295"><a href="#cb1-295" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-296"><a href="#cb1-296" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     cluster</span></span>
<span id="cb1-297"><a href="#cb1-297" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-298"><a href="#cb1-298" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-299"><a href="#cb1-299" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: scater</span></span>
<span id="cb1-300"><a href="#cb1-300" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-301"><a href="#cb1-301" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: scuttle</span></span>
<span id="cb1-302"><a href="#cb1-302" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-303"><a href="#cb1-303" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: ggsignif</span></span>
<span id="cb1-304"><a href="#cb1-304" aria-hidden="true" tabindex="-1"></a>pkgs_not_loaded <span class="ot">&lt;-</span> <span class="fu">names</span>(pkgs_not_loaded)[ pkgs_not_loaded ]</span>
<span id="cb1-305"><a href="#cb1-305" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="fu">length</span>(pkgs_not_loaded) <span class="sc">&gt;</span> <span class="dv">0</span> ){</span>
<span id="cb1-306"><a href="#cb1-306" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Error in loading the following packages into the session: '"</span>, <span class="fu">paste0</span>(pkgs_not_loaded, <span class="at">collapse =</span> <span class="st">"', '"</span>), <span class="st">"'"</span>)</span>
<span id="cb1-307"><a href="#cb1-307" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Provide Java with RAM</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">java.parameters =</span> <span class="st">"-Xmx8g"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-data" class="level2">
<h2 class="anchored" data-anchor-id="import-data">Import data</h2>
<p>We start the workflow from data retrieval. We will use <a href="https://www.holofooddata.org/samples/?animal__system=salmon">salmon data</a> and its associated fatty acid and metagenomic amplicon as an example.</p>
<section id="retrieve-holofood-data" class="level3">
<h3 class="anchored" data-anchor-id="retrieve-holofood-data">Retrieve HoloFood data</h3>
<p>First of all, we have to query the HoloFood database to retrieve the salmon accession numbers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get salmon samples</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>salmons <span class="ot">&lt;-</span> HoloFoodR<span class="sc">::</span><span class="fu">doQuery</span>(<span class="st">"animals"</span>, <span class="at">system =</span> <span class="st">"salmon"</span>, <span class="at">use.cache =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get only the data that has both metagenomic amplicion and fatty acid</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># data</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>salmons <span class="ot">&lt;-</span> salmons <span class="sc">|&gt;</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(fatty_acids <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> metagenomic_amplicon <span class="sc">==</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(salmons)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] "accession"            "system"               "canonical_url"       </span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] "fatty_acids"          "heavy_metals"         "host_genomic"        </span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] "iodine"               "metabolomic"          "metagenomic_amplicon"</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] "metagenomic_assembly" "metatranscriptomic"   "transcriptomic"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we can retrieve the data associated with each salmon.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get salmon data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>salmon_data <span class="ot">&lt;-</span> HoloFoodR<span class="sc">::</span><span class="fu">getData</span>(</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">accession.type =</span> <span class="st">"animals"</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">accession =</span> salmons[[<span class="st">"accession"</span>]],</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">use.cache =</span> <span class="cn">TRUE</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get salmon samples</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>salmon_samples <span class="ot">&lt;-</span> salmon_data[[<span class="st">"samples"</span>]]</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sample IDs</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>salmon_sample_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(salmon_samples[[<span class="st">"accession"</span>]])</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(salmon_sample_ids)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "SAMEA112750580" "SAMEA112950027" "SAMEA112953473" "SAMEA112750416"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] "SAMEA112950208" "SAMEA112951431"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The data returned above is a list of all sample accession numbers that are associated with all salmons. For example, metagenomic amplicon samples, such as <a href="https://www.holofooddata.org/sample/SAMEA112750580">SAMEA112750580</a> or genomic sample, such as <a href="https://www.holofooddata.org/sample/SAMEA9068691">SAMEA9068691</a>.</p>
<p>We can use these accession numbers to fetch the data associated with each sample type and store them as <code>experiments</code> in a MultiAssayExperiment (MAE) object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get salmon &lt;- experiments as MAE object</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> HoloFoodR<span class="sc">::</span><span class="fu">getResult</span>(</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  salmon_sample_ids,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">use.cache =</span> <span class="cn">TRUE</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Data for the following samples cannot be found. The sample types are metagenomic_amplicon, metabolomic, heavy_metals, transcriptomic, metagenomic_assembly, metatranscriptomic and host_genomic. (Note that metagenomic assemblies can be found from the MGnify database. See MGnifyR package.):</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'SAMEA112750580', 'SAMEA112953473', 'SAMEA112750416', 'SAMEA112951431', 'SAMEA112953296', 'SAMEA112750432', 'SAMEA112951736', 'SAMEA112953315', 'SAMEA112750492', 'SAMEA112750484', 'SAMEA112951604', 'SAMEA112953361', 'SAMEA112750452', 'SAMEA112953336', 'SAMEA112369595', 'SAMEA112642746', 'SAMEA112951566', 'SAMEA112952768', 'SAMEA7687944', 'SAMEA8141602', 'SAMEA9068721', 'SAMEA112750502', 'SAMEA112951550', 'SAMEA112953389', 'SAMEA112750472', 'SAMEA112951513', 'SAMEA112953355', 'SAMEA112369556', 'SAMEA112642736', 'SAMEA112951704', 'SAMEA112952725', 'SAMEA7687900', 'SAMEA8141559', 'SAMEA9068669', 'SAMEA112369594', 'SAMEA112642745', 'SAMEA112951536', 'SAMEA112952767', 'SAMEA7687943', 'SAMEA8141601', 'SAMEA9068720', 'SAMEA112750490', 'SAMEA112953376', 'SAMEA112750463', 'SAMEA112951749', 'SAMEA112953341', 'SAMEA112750501', 'SAMEA112953386', 'SAMEA112750579', 'SAMEA112951380', 'SAMEA112953472', 'SAMEA112369600', 'SAMEA112642749', 'SAMEA112951717', 'SAMEA112952773', 'SAMEA7687948', 'SAMEA8141607', 'SAMEA9068725', 'SAMEA112750413', 'SAMEA112951337', 'SAMEA112953290', 'SAMEA112750430', 'SAMEA112951676', 'SAMEA112953312', 'SAMEA112750527', 'SAMEA112750473', 'SAMEA112369604', 'SAMEA112642751', 'SAMEA112951796', 'SAMEA112952777', 'SAMEA7687952', 'SAMEA8141611', 'SAMEA9068729', 'SAMEA112750491', 'SAMEA112750536', 'SAMEA112951556', 'SAMEA112953421', 'SAMEA112750575', 'SAMEA112951801', 'SAMEA112953468', 'SAMEA112442004', 'SAMEA112642735', 'SAMEA112951644', 'SAMEA112952723', 'SAMEA7687898', 'SAMEA8141557', 'SAMEA9068677', 'SAMEA112750445', 'SAMEA112951446', 'SAMEA112953321', 'SAMEA112750429', 'SAMEA112953311', 'SAMEA112750437', 'SAMEA112369615', 'SAMEA112642755', 'SAMEA112951744', 'SAMEA112952791', 'SAMEA7687966', 'SAMEA8141625', 'SAMEA9068743', 'SAMEA112750523', 'SAMEA112953408', 'SAMEA112750510', 'SAMEA112750592', 'SAMEA112953483', 'SAMEA112369694', 'SAMEA112642776', 'SAMEA112951762', 'SAMEA112952871', 'SAMEA7688046', 'SAMEA8141706', 'SAMEA9068820', 'SAMEA112750453', 'SAMEA112951690', 'SAMEA112953337', 'SAMEA112750384', 'SAMEA112369621', 'SAMEA112642756', 'SAMEA112951355', 'SAMEA112952797', 'SAMEA7687972', 'SAMEA8141631', 'SAMEA9068747', 'SAMEA112369683', 'SAMEA112642772', 'SAMEA112951428', 'SAMEA112952860', 'SAMEA7688035', 'SAMEA8141695', 'SAMEA9068810', 'SAMEA112750397', 'SAMEA112951549', 'SAMEA112953273', 'SAMEA112750393', 'SAMEA112953266', 'SAMEA112750594', 'SAMEA112951597', 'SAMEA112953488', 'SAMEA112750418', 'SAMEA112953299', 'SAMEA112750412', 'SAMEA112953287', 'SAMEA112750540', 'SAMEA112951647', 'SAMEA112953427', 'SAMEA112369648', 'SAMEA112642763', 'SAMEA112951602', 'SAMEA112952825', 'SAMEA7688000', 'SAMEA8141659', 'SAMEA9068775', 'SAMEA112750436', 'SAMEA112951794', 'SAMEA112953319', 'SAMEA112750573', 'SAMEA112953464', 'SAMEA112750391', 'SAMEA112953262', 'SAMEA112750526', 'SAMEA112951400', 'SAMEA112953411', 'SAMEA112369624', 'SAMEA112642757', 'SAMEA112951381', 'SAMEA112952800', 'SAMEA7687975', 'SAMEA8141634', 'SAMEA9068751', 'SAMEA112750420', 'SAMEA112750519', 'SAMEA112951760', 'SAMEA112953403', 'SAMEA112750455', 'SAMEA112369650', 'SAMEA112642764', 'SAMEA112951663', 'SAMEA112952827', 'SAMEA7688002', 'SAMEA8141661', 'SAMEA9068780', 'SAMEA112369669', 'SAMEA112642768', 'SAMEA112951637', 'SAMEA112952846', 'SAMEA7688021', 'SAMEA8141681', 'SAMEA9068793', 'SAMEA112750538', 'SAMEA112951586', 'SAMEA112953423', 'SAMEA112750542', 'SAMEA112953434', 'SAMEA112750520', 'SAMEA112951789', 'SAMEA112953405', 'SAMEA112750598', 'SAMEA112953495', 'SAMEA112750521', 'SAMEA112953406', 'SAMEA112750528', 'SAMEA112750411', 'SAMEA112953285', 'SAMEA112369543', 'SAMEA112642734', 'SAMEA112951405', 'SAMEA112952706', 'SAMEA7687882', 'SAMEA8141540', 'SAMEA9068662', 'SAMEA112750400', 'SAMEA112953276', 'SAMEA112750572', 'SAMEA112951713', 'SAMEA112953463', 'SAMEA112750415', 'SAMEA112951399', 'SAMEA112953294', 'SAMEA112750503', 'SAMEA112953390', 'SAMEA112750471', 'SAMEA112750448', 'SAMEA112953328', 'SAMEA112750409', 'SAMEA112951700', 'SAMEA112953282', 'SAMEA112750557', 'SAMEA112951438', 'SAMEA112953445', 'SAMEA112750456', 'SAMEA112750539', 'SAMEA112953426', 'SAMEA112369661', 'SAMEA112642765', 'SAMEA112951389', 'SAMEA112952838', 'SAMEA7688013', 'SAMEA8141673', 'SAMEA9068786', 'SAMEA112750464', 'SAMEA112953342', 'SAMEA112369557', 'SAMEA112642737', 'SAMEA112951763', 'SAMEA112952727', 'SAMEA7687902', 'SAMEA8141561', 'SAMEA9068679', 'SAMEA112750469', 'SAMEA112953351', 'SAMEA112369573', 'SAMEA112642742', 'SAMEA112951680', 'SAMEA112952745', 'SAMEA7687921', 'SAMEA8141579', 'SAMEA9068699', 'SAMEA112750543', 'SAMEA112951766', 'SAMEA112953435', 'SAMEA112369693', 'SAMEA112642775', 'SAMEA112951733', 'SAMEA112952870', 'SAMEA7688045', 'SAMEA8141705', 'SAMEA9068819', 'SAMEA112750489', 'SAMEA112951332', 'SAMEA112953375', 'SAMEA112750581', 'SAMEA112750434', 'SAMEA112951765', 'SAMEA112953317', 'SAMEA112750399', 'SAMEA112951579', 'SAMEA112953275', 'SAMEA112369560', 'SAMEA112642738', 'SAMEA112951500', 'SAMEA112952729', 'SAMEA7687904', 'SAMEA8141563', 'SAMEA9068681', 'SAMEA112750558', 'SAMEA112951501', 'SAMEA112953449', 'SAMEA112750449', 'SAMEA112951569', 'SAMEA112953329', 'SAMEA112750396', 'SAMEA112951518', 'SAMEA112953271', 'SAMEA112750518', 'SAMEA112953402', 'SAMEA112750563', 'SAMEA112750482', 'SAMEA112951574', 'SAMEA112953359', 'SAMEA112750487', 'SAMEA112953366', 'SAMEA112369668', 'SAMEA112642767', 'SAMEA112951606', 'SAMEA112952845', 'SAMEA7688020', 'SAMEA8141680', 'SAMEA9068791', 'SAMEA112750590', 'SAMEA112953481', 'SAMEA112369632', 'SAMEA112642760', 'SAMEA112951598', 'SAMEA112952808', 'SAMEA7687983', 'SAMEA8141642', 'SAMEA9068761', 'SAMEA112750435', 'SAMEA112953318', 'SAMEA112369662', 'SAMEA112642766', 'SAMEA112951421', 'SAMEA112952839', 'SAMEA7688014', 'SAMEA8141674', 'SAMEA9068787', 'SAMEA112369698', 'SAMEA112642777', 'SAMEA112951402', 'SAMEA112952875', 'SAMEA7688050', 'SAMEA8141710', 'SAMEA9068823', 'SAMEA112750468', 'SAMEA112951419', 'SAMEA112953350', 'SAMEA112750559', 'SAMEA112953452', 'SAMEA112369610', 'SAMEA112642754', 'SAMEA112951563', 'SAMEA112952785', 'SAMEA7687960', 'SAMEA8141619', 'SAMEA9068737', 'SAMEA112750402', 'SAMEA112750524', 'SAMEA112951368', 'SAMEA112953409', 'SAMEA112750555', 'SAMEA112953442', 'SAMEA112750593', 'SAMEA112953487', 'SAMEA112750454', 'SAMEA112953338', 'SAMEA112750467', 'SAMEA112953349', 'SAMEA112750578', 'SAMEA112953471', 'SAMEA112750554', 'SAMEA112951374', 'SAMEA112953441', 'SAMEA112750485', 'SAMEA112953362', 'SAMEA112369561', 'SAMEA112642739', 'SAMEA112951530', 'SAMEA112952730', 'SAMEA7687905', 'SAMEA8141564', 'SAMEA9068684', 'SAMEA112369597', 'SAMEA112642748', 'SAMEA112951627', 'SAMEA112952770', 'SAMEA8141604', 'SAMEA112369688', 'SAMEA112642773', 'SAMEA112951582', 'SAMEA112952865', 'SAMEA7688040', 'SAMEA8141700', 'SAMEA9068814', 'SAMEA112750505', 'SAMEA112953392', 'SAMEA112750507', 'SAMEA112951641', 'SAMEA112953395', 'SAMEA112750428', 'SAMEA112951615', 'SAMEA112953308', 'SAMEA112750535', 'SAMEA112953420', 'SAMEA112369633', 'SAMEA112642761', 'SAMEA112951629', 'SAMEA112952809', 'SAMEA7687984', 'SAMEA8141643', 'SAMEA9068759', 'SAMEA112750591', 'SAMEA112951506', 'SAMEA112953482', 'SAMEA112750483', 'SAMEA112953360', 'SAMEA112750470', 'SAMEA112951483', 'SAMEA112953354', 'SAMEA112750419', 'SAMEA112369602', 'SAMEA112642750', 'SAMEA112951738', 'SAMEA112952775', 'SAMEA7687950', 'SAMEA8141609', 'SAMEA9068727', 'SAMEA112369605', 'SAMEA112642753', 'SAMEA112951375', 'SAMEA112952779', 'SAMEA7687954', 'SAMEA8141613', 'SAMEA9068731', 'SAMEA112750595', 'SAMEA112953489', 'SAMEA112750544', 'SAMEA112953436', 'SAMEA112750537', 'SAMEA112953422', 'SAMEA112369691', 'SAMEA112642774', 'SAMEA112951673', 'SAMEA112952868', 'SAMEA7688043', 'SAMEA8141703', 'SAMEA9068818', 'SAMEA112369596', 'SAMEA112642747', 'SAMEA112951596', 'SAMEA112952769', 'SAMEA7687945', 'SAMEA8141603', 'SAMEA9068722', 'SAMEA112750414', 'SAMEA112953291', 'SAMEA112750446', 'SAMEA112953322', 'SAMEA112750597', 'SAMEA112951688', 'SAMEA1129534</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Save salmon MAE</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> mae, <span class="at">file =</span> <span class="st">"../inst/extdata/salmon_mae_without_mgnify.RDS"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename FATTY ACIDS MG to more readable format</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mae)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">names</span>(mae)[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fetch-metagenomic-data-from-mgnify" class="level3">
<h3 class="anchored" data-anchor-id="fetch-metagenomic-data-from-mgnify">Fetch metagenomic data from MGnify</h3>
<p>HoloFood database does not include the data for metagenomic data. This data can be retrieved from the <a href="https://www.ebi.ac.uk/metagenomics">MGnify portal</a>. For this purpose, we will use MGnifyR package, which in a similar fashion to HoloFoodR, allows simple interaction with MGnify API.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create MGnify object</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>mg <span class="ot">&lt;-</span> <span class="fu">MgnifyClient</span>(</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">useCache =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cacheDir =</span> <span class="st">".MGnifyR_cache"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only metagenomic_amplicon sample type</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>metagenomic_salmon_samples <span class="ot">&lt;-</span> salmon_samples[</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  salmon_samples[[<span class="st">"sample_type"</span>]] <span class="sc">==</span> <span class="st">"metagenomic_amplicon"</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Search for sample IDs in MGnify database</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>salmon_analysis_ids <span class="ot">&lt;-</span> <span class="fu">searchAnalysis</span>(</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  mg,</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"samples"</span>,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  metagenomic_salmon_samples[[<span class="st">"accession"</span>]]</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fetching analyses...</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750580: No data found.</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750580: No data found.</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750580</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">1</span>%</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750416: No data found.</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750416: No data found.</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750416</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=</span>                                                                     <span class="er">|</span>   <span class="dv">1</span>%</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750432: No data found.</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750432: No data found.</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750432</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=</span>                                                                     <span class="er">|</span>   <span class="dv">2</span>%</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750492: No data found.</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750492: No data found.</span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750492</span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750484: No data found.</span></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750484: No data found.</span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750484</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==</span>                                                                    <span class="er">|</span>   <span class="dv">3</span>%</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750452: No data found.</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750452: No data found.</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750452</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===</span>                                                                   <span class="er">|</span>   <span class="dv">4</span>%</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750502: No data found.</span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750502: No data found.</span></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750502</span></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750472: No data found.</span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750472: No data found.</span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750472</span></span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===</span>                                                                   <span class="er">|</span>   <span class="dv">5</span>%</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====</span>                                                                  <span class="er">|</span>   <span class="dv">5</span>%</span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====</span>                                                                  <span class="er">|</span>   <span class="dv">6</span>%</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750490: No data found.</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750490: No data found.</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750490</span></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750463: No data found.</span></span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750463: No data found.</span></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750463</span></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====</span>                                                                 <span class="er">|</span>   <span class="dv">7</span>%</span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750501: No data found.</span></span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750501: No data found.</span></span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750501</span></span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750579: No data found.</span></span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750579: No data found.</span></span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750579</span></span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======</span>                                                                <span class="er">|</span>   <span class="dv">8</span>%</span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750413: No data found.</span></span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750413: No data found.</span></span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750413</span></span>
<span id="cb7-128"><a href="#cb7-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-129"><a href="#cb7-129" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-130"><a href="#cb7-130" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======</span>                                                                <span class="er">|</span>   <span class="dv">9</span>%</span>
<span id="cb7-131"><a href="#cb7-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-132"><a href="#cb7-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750430: No data found.</span></span>
<span id="cb7-133"><a href="#cb7-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-134"><a href="#cb7-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750430: No data found.</span></span>
<span id="cb7-135"><a href="#cb7-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-136"><a href="#cb7-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750430</span></span>
<span id="cb7-137"><a href="#cb7-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-138"><a href="#cb7-138" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-139"><a href="#cb7-139" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======</span>                                                               <span class="er">|</span>  <span class="dv">10</span>%</span>
<span id="cb7-140"><a href="#cb7-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-141"><a href="#cb7-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750527: No data found.</span></span>
<span id="cb7-142"><a href="#cb7-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-143"><a href="#cb7-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750527: No data found.</span></span>
<span id="cb7-144"><a href="#cb7-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-145"><a href="#cb7-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750527</span></span>
<span id="cb7-146"><a href="#cb7-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-147"><a href="#cb7-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750473: No data found.</span></span>
<span id="cb7-148"><a href="#cb7-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-149"><a href="#cb7-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750473: No data found.</span></span>
<span id="cb7-150"><a href="#cb7-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-151"><a href="#cb7-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750473</span></span>
<span id="cb7-152"><a href="#cb7-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-153"><a href="#cb7-153" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-154"><a href="#cb7-154" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======</span>                                                               <span class="er">|</span>  <span class="dv">11</span>%</span>
<span id="cb7-155"><a href="#cb7-155" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-156"><a href="#cb7-156" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========</span>                                                              <span class="er">|</span>  <span class="dv">11</span>%</span>
<span id="cb7-157"><a href="#cb7-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-158"><a href="#cb7-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750491: No data found.</span></span>
<span id="cb7-159"><a href="#cb7-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-160"><a href="#cb7-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750491: No data found.</span></span>
<span id="cb7-161"><a href="#cb7-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-162"><a href="#cb7-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750491</span></span>
<span id="cb7-163"><a href="#cb7-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-164"><a href="#cb7-164" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-165"><a href="#cb7-165" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========</span>                                                              <span class="er">|</span>  <span class="dv">12</span>%</span>
<span id="cb7-166"><a href="#cb7-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-167"><a href="#cb7-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750536: No data found.</span></span>
<span id="cb7-168"><a href="#cb7-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-169"><a href="#cb7-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750536: No data found.</span></span>
<span id="cb7-170"><a href="#cb7-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-171"><a href="#cb7-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750536</span></span>
<span id="cb7-172"><a href="#cb7-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-173"><a href="#cb7-173" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-174"><a href="#cb7-174" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========</span>                                                             <span class="er">|</span>  <span class="dv">12</span>%</span>
<span id="cb7-175"><a href="#cb7-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-176"><a href="#cb7-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750575: No data found.</span></span>
<span id="cb7-177"><a href="#cb7-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-178"><a href="#cb7-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750575: No data found.</span></span>
<span id="cb7-179"><a href="#cb7-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-180"><a href="#cb7-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750575</span></span>
<span id="cb7-181"><a href="#cb7-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-182"><a href="#cb7-182" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-183"><a href="#cb7-183" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========</span>                                                             <span class="er">|</span>  <span class="dv">13</span>%</span>
<span id="cb7-184"><a href="#cb7-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-185"><a href="#cb7-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750445: No data found.</span></span>
<span id="cb7-186"><a href="#cb7-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-187"><a href="#cb7-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750445: No data found.</span></span>
<span id="cb7-188"><a href="#cb7-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-189"><a href="#cb7-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750445</span></span>
<span id="cb7-190"><a href="#cb7-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-191"><a href="#cb7-191" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-192"><a href="#cb7-192" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========</span>                                                            <span class="er">|</span>  <span class="dv">14</span>%</span>
<span id="cb7-193"><a href="#cb7-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-194"><a href="#cb7-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750429: No data found.</span></span>
<span id="cb7-195"><a href="#cb7-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-196"><a href="#cb7-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750429: No data found.</span></span>
<span id="cb7-197"><a href="#cb7-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-198"><a href="#cb7-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750429</span></span>
<span id="cb7-199"><a href="#cb7-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-200"><a href="#cb7-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750437: No data found.</span></span>
<span id="cb7-201"><a href="#cb7-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-202"><a href="#cb7-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750437: No data found.</span></span>
<span id="cb7-203"><a href="#cb7-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-204"><a href="#cb7-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750437</span></span>
<span id="cb7-205"><a href="#cb7-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-206"><a href="#cb7-206" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-207"><a href="#cb7-207" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========</span>                                                            <span class="er">|</span>  <span class="dv">15</span>%</span>
<span id="cb7-208"><a href="#cb7-208" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-209"><a href="#cb7-209" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========</span>                                                           <span class="er">|</span>  <span class="dv">15</span>%</span>
<span id="cb7-210"><a href="#cb7-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-211"><a href="#cb7-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750523: No data found.</span></span>
<span id="cb7-212"><a href="#cb7-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-213"><a href="#cb7-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750523: No data found.</span></span>
<span id="cb7-214"><a href="#cb7-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-215"><a href="#cb7-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750523</span></span>
<span id="cb7-216"><a href="#cb7-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-217"><a href="#cb7-217" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-218"><a href="#cb7-218" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========</span>                                                           <span class="er">|</span>  <span class="dv">16</span>%</span>
<span id="cb7-219"><a href="#cb7-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-220"><a href="#cb7-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750510: No data found.</span></span>
<span id="cb7-221"><a href="#cb7-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-222"><a href="#cb7-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750510: No data found.</span></span>
<span id="cb7-223"><a href="#cb7-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-224"><a href="#cb7-224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750510</span></span>
<span id="cb7-225"><a href="#cb7-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-226"><a href="#cb7-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750592: No data found.</span></span>
<span id="cb7-227"><a href="#cb7-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-228"><a href="#cb7-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750592: No data found.</span></span>
<span id="cb7-229"><a href="#cb7-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-230"><a href="#cb7-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750592</span></span>
<span id="cb7-231"><a href="#cb7-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-232"><a href="#cb7-232" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-233"><a href="#cb7-233" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============</span>                                                          <span class="er">|</span>  <span class="dv">17</span>%</span>
<span id="cb7-234"><a href="#cb7-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-235"><a href="#cb7-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750453: No data found.</span></span>
<span id="cb7-236"><a href="#cb7-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-237"><a href="#cb7-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750453: No data found.</span></span>
<span id="cb7-238"><a href="#cb7-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-239"><a href="#cb7-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750453</span></span>
<span id="cb7-240"><a href="#cb7-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-241"><a href="#cb7-241" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-242"><a href="#cb7-242" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============</span>                                                         <span class="er">|</span>  <span class="dv">18</span>%</span>
<span id="cb7-243"><a href="#cb7-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-244"><a href="#cb7-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750384: No data found.</span></span>
<span id="cb7-245"><a href="#cb7-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-246"><a href="#cb7-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750384: No data found.</span></span>
<span id="cb7-247"><a href="#cb7-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-248"><a href="#cb7-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750384</span></span>
<span id="cb7-249"><a href="#cb7-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-250"><a href="#cb7-250" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-251"><a href="#cb7-251" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============</span>                                                         <span class="er">|</span>  <span class="dv">19</span>%</span>
<span id="cb7-252"><a href="#cb7-252" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-253"><a href="#cb7-253" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============</span>                                                        <span class="er">|</span>  <span class="dv">20</span>%</span>
<span id="cb7-254"><a href="#cb7-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-255"><a href="#cb7-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750397: No data found.</span></span>
<span id="cb7-256"><a href="#cb7-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-257"><a href="#cb7-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750397: No data found.</span></span>
<span id="cb7-258"><a href="#cb7-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-259"><a href="#cb7-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750397</span></span>
<span id="cb7-260"><a href="#cb7-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-261"><a href="#cb7-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750393: No data found.</span></span>
<span id="cb7-262"><a href="#cb7-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-263"><a href="#cb7-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750393: No data found.</span></span>
<span id="cb7-264"><a href="#cb7-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-265"><a href="#cb7-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750393</span></span>
<span id="cb7-266"><a href="#cb7-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-267"><a href="#cb7-267" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-268"><a href="#cb7-268" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============</span>                                                        <span class="er">|</span>  <span class="dv">21</span>%</span>
<span id="cb7-269"><a href="#cb7-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-270"><a href="#cb7-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750594: No data found.</span></span>
<span id="cb7-271"><a href="#cb7-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-272"><a href="#cb7-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750594: No data found.</span></span>
<span id="cb7-273"><a href="#cb7-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-274"><a href="#cb7-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750594</span></span>
<span id="cb7-275"><a href="#cb7-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-276"><a href="#cb7-276" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-277"><a href="#cb7-277" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============</span>                                                       <span class="er">|</span>  <span class="dv">21</span>%</span>
<span id="cb7-278"><a href="#cb7-278" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-279"><a href="#cb7-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750418: No data found.</span></span>
<span id="cb7-280"><a href="#cb7-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-281"><a href="#cb7-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750418: No data found.</span></span>
<span id="cb7-282"><a href="#cb7-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-283"><a href="#cb7-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750418</span></span>
<span id="cb7-284"><a href="#cb7-284" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-285"><a href="#cb7-285" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-286"><a href="#cb7-286" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============</span>                                                       <span class="er">|</span>  <span class="dv">22</span>%</span>
<span id="cb7-287"><a href="#cb7-287" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-288"><a href="#cb7-288" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750412: No data found.</span></span>
<span id="cb7-289"><a href="#cb7-289" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-290"><a href="#cb7-290" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750412: No data found.</span></span>
<span id="cb7-291"><a href="#cb7-291" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-292"><a href="#cb7-292" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750412</span></span>
<span id="cb7-293"><a href="#cb7-293" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-294"><a href="#cb7-294" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-295"><a href="#cb7-295" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================</span>                                                      <span class="er">|</span>  <span class="dv">22</span>%</span>
<span id="cb7-296"><a href="#cb7-296" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-297"><a href="#cb7-297" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750540: No data found.</span></span>
<span id="cb7-298"><a href="#cb7-298" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-299"><a href="#cb7-299" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750540: No data found.</span></span>
<span id="cb7-300"><a href="#cb7-300" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-301"><a href="#cb7-301" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750540</span></span>
<span id="cb7-302"><a href="#cb7-302" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-303"><a href="#cb7-303" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-304"><a href="#cb7-304" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================</span>                                                      <span class="er">|</span>  <span class="dv">23</span>%</span>
<span id="cb7-305"><a href="#cb7-305" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-306"><a href="#cb7-306" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750436: No data found.</span></span>
<span id="cb7-307"><a href="#cb7-307" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-308"><a href="#cb7-308" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750436: No data found.</span></span>
<span id="cb7-309"><a href="#cb7-309" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-310"><a href="#cb7-310" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750436</span></span>
<span id="cb7-311"><a href="#cb7-311" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-312"><a href="#cb7-312" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-313"><a href="#cb7-313" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================</span>                                                     <span class="er">|</span>  <span class="dv">24</span>%</span>
<span id="cb7-314"><a href="#cb7-314" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-315"><a href="#cb7-315" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750573: No data found.</span></span>
<span id="cb7-316"><a href="#cb7-316" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-317"><a href="#cb7-317" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750573: No data found.</span></span>
<span id="cb7-318"><a href="#cb7-318" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-319"><a href="#cb7-319" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750573</span></span>
<span id="cb7-320"><a href="#cb7-320" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-321"><a href="#cb7-321" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750391: No data found.</span></span>
<span id="cb7-322"><a href="#cb7-322" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-323"><a href="#cb7-323" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750391: No data found.</span></span>
<span id="cb7-324"><a href="#cb7-324" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-325"><a href="#cb7-325" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750391</span></span>
<span id="cb7-326"><a href="#cb7-326" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-327"><a href="#cb7-327" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-328"><a href="#cb7-328" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================</span>                                                     <span class="er">|</span>  <span class="dv">25</span>%</span>
<span id="cb7-329"><a href="#cb7-329" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-330"><a href="#cb7-330" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750526: No data found.</span></span>
<span id="cb7-331"><a href="#cb7-331" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-332"><a href="#cb7-332" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750526: No data found.</span></span>
<span id="cb7-333"><a href="#cb7-333" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-334"><a href="#cb7-334" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750526</span></span>
<span id="cb7-335"><a href="#cb7-335" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-336"><a href="#cb7-336" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-337"><a href="#cb7-337" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================</span>                                                    <span class="er">|</span>  <span class="dv">25</span>%</span>
<span id="cb7-338"><a href="#cb7-338" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-339"><a href="#cb7-339" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================</span>                                                    <span class="er">|</span>  <span class="dv">26</span>%</span>
<span id="cb7-340"><a href="#cb7-340" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-341"><a href="#cb7-341" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750420: No data found.</span></span>
<span id="cb7-342"><a href="#cb7-342" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-343"><a href="#cb7-343" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750420: No data found.</span></span>
<span id="cb7-344"><a href="#cb7-344" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-345"><a href="#cb7-345" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750420</span></span>
<span id="cb7-346"><a href="#cb7-346" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-347"><a href="#cb7-347" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-348"><a href="#cb7-348" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================</span>                                                   <span class="er">|</span>  <span class="dv">26</span>%</span>
<span id="cb7-349"><a href="#cb7-349" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-350"><a href="#cb7-350" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750519: No data found.</span></span>
<span id="cb7-351"><a href="#cb7-351" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-352"><a href="#cb7-352" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750519: No data found.</span></span>
<span id="cb7-353"><a href="#cb7-353" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-354"><a href="#cb7-354" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750519</span></span>
<span id="cb7-355"><a href="#cb7-355" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-356"><a href="#cb7-356" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-357"><a href="#cb7-357" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================</span>                                                   <span class="er">|</span>  <span class="dv">27</span>%</span>
<span id="cb7-358"><a href="#cb7-358" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-359"><a href="#cb7-359" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750455: No data found.</span></span>
<span id="cb7-360"><a href="#cb7-360" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-361"><a href="#cb7-361" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750455: No data found.</span></span>
<span id="cb7-362"><a href="#cb7-362" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-363"><a href="#cb7-363" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750455</span></span>
<span id="cb7-364"><a href="#cb7-364" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-365"><a href="#cb7-365" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-366"><a href="#cb7-366" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================</span>                                                   <span class="er">|</span>  <span class="dv">28</span>%</span>
<span id="cb7-367"><a href="#cb7-367" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-368"><a href="#cb7-368" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================</span>                                                  <span class="er">|</span>  <span class="dv">28</span>%</span>
<span id="cb7-369"><a href="#cb7-369" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-370"><a href="#cb7-370" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================</span>                                                  <span class="er">|</span>  <span class="dv">29</span>%</span>
<span id="cb7-371"><a href="#cb7-371" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-372"><a href="#cb7-372" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750538: No data found.</span></span>
<span id="cb7-373"><a href="#cb7-373" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-374"><a href="#cb7-374" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750538: No data found.</span></span>
<span id="cb7-375"><a href="#cb7-375" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-376"><a href="#cb7-376" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750538</span></span>
<span id="cb7-377"><a href="#cb7-377" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-378"><a href="#cb7-378" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750542: No data found.</span></span>
<span id="cb7-379"><a href="#cb7-379" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-380"><a href="#cb7-380" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750542: No data found.</span></span>
<span id="cb7-381"><a href="#cb7-381" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-382"><a href="#cb7-382" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750542</span></span>
<span id="cb7-383"><a href="#cb7-383" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-384"><a href="#cb7-384" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-385"><a href="#cb7-385" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================</span>                                                 <span class="er">|</span>  <span class="dv">30</span>%</span>
<span id="cb7-386"><a href="#cb7-386" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-387"><a href="#cb7-387" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750520: No data found.</span></span>
<span id="cb7-388"><a href="#cb7-388" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-389"><a href="#cb7-389" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750520: No data found.</span></span>
<span id="cb7-390"><a href="#cb7-390" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-391"><a href="#cb7-391" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750520</span></span>
<span id="cb7-392"><a href="#cb7-392" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-393"><a href="#cb7-393" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750598: No data found.</span></span>
<span id="cb7-394"><a href="#cb7-394" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-395"><a href="#cb7-395" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750598: No data found.</span></span>
<span id="cb7-396"><a href="#cb7-396" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-397"><a href="#cb7-397" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750598</span></span>
<span id="cb7-398"><a href="#cb7-398" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-399"><a href="#cb7-399" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-400"><a href="#cb7-400" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================</span>                                                 <span class="er">|</span>  <span class="dv">31</span>%</span>
<span id="cb7-401"><a href="#cb7-401" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-402"><a href="#cb7-402" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750521: No data found.</span></span>
<span id="cb7-403"><a href="#cb7-403" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-404"><a href="#cb7-404" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750521: No data found.</span></span>
<span id="cb7-405"><a href="#cb7-405" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-406"><a href="#cb7-406" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750521</span></span>
<span id="cb7-407"><a href="#cb7-407" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-408"><a href="#cb7-408" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-409"><a href="#cb7-409" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================</span>                                                <span class="er">|</span>  <span class="dv">31</span>%</span>
<span id="cb7-410"><a href="#cb7-410" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-411"><a href="#cb7-411" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750528: No data found.</span></span>
<span id="cb7-412"><a href="#cb7-412" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-413"><a href="#cb7-413" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750528: No data found.</span></span>
<span id="cb7-414"><a href="#cb7-414" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-415"><a href="#cb7-415" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750528</span></span>
<span id="cb7-416"><a href="#cb7-416" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-417"><a href="#cb7-417" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-418"><a href="#cb7-418" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================</span>                                                <span class="er">|</span>  <span class="dv">32</span>%</span>
<span id="cb7-419"><a href="#cb7-419" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-420"><a href="#cb7-420" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750411: No data found.</span></span>
<span id="cb7-421"><a href="#cb7-421" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-422"><a href="#cb7-422" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750411: No data found.</span></span>
<span id="cb7-423"><a href="#cb7-423" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-424"><a href="#cb7-424" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750411</span></span>
<span id="cb7-425"><a href="#cb7-425" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-426"><a href="#cb7-426" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-427"><a href="#cb7-427" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">32</span>%</span>
<span id="cb7-428"><a href="#cb7-428" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-429"><a href="#cb7-429" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">33</span>%</span>
<span id="cb7-430"><a href="#cb7-430" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-431"><a href="#cb7-431" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750400: No data found.</span></span>
<span id="cb7-432"><a href="#cb7-432" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-433"><a href="#cb7-433" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750400: No data found.</span></span>
<span id="cb7-434"><a href="#cb7-434" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-435"><a href="#cb7-435" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750400</span></span>
<span id="cb7-436"><a href="#cb7-436" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-437"><a href="#cb7-437" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750572: No data found.</span></span>
<span id="cb7-438"><a href="#cb7-438" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-439"><a href="#cb7-439" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750572: No data found.</span></span>
<span id="cb7-440"><a href="#cb7-440" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-441"><a href="#cb7-441" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750572</span></span>
<span id="cb7-442"><a href="#cb7-442" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-443"><a href="#cb7-443" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-444"><a href="#cb7-444" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================</span>                                              <span class="er">|</span>  <span class="dv">34</span>%</span>
<span id="cb7-445"><a href="#cb7-445" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-446"><a href="#cb7-446" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750415: No data found.</span></span>
<span id="cb7-447"><a href="#cb7-447" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-448"><a href="#cb7-448" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750415: No data found.</span></span>
<span id="cb7-449"><a href="#cb7-449" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-450"><a href="#cb7-450" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750415</span></span>
<span id="cb7-451"><a href="#cb7-451" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-452"><a href="#cb7-452" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750503: No data found.</span></span>
<span id="cb7-453"><a href="#cb7-453" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-454"><a href="#cb7-454" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750503: No data found.</span></span>
<span id="cb7-455"><a href="#cb7-455" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-456"><a href="#cb7-456" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750503</span></span>
<span id="cb7-457"><a href="#cb7-457" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-458"><a href="#cb7-458" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-459"><a href="#cb7-459" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================</span>                                              <span class="er">|</span>  <span class="dv">35</span>%</span>
<span id="cb7-460"><a href="#cb7-460" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-461"><a href="#cb7-461" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750471: No data found.</span></span>
<span id="cb7-462"><a href="#cb7-462" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-463"><a href="#cb7-463" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750471: No data found.</span></span>
<span id="cb7-464"><a href="#cb7-464" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-465"><a href="#cb7-465" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750471</span></span>
<span id="cb7-466"><a href="#cb7-466" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-467"><a href="#cb7-467" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-468"><a href="#cb7-468" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================</span>                                             <span class="er">|</span>  <span class="dv">35</span>%</span>
<span id="cb7-469"><a href="#cb7-469" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-470"><a href="#cb7-470" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750448: No data found.</span></span>
<span id="cb7-471"><a href="#cb7-471" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-472"><a href="#cb7-472" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750448: No data found.</span></span>
<span id="cb7-473"><a href="#cb7-473" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-474"><a href="#cb7-474" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750448</span></span>
<span id="cb7-475"><a href="#cb7-475" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-476"><a href="#cb7-476" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-477"><a href="#cb7-477" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================</span>                                             <span class="er">|</span>  <span class="dv">36</span>%</span>
<span id="cb7-478"><a href="#cb7-478" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-479"><a href="#cb7-479" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750409: No data found.</span></span>
<span id="cb7-480"><a href="#cb7-480" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-481"><a href="#cb7-481" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750409: No data found.</span></span>
<span id="cb7-482"><a href="#cb7-482" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-483"><a href="#cb7-483" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750409</span></span>
<span id="cb7-484"><a href="#cb7-484" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-485"><a href="#cb7-485" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-486"><a href="#cb7-486" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================</span>                                            <span class="er">|</span>  <span class="dv">37</span>%</span>
<span id="cb7-487"><a href="#cb7-487" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-488"><a href="#cb7-488" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750557: No data found.</span></span>
<span id="cb7-489"><a href="#cb7-489" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-490"><a href="#cb7-490" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750557: No data found.</span></span>
<span id="cb7-491"><a href="#cb7-491" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-492"><a href="#cb7-492" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750557</span></span>
<span id="cb7-493"><a href="#cb7-493" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-494"><a href="#cb7-494" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750456: No data found.</span></span>
<span id="cb7-495"><a href="#cb7-495" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-496"><a href="#cb7-496" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750456: No data found.</span></span>
<span id="cb7-497"><a href="#cb7-497" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-498"><a href="#cb7-498" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750456</span></span>
<span id="cb7-499"><a href="#cb7-499" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-500"><a href="#cb7-500" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-501"><a href="#cb7-501" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================</span>                                            <span class="er">|</span>  <span class="dv">38</span>%</span>
<span id="cb7-502"><a href="#cb7-502" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-503"><a href="#cb7-503" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750539: No data found.</span></span>
<span id="cb7-504"><a href="#cb7-504" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-505"><a href="#cb7-505" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750539: No data found.</span></span>
<span id="cb7-506"><a href="#cb7-506" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-507"><a href="#cb7-507" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750539</span></span>
<span id="cb7-508"><a href="#cb7-508" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-509"><a href="#cb7-509" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-510"><a href="#cb7-510" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================</span>                                           <span class="er">|</span>  <span class="dv">38</span>%</span>
<span id="cb7-511"><a href="#cb7-511" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-512"><a href="#cb7-512" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================</span>                                           <span class="er">|</span>  <span class="dv">39</span>%</span>
<span id="cb7-513"><a href="#cb7-513" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-514"><a href="#cb7-514" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750464: No data found.</span></span>
<span id="cb7-515"><a href="#cb7-515" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-516"><a href="#cb7-516" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750464: No data found.</span></span>
<span id="cb7-517"><a href="#cb7-517" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-518"><a href="#cb7-518" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750464</span></span>
<span id="cb7-519"><a href="#cb7-519" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-520"><a href="#cb7-520" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-521"><a href="#cb7-521" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================</span>                                          <span class="er">|</span>  <span class="dv">40</span>%</span>
<span id="cb7-522"><a href="#cb7-522" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-523"><a href="#cb7-523" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750469: No data found.</span></span>
<span id="cb7-524"><a href="#cb7-524" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-525"><a href="#cb7-525" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750469: No data found.</span></span>
<span id="cb7-526"><a href="#cb7-526" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-527"><a href="#cb7-527" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750469</span></span>
<span id="cb7-528"><a href="#cb7-528" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-529"><a href="#cb7-529" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-530"><a href="#cb7-530" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================</span>                                         <span class="er">|</span>  <span class="dv">41</span>%</span>
<span id="cb7-531"><a href="#cb7-531" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-532"><a href="#cb7-532" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750543: No data found.</span></span>
<span id="cb7-533"><a href="#cb7-533" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-534"><a href="#cb7-534" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750543: No data found.</span></span>
<span id="cb7-535"><a href="#cb7-535" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-536"><a href="#cb7-536" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750543</span></span>
<span id="cb7-537"><a href="#cb7-537" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-538"><a href="#cb7-538" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-539"><a href="#cb7-539" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================</span>                                         <span class="er">|</span>  <span class="dv">42</span>%</span>
<span id="cb7-540"><a href="#cb7-540" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-541"><a href="#cb7-541" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750489: No data found.</span></span>
<span id="cb7-542"><a href="#cb7-542" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-543"><a href="#cb7-543" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750489: No data found.</span></span>
<span id="cb7-544"><a href="#cb7-544" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-545"><a href="#cb7-545" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750489</span></span>
<span id="cb7-546"><a href="#cb7-546" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-547"><a href="#cb7-547" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-548"><a href="#cb7-548" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================</span>                                        <span class="er">|</span>  <span class="dv">42</span>%</span>
<span id="cb7-549"><a href="#cb7-549" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-550"><a href="#cb7-550" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750581: No data found.</span></span>
<span id="cb7-551"><a href="#cb7-551" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-552"><a href="#cb7-552" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750581: No data found.</span></span>
<span id="cb7-553"><a href="#cb7-553" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-554"><a href="#cb7-554" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750581</span></span>
<span id="cb7-555"><a href="#cb7-555" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-556"><a href="#cb7-556" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-557"><a href="#cb7-557" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================</span>                                        <span class="er">|</span>  <span class="dv">43</span>%</span>
<span id="cb7-558"><a href="#cb7-558" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-559"><a href="#cb7-559" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750434: No data found.</span></span>
<span id="cb7-560"><a href="#cb7-560" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-561"><a href="#cb7-561" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750434: No data found.</span></span>
<span id="cb7-562"><a href="#cb7-562" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-563"><a href="#cb7-563" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750434</span></span>
<span id="cb7-564"><a href="#cb7-564" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-565"><a href="#cb7-565" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750399: No data found.</span></span>
<span id="cb7-566"><a href="#cb7-566" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-567"><a href="#cb7-567" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750399: No data found.</span></span>
<span id="cb7-568"><a href="#cb7-568" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-569"><a href="#cb7-569" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750399</span></span>
<span id="cb7-570"><a href="#cb7-570" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-571"><a href="#cb7-571" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-572"><a href="#cb7-572" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================</span>                                       <span class="er">|</span>  <span class="dv">44</span>%</span>
<span id="cb7-573"><a href="#cb7-573" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-574"><a href="#cb7-574" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750558: No data found.</span></span>
<span id="cb7-575"><a href="#cb7-575" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-576"><a href="#cb7-576" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750558: No data found.</span></span>
<span id="cb7-577"><a href="#cb7-577" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-578"><a href="#cb7-578" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750558</span></span>
<span id="cb7-579"><a href="#cb7-579" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-580"><a href="#cb7-580" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-581"><a href="#cb7-581" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================</span>                                       <span class="er">|</span>  <span class="dv">45</span>%</span>
<span id="cb7-582"><a href="#cb7-582" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-583"><a href="#cb7-583" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750449: No data found.</span></span>
<span id="cb7-584"><a href="#cb7-584" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-585"><a href="#cb7-585" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750449: No data found.</span></span>
<span id="cb7-586"><a href="#cb7-586" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-587"><a href="#cb7-587" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750449</span></span>
<span id="cb7-588"><a href="#cb7-588" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-589"><a href="#cb7-589" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-590"><a href="#cb7-590" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================</span>                                      <span class="er">|</span>  <span class="dv">46</span>%</span>
<span id="cb7-591"><a href="#cb7-591" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-592"><a href="#cb7-592" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750396: No data found.</span></span>
<span id="cb7-593"><a href="#cb7-593" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-594"><a href="#cb7-594" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750396: No data found.</span></span>
<span id="cb7-595"><a href="#cb7-595" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-596"><a href="#cb7-596" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750396</span></span>
<span id="cb7-597"><a href="#cb7-597" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-598"><a href="#cb7-598" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750518: No data found.</span></span>
<span id="cb7-599"><a href="#cb7-599" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-600"><a href="#cb7-600" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750518: No data found.</span></span>
<span id="cb7-601"><a href="#cb7-601" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-602"><a href="#cb7-602" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750518</span></span>
<span id="cb7-603"><a href="#cb7-603" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-604"><a href="#cb7-604" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-605"><a href="#cb7-605" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================</span>                                     <span class="er">|</span>  <span class="dv">47</span>%</span>
<span id="cb7-606"><a href="#cb7-606" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-607"><a href="#cb7-607" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750563: No data found.</span></span>
<span id="cb7-608"><a href="#cb7-608" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-609"><a href="#cb7-609" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750563: No data found.</span></span>
<span id="cb7-610"><a href="#cb7-610" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-611"><a href="#cb7-611" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750563</span></span>
<span id="cb7-612"><a href="#cb7-612" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-613"><a href="#cb7-613" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750482: No data found.</span></span>
<span id="cb7-614"><a href="#cb7-614" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-615"><a href="#cb7-615" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750482: No data found.</span></span>
<span id="cb7-616"><a href="#cb7-616" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-617"><a href="#cb7-617" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750482</span></span>
<span id="cb7-618"><a href="#cb7-618" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-619"><a href="#cb7-619" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-620"><a href="#cb7-620" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================</span>                                     <span class="er">|</span>  <span class="dv">48</span>%</span>
<span id="cb7-621"><a href="#cb7-621" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-622"><a href="#cb7-622" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750487: No data found.</span></span>
<span id="cb7-623"><a href="#cb7-623" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-624"><a href="#cb7-624" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750487: No data found.</span></span>
<span id="cb7-625"><a href="#cb7-625" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-626"><a href="#cb7-626" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750487</span></span>
<span id="cb7-627"><a href="#cb7-627" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-628"><a href="#cb7-628" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-629"><a href="#cb7-629" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================</span>                                    <span class="er">|</span>  <span class="dv">48</span>%</span>
<span id="cb7-630"><a href="#cb7-630" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-631"><a href="#cb7-631" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================</span>                                    <span class="er">|</span>  <span class="dv">49</span>%</span>
<span id="cb7-632"><a href="#cb7-632" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-633"><a href="#cb7-633" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750590: No data found.</span></span>
<span id="cb7-634"><a href="#cb7-634" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-635"><a href="#cb7-635" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750590: No data found.</span></span>
<span id="cb7-636"><a href="#cb7-636" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-637"><a href="#cb7-637" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750590</span></span>
<span id="cb7-638"><a href="#cb7-638" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-639"><a href="#cb7-639" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-640"><a href="#cb7-640" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================</span>                                   <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb7-641"><a href="#cb7-641" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-642"><a href="#cb7-642" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750435: No data found.</span></span>
<span id="cb7-643"><a href="#cb7-643" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-644"><a href="#cb7-644" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750435: No data found.</span></span>
<span id="cb7-645"><a href="#cb7-645" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-646"><a href="#cb7-646" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750435</span></span>
<span id="cb7-647"><a href="#cb7-647" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-648"><a href="#cb7-648" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-649"><a href="#cb7-649" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================</span>                                  <span class="er">|</span>  <span class="dv">51</span>%</span>
<span id="cb7-650"><a href="#cb7-650" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-651"><a href="#cb7-651" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750468: No data found.</span></span>
<span id="cb7-652"><a href="#cb7-652" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-653"><a href="#cb7-653" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750468: No data found.</span></span>
<span id="cb7-654"><a href="#cb7-654" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-655"><a href="#cb7-655" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750468</span></span>
<span id="cb7-656"><a href="#cb7-656" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-657"><a href="#cb7-657" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-658"><a href="#cb7-658" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================</span>                                  <span class="er">|</span>  <span class="dv">52</span>%</span>
<span id="cb7-659"><a href="#cb7-659" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-660"><a href="#cb7-660" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750559: No data found.</span></span>
<span id="cb7-661"><a href="#cb7-661" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-662"><a href="#cb7-662" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750559: No data found.</span></span>
<span id="cb7-663"><a href="#cb7-663" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-664"><a href="#cb7-664" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750559</span></span>
<span id="cb7-665"><a href="#cb7-665" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-666"><a href="#cb7-666" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-667"><a href="#cb7-667" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================</span>                                 <span class="er">|</span>  <span class="dv">52</span>%</span>
<span id="cb7-668"><a href="#cb7-668" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-669"><a href="#cb7-669" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================</span>                                 <span class="er">|</span>  <span class="dv">53</span>%</span>
<span id="cb7-670"><a href="#cb7-670" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-671"><a href="#cb7-671" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750402: No data found.</span></span>
<span id="cb7-672"><a href="#cb7-672" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-673"><a href="#cb7-673" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750402: No data found.</span></span>
<span id="cb7-674"><a href="#cb7-674" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-675"><a href="#cb7-675" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750402</span></span>
<span id="cb7-676"><a href="#cb7-676" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-677"><a href="#cb7-677" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750524: No data found.</span></span>
<span id="cb7-678"><a href="#cb7-678" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-679"><a href="#cb7-679" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750524: No data found.</span></span>
<span id="cb7-680"><a href="#cb7-680" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-681"><a href="#cb7-681" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750524</span></span>
<span id="cb7-682"><a href="#cb7-682" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-683"><a href="#cb7-683" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-684"><a href="#cb7-684" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================</span>                                <span class="er">|</span>  <span class="dv">54</span>%</span>
<span id="cb7-685"><a href="#cb7-685" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-686"><a href="#cb7-686" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750555: No data found.</span></span>
<span id="cb7-687"><a href="#cb7-687" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-688"><a href="#cb7-688" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750555: No data found.</span></span>
<span id="cb7-689"><a href="#cb7-689" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-690"><a href="#cb7-690" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750555</span></span>
<span id="cb7-691"><a href="#cb7-691" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-692"><a href="#cb7-692" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750593: No data found.</span></span>
<span id="cb7-693"><a href="#cb7-693" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-694"><a href="#cb7-694" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750593: No data found.</span></span>
<span id="cb7-695"><a href="#cb7-695" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-696"><a href="#cb7-696" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750593</span></span>
<span id="cb7-697"><a href="#cb7-697" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-698"><a href="#cb7-698" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-699"><a href="#cb7-699" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================</span>                               <span class="er">|</span>  <span class="dv">55</span>%</span>
<span id="cb7-700"><a href="#cb7-700" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-701"><a href="#cb7-701" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750454: No data found.</span></span>
<span id="cb7-702"><a href="#cb7-702" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-703"><a href="#cb7-703" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750454: No data found.</span></span>
<span id="cb7-704"><a href="#cb7-704" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-705"><a href="#cb7-705" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750454</span></span>
<span id="cb7-706"><a href="#cb7-706" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-707"><a href="#cb7-707" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-708"><a href="#cb7-708" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================</span>                               <span class="er">|</span>  <span class="dv">56</span>%</span>
<span id="cb7-709"><a href="#cb7-709" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-710"><a href="#cb7-710" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750467: No data found.</span></span>
<span id="cb7-711"><a href="#cb7-711" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-712"><a href="#cb7-712" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750467: No data found.</span></span>
<span id="cb7-713"><a href="#cb7-713" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-714"><a href="#cb7-714" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750467</span></span>
<span id="cb7-715"><a href="#cb7-715" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-716"><a href="#cb7-716" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750578: No data found.</span></span>
<span id="cb7-717"><a href="#cb7-717" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-718"><a href="#cb7-718" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750578: No data found.</span></span>
<span id="cb7-719"><a href="#cb7-719" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-720"><a href="#cb7-720" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750578</span></span>
<span id="cb7-721"><a href="#cb7-721" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-722"><a href="#cb7-722" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-723"><a href="#cb7-723" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================</span>                              <span class="er">|</span>  <span class="dv">57</span>%</span>
<span id="cb7-724"><a href="#cb7-724" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-725"><a href="#cb7-725" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750554: No data found.</span></span>
<span id="cb7-726"><a href="#cb7-726" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-727"><a href="#cb7-727" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750554: No data found.</span></span>
<span id="cb7-728"><a href="#cb7-728" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-729"><a href="#cb7-729" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750554</span></span>
<span id="cb7-730"><a href="#cb7-730" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-731"><a href="#cb7-731" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750485: No data found.</span></span>
<span id="cb7-732"><a href="#cb7-732" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-733"><a href="#cb7-733" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750485: No data found.</span></span>
<span id="cb7-734"><a href="#cb7-734" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-735"><a href="#cb7-735" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750485</span></span>
<span id="cb7-736"><a href="#cb7-736" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-737"><a href="#cb7-737" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-738"><a href="#cb7-738" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================</span>                              <span class="er">|</span>  <span class="dv">58</span>%</span>
<span id="cb7-739"><a href="#cb7-739" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-740"><a href="#cb7-740" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================================</span>                             <span class="er">|</span>  <span class="dv">58</span>%</span>
<span id="cb7-741"><a href="#cb7-741" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-742"><a href="#cb7-742" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================================</span>                             <span class="er">|</span>  <span class="dv">59</span>%</span>
<span id="cb7-743"><a href="#cb7-743" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-744"><a href="#cb7-744" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750505: No data found.</span></span>
<span id="cb7-745"><a href="#cb7-745" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-746"><a href="#cb7-746" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750505: No data found.</span></span>
<span id="cb7-747"><a href="#cb7-747" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-748"><a href="#cb7-748" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750505</span></span>
<span id="cb7-749"><a href="#cb7-749" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-750"><a href="#cb7-750" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-751"><a href="#cb7-751" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================</span>                            <span class="er">|</span>  <span class="dv">60</span>%</span>
<span id="cb7-752"><a href="#cb7-752" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-753"><a href="#cb7-753" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750507: No data found.</span></span>
<span id="cb7-754"><a href="#cb7-754" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-755"><a href="#cb7-755" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750507: No data found.</span></span>
<span id="cb7-756"><a href="#cb7-756" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-757"><a href="#cb7-757" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750507</span></span>
<span id="cb7-758"><a href="#cb7-758" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-759"><a href="#cb7-759" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750428: No data found.</span></span>
<span id="cb7-760"><a href="#cb7-760" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-761"><a href="#cb7-761" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750428: No data found.</span></span>
<span id="cb7-762"><a href="#cb7-762" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-763"><a href="#cb7-763" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750428</span></span>
<span id="cb7-764"><a href="#cb7-764" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-765"><a href="#cb7-765" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-766"><a href="#cb7-766" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================</span>                           <span class="er">|</span>  <span class="dv">61</span>%</span>
<span id="cb7-767"><a href="#cb7-767" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-768"><a href="#cb7-768" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750535: No data found.</span></span>
<span id="cb7-769"><a href="#cb7-769" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-770"><a href="#cb7-770" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750535: No data found.</span></span>
<span id="cb7-771"><a href="#cb7-771" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-772"><a href="#cb7-772" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750535</span></span>
<span id="cb7-773"><a href="#cb7-773" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-774"><a href="#cb7-774" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-775"><a href="#cb7-775" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================</span>                           <span class="er">|</span>  <span class="dv">62</span>%</span>
<span id="cb7-776"><a href="#cb7-776" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-777"><a href="#cb7-777" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750591: No data found.</span></span>
<span id="cb7-778"><a href="#cb7-778" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-779"><a href="#cb7-779" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750591: No data found.</span></span>
<span id="cb7-780"><a href="#cb7-780" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-781"><a href="#cb7-781" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750591</span></span>
<span id="cb7-782"><a href="#cb7-782" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-783"><a href="#cb7-783" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-784"><a href="#cb7-784" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================</span>                          <span class="er">|</span>  <span class="dv">62</span>%</span>
<span id="cb7-785"><a href="#cb7-785" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-786"><a href="#cb7-786" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750483: No data found.</span></span>
<span id="cb7-787"><a href="#cb7-787" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-788"><a href="#cb7-788" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750483: No data found.</span></span>
<span id="cb7-789"><a href="#cb7-789" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-790"><a href="#cb7-790" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750483</span></span>
<span id="cb7-791"><a href="#cb7-791" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-792"><a href="#cb7-792" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-793"><a href="#cb7-793" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================</span>                          <span class="er">|</span>  <span class="dv">63</span>%</span>
<span id="cb7-794"><a href="#cb7-794" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-795"><a href="#cb7-795" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750470: No data found.</span></span>
<span id="cb7-796"><a href="#cb7-796" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-797"><a href="#cb7-797" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750470: No data found.</span></span>
<span id="cb7-798"><a href="#cb7-798" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-799"><a href="#cb7-799" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750470</span></span>
<span id="cb7-800"><a href="#cb7-800" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-801"><a href="#cb7-801" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750419: No data found.</span></span>
<span id="cb7-802"><a href="#cb7-802" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-803"><a href="#cb7-803" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750419: No data found.</span></span>
<span id="cb7-804"><a href="#cb7-804" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-805"><a href="#cb7-805" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750419</span></span>
<span id="cb7-806"><a href="#cb7-806" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-807"><a href="#cb7-807" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-808"><a href="#cb7-808" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================</span>                         <span class="er">|</span>  <span class="dv">64</span>%</span>
<span id="cb7-809"><a href="#cb7-809" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-810"><a href="#cb7-810" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================</span>                         <span class="er">|</span>  <span class="dv">65</span>%</span>
<span id="cb7-811"><a href="#cb7-811" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-812"><a href="#cb7-812" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================</span>                        <span class="er">|</span>  <span class="dv">65</span>%</span>
<span id="cb7-813"><a href="#cb7-813" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-814"><a href="#cb7-814" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750595: No data found.</span></span>
<span id="cb7-815"><a href="#cb7-815" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-816"><a href="#cb7-816" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750595: No data found.</span></span>
<span id="cb7-817"><a href="#cb7-817" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-818"><a href="#cb7-818" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750595</span></span>
<span id="cb7-819"><a href="#cb7-819" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-820"><a href="#cb7-820" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-821"><a href="#cb7-821" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================</span>                        <span class="er">|</span>  <span class="dv">66</span>%</span>
<span id="cb7-822"><a href="#cb7-822" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-823"><a href="#cb7-823" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750544: No data found.</span></span>
<span id="cb7-824"><a href="#cb7-824" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-825"><a href="#cb7-825" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750544: No data found.</span></span>
<span id="cb7-826"><a href="#cb7-826" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-827"><a href="#cb7-827" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750544</span></span>
<span id="cb7-828"><a href="#cb7-828" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-829"><a href="#cb7-829" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750537: No data found.</span></span>
<span id="cb7-830"><a href="#cb7-830" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-831"><a href="#cb7-831" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750537: No data found.</span></span>
<span id="cb7-832"><a href="#cb7-832" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-833"><a href="#cb7-833" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750537</span></span>
<span id="cb7-834"><a href="#cb7-834" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-835"><a href="#cb7-835" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-836"><a href="#cb7-836" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">67</span>%</span>
<span id="cb7-837"><a href="#cb7-837" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-838"><a href="#cb7-838" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">68</span>%</span>
<span id="cb7-839"><a href="#cb7-839" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-840"><a href="#cb7-840" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750414: No data found.</span></span>
<span id="cb7-841"><a href="#cb7-841" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-842"><a href="#cb7-842" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750414: No data found.</span></span>
<span id="cb7-843"><a href="#cb7-843" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-844"><a href="#cb7-844" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750414</span></span>
<span id="cb7-845"><a href="#cb7-845" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-846"><a href="#cb7-846" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-847"><a href="#cb7-847" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================</span>                      <span class="er">|</span>  <span class="dv">68</span>%</span>
<span id="cb7-848"><a href="#cb7-848" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-849"><a href="#cb7-849" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750446: No data found.</span></span>
<span id="cb7-850"><a href="#cb7-850" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-851"><a href="#cb7-851" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750446: No data found.</span></span>
<span id="cb7-852"><a href="#cb7-852" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-853"><a href="#cb7-853" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750446</span></span>
<span id="cb7-854"><a href="#cb7-854" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-855"><a href="#cb7-855" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-856"><a href="#cb7-856" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================</span>                      <span class="er">|</span>  <span class="dv">69</span>%</span>
<span id="cb7-857"><a href="#cb7-857" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-858"><a href="#cb7-858" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750597: No data found.</span></span>
<span id="cb7-859"><a href="#cb7-859" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-860"><a href="#cb7-860" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750597: No data found.</span></span>
<span id="cb7-861"><a href="#cb7-861" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-862"><a href="#cb7-862" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750597</span></span>
<span id="cb7-863"><a href="#cb7-863" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-864"><a href="#cb7-864" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-865"><a href="#cb7-865" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================================</span>                     <span class="er">|</span>  <span class="dv">69</span>%</span>
<span id="cb7-866"><a href="#cb7-866" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-867"><a href="#cb7-867" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750488: No data found.</span></span>
<span id="cb7-868"><a href="#cb7-868" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-869"><a href="#cb7-869" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750488: No data found.</span></span>
<span id="cb7-870"><a href="#cb7-870" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-871"><a href="#cb7-871" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750488</span></span>
<span id="cb7-872"><a href="#cb7-872" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-873"><a href="#cb7-873" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-874"><a href="#cb7-874" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================================</span>                     <span class="er">|</span>  <span class="dv">70</span>%</span>
<span id="cb7-875"><a href="#cb7-875" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-876"><a href="#cb7-876" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750383: No data found.</span></span>
<span id="cb7-877"><a href="#cb7-877" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-878"><a href="#cb7-878" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750383: No data found.</span></span>
<span id="cb7-879"><a href="#cb7-879" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-880"><a href="#cb7-880" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750383</span></span>
<span id="cb7-881"><a href="#cb7-881" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-882"><a href="#cb7-882" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750447: No data found.</span></span>
<span id="cb7-883"><a href="#cb7-883" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-884"><a href="#cb7-884" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750447: No data found.</span></span>
<span id="cb7-885"><a href="#cb7-885" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-886"><a href="#cb7-886" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750447</span></span>
<span id="cb7-887"><a href="#cb7-887" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-888"><a href="#cb7-888" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-889"><a href="#cb7-889" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================</span>                    <span class="er">|</span>  <span class="dv">71</span>%</span>
<span id="cb7-890"><a href="#cb7-890" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-891"><a href="#cb7-891" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750392: No data found.</span></span>
<span id="cb7-892"><a href="#cb7-892" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-893"><a href="#cb7-893" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750392: No data found.</span></span>
<span id="cb7-894"><a href="#cb7-894" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-895"><a href="#cb7-895" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750392</span></span>
<span id="cb7-896"><a href="#cb7-896" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-897"><a href="#cb7-897" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750561: No data found.</span></span>
<span id="cb7-898"><a href="#cb7-898" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-899"><a href="#cb7-899" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750561: No data found.</span></span>
<span id="cb7-900"><a href="#cb7-900" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-901"><a href="#cb7-901" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750561</span></span>
<span id="cb7-902"><a href="#cb7-902" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-903"><a href="#cb7-903" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-904"><a href="#cb7-904" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================</span>                    <span class="er">|</span>  <span class="dv">72</span>%</span>
<span id="cb7-905"><a href="#cb7-905" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-906"><a href="#cb7-906" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750499: No data found.</span></span>
<span id="cb7-907"><a href="#cb7-907" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-908"><a href="#cb7-908" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750499: No data found.</span></span>
<span id="cb7-909"><a href="#cb7-909" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-910"><a href="#cb7-910" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750499</span></span>
<span id="cb7-911"><a href="#cb7-911" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-912"><a href="#cb7-912" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-913"><a href="#cb7-913" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================</span>                   <span class="er">|</span>  <span class="dv">72</span>%</span>
<span id="cb7-914"><a href="#cb7-914" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-915"><a href="#cb7-915" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750427: No data found.</span></span>
<span id="cb7-916"><a href="#cb7-916" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-917"><a href="#cb7-917" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750427: No data found.</span></span>
<span id="cb7-918"><a href="#cb7-918" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-919"><a href="#cb7-919" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750427</span></span>
<span id="cb7-920"><a href="#cb7-920" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-921"><a href="#cb7-921" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-922"><a href="#cb7-922" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================</span>                   <span class="er">|</span>  <span class="dv">73</span>%</span>
<span id="cb7-923"><a href="#cb7-923" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-924"><a href="#cb7-924" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750545: No data found.</span></span>
<span id="cb7-925"><a href="#cb7-925" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-926"><a href="#cb7-926" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750545: No data found.</span></span>
<span id="cb7-927"><a href="#cb7-927" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-928"><a href="#cb7-928" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750545</span></span>
<span id="cb7-929"><a href="#cb7-929" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-930"><a href="#cb7-930" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-931"><a href="#cb7-931" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================</span>                   <span class="er">|</span>  <span class="dv">74</span>%</span>
<span id="cb7-932"><a href="#cb7-932" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-933"><a href="#cb7-933" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================</span>                  <span class="er">|</span>  <span class="dv">74</span>%</span>
<span id="cb7-934"><a href="#cb7-934" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-935"><a href="#cb7-935" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================</span>                  <span class="er">|</span>  <span class="dv">75</span>%</span>
<span id="cb7-936"><a href="#cb7-936" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-937"><a href="#cb7-937" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================================</span>                 <span class="er">|</span>  <span class="dv">75</span>%</span>
<span id="cb7-938"><a href="#cb7-938" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-939"><a href="#cb7-939" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750433: No data found.</span></span>
<span id="cb7-940"><a href="#cb7-940" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-941"><a href="#cb7-941" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750433: No data found.</span></span>
<span id="cb7-942"><a href="#cb7-942" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-943"><a href="#cb7-943" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750433</span></span>
<span id="cb7-944"><a href="#cb7-944" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-945"><a href="#cb7-945" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-946"><a href="#cb7-946" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================================</span>                 <span class="er">|</span>  <span class="dv">76</span>%</span>
<span id="cb7-947"><a href="#cb7-947" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-948"><a href="#cb7-948" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750486: No data found.</span></span>
<span id="cb7-949"><a href="#cb7-949" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-950"><a href="#cb7-950" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750486: No data found.</span></span>
<span id="cb7-951"><a href="#cb7-951" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-952"><a href="#cb7-952" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750486</span></span>
<span id="cb7-953"><a href="#cb7-953" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-954"><a href="#cb7-954" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750431: No data found.</span></span>
<span id="cb7-955"><a href="#cb7-955" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-956"><a href="#cb7-956" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750431: No data found.</span></span>
<span id="cb7-957"><a href="#cb7-957" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-958"><a href="#cb7-958" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750431</span></span>
<span id="cb7-959"><a href="#cb7-959" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-960"><a href="#cb7-960" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-961"><a href="#cb7-961" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================</span>                <span class="er">|</span>  <span class="dv">77</span>%</span>
<span id="cb7-962"><a href="#cb7-962" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-963"><a href="#cb7-963" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750553: No data found.</span></span>
<span id="cb7-964"><a href="#cb7-964" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-965"><a href="#cb7-965" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750553: No data found.</span></span>
<span id="cb7-966"><a href="#cb7-966" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-967"><a href="#cb7-967" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750553</span></span>
<span id="cb7-968"><a href="#cb7-968" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-969"><a href="#cb7-969" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-970"><a href="#cb7-970" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================</span>                <span class="er">|</span>  <span class="dv">78</span>%</span>
<span id="cb7-971"><a href="#cb7-971" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-972"><a href="#cb7-972" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750562: No data found.</span></span>
<span id="cb7-973"><a href="#cb7-973" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-974"><a href="#cb7-974" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750562: No data found.</span></span>
<span id="cb7-975"><a href="#cb7-975" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-976"><a href="#cb7-976" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750562</span></span>
<span id="cb7-977"><a href="#cb7-977" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-978"><a href="#cb7-978" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-979"><a href="#cb7-979" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================================</span>               <span class="er">|</span>  <span class="dv">78</span>%</span>
<span id="cb7-980"><a href="#cb7-980" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-981"><a href="#cb7-981" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750504: No data found.</span></span>
<span id="cb7-982"><a href="#cb7-982" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-983"><a href="#cb7-983" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750504: No data found.</span></span>
<span id="cb7-984"><a href="#cb7-984" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-985"><a href="#cb7-985" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750504</span></span>
<span id="cb7-986"><a href="#cb7-986" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-987"><a href="#cb7-987" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-988"><a href="#cb7-988" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================================</span>               <span class="er">|</span>  <span class="dv">79</span>%</span>
<span id="cb7-989"><a href="#cb7-989" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-990"><a href="#cb7-990" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750506: No data found.</span></span>
<span id="cb7-991"><a href="#cb7-991" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-992"><a href="#cb7-992" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750506: No data found.</span></span>
<span id="cb7-993"><a href="#cb7-993" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-994"><a href="#cb7-994" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750506</span></span>
<span id="cb7-995"><a href="#cb7-995" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-996"><a href="#cb7-996" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-997"><a href="#cb7-997" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================================</span>              <span class="er">|</span>  <span class="dv">79</span>%</span>
<span id="cb7-998"><a href="#cb7-998" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-999"><a href="#cb7-999" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================================</span>              <span class="er">|</span>  <span class="dv">80</span>%</span>
<span id="cb7-1000"><a href="#cb7-1000" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1001"><a href="#cb7-1001" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750576: No data found.</span></span>
<span id="cb7-1002"><a href="#cb7-1002" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1003"><a href="#cb7-1003" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750576: No data found.</span></span>
<span id="cb7-1004"><a href="#cb7-1004" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1005"><a href="#cb7-1005" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750576</span></span>
<span id="cb7-1006"><a href="#cb7-1006" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1007"><a href="#cb7-1007" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750577: No data found.</span></span>
<span id="cb7-1008"><a href="#cb7-1008" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1009"><a href="#cb7-1009" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750577: No data found.</span></span>
<span id="cb7-1010"><a href="#cb7-1010" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1011"><a href="#cb7-1011" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750577</span></span>
<span id="cb7-1012"><a href="#cb7-1012" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1013"><a href="#cb7-1013" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1014"><a href="#cb7-1014" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================================================</span>             <span class="er">|</span>  <span class="dv">81</span>%</span>
<span id="cb7-1015"><a href="#cb7-1015" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1016"><a href="#cb7-1016" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750574: No data found.</span></span>
<span id="cb7-1017"><a href="#cb7-1017" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1018"><a href="#cb7-1018" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750574: No data found.</span></span>
<span id="cb7-1019"><a href="#cb7-1019" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1020"><a href="#cb7-1020" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750574</span></span>
<span id="cb7-1021"><a href="#cb7-1021" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1022"><a href="#cb7-1022" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1023"><a href="#cb7-1023" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================================================</span>             <span class="er">|</span>  <span class="dv">82</span>%</span>
<span id="cb7-1024"><a href="#cb7-1024" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1025"><a href="#cb7-1025" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================================</span>            <span class="er">|</span>  <span class="dv">83</span>%</span>
<span id="cb7-1026"><a href="#cb7-1026" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1027"><a href="#cb7-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750508: No data found.</span></span>
<span id="cb7-1028"><a href="#cb7-1028" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1029"><a href="#cb7-1029" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750508: No data found.</span></span>
<span id="cb7-1030"><a href="#cb7-1030" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1031"><a href="#cb7-1031" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750508</span></span>
<span id="cb7-1032"><a href="#cb7-1032" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1033"><a href="#cb7-1033" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750450: No data found.</span></span>
<span id="cb7-1034"><a href="#cb7-1034" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1035"><a href="#cb7-1035" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750450: No data found.</span></span>
<span id="cb7-1036"><a href="#cb7-1036" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1037"><a href="#cb7-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750450</span></span>
<span id="cb7-1038"><a href="#cb7-1038" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1039"><a href="#cb7-1039" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1040"><a href="#cb7-1040" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================================</span>           <span class="er">|</span>  <span class="dv">84</span>%</span>
<span id="cb7-1041"><a href="#cb7-1041" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1042"><a href="#cb7-1042" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750571: No data found.</span></span>
<span id="cb7-1043"><a href="#cb7-1043" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1044"><a href="#cb7-1044" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750571: No data found.</span></span>
<span id="cb7-1045"><a href="#cb7-1045" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1046"><a href="#cb7-1046" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750571</span></span>
<span id="cb7-1047"><a href="#cb7-1047" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1048"><a href="#cb7-1048" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750401: No data found.</span></span>
<span id="cb7-1049"><a href="#cb7-1049" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1050"><a href="#cb7-1050" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750401: No data found.</span></span>
<span id="cb7-1051"><a href="#cb7-1051" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1052"><a href="#cb7-1052" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750401</span></span>
<span id="cb7-1053"><a href="#cb7-1053" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1054"><a href="#cb7-1054" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1055"><a href="#cb7-1055" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================================</span>           <span class="er">|</span>  <span class="dv">85</span>%</span>
<span id="cb7-1056"><a href="#cb7-1056" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1057"><a href="#cb7-1057" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750451: No data found.</span></span>
<span id="cb7-1058"><a href="#cb7-1058" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1059"><a href="#cb7-1059" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750451: No data found.</span></span>
<span id="cb7-1060"><a href="#cb7-1060" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1061"><a href="#cb7-1061" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750451</span></span>
<span id="cb7-1062"><a href="#cb7-1062" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1063"><a href="#cb7-1063" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1064"><a href="#cb7-1064" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================================</span>          <span class="er">|</span>  <span class="dv">85</span>%</span>
<span id="cb7-1065"><a href="#cb7-1065" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1066"><a href="#cb7-1066" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750481: No data found.</span></span>
<span id="cb7-1067"><a href="#cb7-1067" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1068"><a href="#cb7-1068" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750481: No data found.</span></span>
<span id="cb7-1069"><a href="#cb7-1069" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1070"><a href="#cb7-1070" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750481</span></span>
<span id="cb7-1071"><a href="#cb7-1071" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1072"><a href="#cb7-1072" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1073"><a href="#cb7-1073" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================================</span>          <span class="er">|</span>  <span class="dv">86</span>%</span>
<span id="cb7-1074"><a href="#cb7-1074" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1075"><a href="#cb7-1075" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750466: No data found.</span></span>
<span id="cb7-1076"><a href="#cb7-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1077"><a href="#cb7-1077" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750466: No data found.</span></span>
<span id="cb7-1078"><a href="#cb7-1078" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1079"><a href="#cb7-1079" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750466</span></span>
<span id="cb7-1080"><a href="#cb7-1080" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1081"><a href="#cb7-1081" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750398: No data found.</span></span>
<span id="cb7-1082"><a href="#cb7-1082" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1083"><a href="#cb7-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750398: No data found.</span></span>
<span id="cb7-1084"><a href="#cb7-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1085"><a href="#cb7-1085" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750398</span></span>
<span id="cb7-1086"><a href="#cb7-1086" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1087"><a href="#cb7-1087" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1088"><a href="#cb7-1088" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================================</span>         <span class="er">|</span>  <span class="dv">87</span>%</span>
<span id="cb7-1089"><a href="#cb7-1089" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1090"><a href="#cb7-1090" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750417: No data found.</span></span>
<span id="cb7-1091"><a href="#cb7-1091" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1092"><a href="#cb7-1092" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750417: No data found.</span></span>
<span id="cb7-1093"><a href="#cb7-1093" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1094"><a href="#cb7-1094" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750417</span></span>
<span id="cb7-1095"><a href="#cb7-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1096"><a href="#cb7-1096" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750564: No data found.</span></span>
<span id="cb7-1097"><a href="#cb7-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1098"><a href="#cb7-1098" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750564: No data found.</span></span>
<span id="cb7-1099"><a href="#cb7-1099" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1100"><a href="#cb7-1100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750564</span></span>
<span id="cb7-1101"><a href="#cb7-1101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1102"><a href="#cb7-1102" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1103"><a href="#cb7-1103" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================================</span>         <span class="er">|</span>  <span class="dv">88</span>%</span>
<span id="cb7-1104"><a href="#cb7-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1105"><a href="#cb7-1105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750394: No data found.</span></span>
<span id="cb7-1106"><a href="#cb7-1106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1107"><a href="#cb7-1107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750394: No data found.</span></span>
<span id="cb7-1108"><a href="#cb7-1108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1109"><a href="#cb7-1109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750394</span></span>
<span id="cb7-1110"><a href="#cb7-1110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1111"><a href="#cb7-1111" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1112"><a href="#cb7-1112" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================================</span>        <span class="er">|</span>  <span class="dv">88</span>%</span>
<span id="cb7-1113"><a href="#cb7-1113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1114"><a href="#cb7-1114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750522: No data found.</span></span>
<span id="cb7-1115"><a href="#cb7-1115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1116"><a href="#cb7-1116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750522: No data found.</span></span>
<span id="cb7-1117"><a href="#cb7-1117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1118"><a href="#cb7-1118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750522</span></span>
<span id="cb7-1119"><a href="#cb7-1119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1120"><a href="#cb7-1120" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1121"><a href="#cb7-1121" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================================</span>        <span class="er">|</span>  <span class="dv">89</span>%</span>
<span id="cb7-1122"><a href="#cb7-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1123"><a href="#cb7-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750438: No data found.</span></span>
<span id="cb7-1124"><a href="#cb7-1124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1125"><a href="#cb7-1125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750438: No data found.</span></span>
<span id="cb7-1126"><a href="#cb7-1126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1127"><a href="#cb7-1127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750438</span></span>
<span id="cb7-1128"><a href="#cb7-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1129"><a href="#cb7-1129" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1130"><a href="#cb7-1130" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================================</span>       <span class="er">|</span>  <span class="dv">89</span>%</span>
<span id="cb7-1131"><a href="#cb7-1131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1132"><a href="#cb7-1132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750465: No data found.</span></span>
<span id="cb7-1133"><a href="#cb7-1133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1134"><a href="#cb7-1134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750465: No data found.</span></span>
<span id="cb7-1135"><a href="#cb7-1135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1136"><a href="#cb7-1136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750465</span></span>
<span id="cb7-1137"><a href="#cb7-1137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1138"><a href="#cb7-1138" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1139"><a href="#cb7-1139" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================================</span>       <span class="er">|</span>  <span class="dv">90</span>%</span>
<span id="cb7-1140"><a href="#cb7-1140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1141"><a href="#cb7-1141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750500: No data found.</span></span>
<span id="cb7-1142"><a href="#cb7-1142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1143"><a href="#cb7-1143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750500: No data found.</span></span>
<span id="cb7-1144"><a href="#cb7-1144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1145"><a href="#cb7-1145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750500</span></span>
<span id="cb7-1146"><a href="#cb7-1146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1147"><a href="#cb7-1147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750582: No data found.</span></span>
<span id="cb7-1148"><a href="#cb7-1148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1149"><a href="#cb7-1149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750582: No data found.</span></span>
<span id="cb7-1150"><a href="#cb7-1150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1151"><a href="#cb7-1151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750582</span></span>
<span id="cb7-1152"><a href="#cb7-1152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1153"><a href="#cb7-1153" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1154"><a href="#cb7-1154" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================================</span>      <span class="er">|</span>  <span class="dv">91</span>%</span>
<span id="cb7-1155"><a href="#cb7-1155" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1156"><a href="#cb7-1156" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================================</span>      <span class="er">|</span>  <span class="dv">92</span>%</span>
<span id="cb7-1157"><a href="#cb7-1157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1158"><a href="#cb7-1158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750589: No data found.</span></span>
<span id="cb7-1159"><a href="#cb7-1159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1160"><a href="#cb7-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750589: No data found.</span></span>
<span id="cb7-1161"><a href="#cb7-1161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1162"><a href="#cb7-1162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750589</span></span>
<span id="cb7-1163"><a href="#cb7-1163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1164"><a href="#cb7-1164" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1165"><a href="#cb7-1165" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================================================</span>     <span class="er">|</span>  <span class="dv">93</span>%</span>
<span id="cb7-1166"><a href="#cb7-1166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1167"><a href="#cb7-1167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750596: No data found.</span></span>
<span id="cb7-1168"><a href="#cb7-1168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1169"><a href="#cb7-1169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750596: No data found.</span></span>
<span id="cb7-1170"><a href="#cb7-1170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1171"><a href="#cb7-1171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750596</span></span>
<span id="cb7-1172"><a href="#cb7-1172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1173"><a href="#cb7-1173" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1174"><a href="#cb7-1174" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================================</span>    <span class="er">|</span>  <span class="dv">94</span>%</span>
<span id="cb7-1175"><a href="#cb7-1175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1176"><a href="#cb7-1176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750556: No data found.</span></span>
<span id="cb7-1177"><a href="#cb7-1177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1178"><a href="#cb7-1178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750556: No data found.</span></span>
<span id="cb7-1179"><a href="#cb7-1179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1180"><a href="#cb7-1180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750556</span></span>
<span id="cb7-1181"><a href="#cb7-1181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1182"><a href="#cb7-1182" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1183"><a href="#cb7-1183" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================================</span>    <span class="er">|</span>  <span class="dv">95</span>%</span>
<span id="cb7-1184"><a href="#cb7-1184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1185"><a href="#cb7-1185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750560: No data found.</span></span>
<span id="cb7-1186"><a href="#cb7-1186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1187"><a href="#cb7-1187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750560: No data found.</span></span>
<span id="cb7-1188"><a href="#cb7-1188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1189"><a href="#cb7-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750560</span></span>
<span id="cb7-1190"><a href="#cb7-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1191"><a href="#cb7-1191" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1192"><a href="#cb7-1192" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================================</span>   <span class="er">|</span>  <span class="dv">95</span>%</span>
<span id="cb7-1193"><a href="#cb7-1193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1194"><a href="#cb7-1194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750517: No data found.</span></span>
<span id="cb7-1195"><a href="#cb7-1195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1196"><a href="#cb7-1196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750517: No data found.</span></span>
<span id="cb7-1197"><a href="#cb7-1197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1198"><a href="#cb7-1198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750517</span></span>
<span id="cb7-1199"><a href="#cb7-1199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1200"><a href="#cb7-1200" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1201"><a href="#cb7-1201" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================================</span>   <span class="er">|</span>  <span class="dv">96</span>%</span>
<span id="cb7-1202"><a href="#cb7-1202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1203"><a href="#cb7-1203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750509: No data found.</span></span>
<span id="cb7-1204"><a href="#cb7-1204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1205"><a href="#cb7-1205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750509: No data found.</span></span>
<span id="cb7-1206"><a href="#cb7-1206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1207"><a href="#cb7-1207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750509</span></span>
<span id="cb7-1208"><a href="#cb7-1208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1209"><a href="#cb7-1209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750541: No data found.</span></span>
<span id="cb7-1210"><a href="#cb7-1210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1211"><a href="#cb7-1211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750541: No data found.</span></span>
<span id="cb7-1212"><a href="#cb7-1212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1213"><a href="#cb7-1213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750541</span></span>
<span id="cb7-1214"><a href="#cb7-1214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1215"><a href="#cb7-1215" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1216"><a href="#cb7-1216" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================================</span>  <span class="er">|</span>  <span class="dv">97</span>%</span>
<span id="cb7-1217"><a href="#cb7-1217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1218"><a href="#cb7-1218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750474: No data found.</span></span>
<span id="cb7-1219"><a href="#cb7-1219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1220"><a href="#cb7-1220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750474: No data found.</span></span>
<span id="cb7-1221"><a href="#cb7-1221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1222"><a href="#cb7-1222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750474</span></span>
<span id="cb7-1223"><a href="#cb7-1223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1224"><a href="#cb7-1224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750546: No data found.</span></span>
<span id="cb7-1225"><a href="#cb7-1225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1226"><a href="#cb7-1226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750546: No data found.</span></span>
<span id="cb7-1227"><a href="#cb7-1227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1228"><a href="#cb7-1228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750546</span></span>
<span id="cb7-1229"><a href="#cb7-1229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1230"><a href="#cb7-1230" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1231"><a href="#cb7-1231" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================================================</span> <span class="er">|</span>  <span class="dv">98</span>%</span>
<span id="cb7-1232"><a href="#cb7-1232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1233"><a href="#cb7-1233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750410: No data found.</span></span>
<span id="cb7-1234"><a href="#cb7-1234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1235"><a href="#cb7-1235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750410: No data found.</span></span>
<span id="cb7-1236"><a href="#cb7-1236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1237"><a href="#cb7-1237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750410</span></span>
<span id="cb7-1238"><a href="#cb7-1238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1239"><a href="#cb7-1239" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1240"><a href="#cb7-1240" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================================================</span> <span class="er">|</span>  <span class="dv">99</span>%</span>
<span id="cb7-1241"><a href="#cb7-1241" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1242"><a href="#cb7-1242" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750395: No data found.</span></span>
<span id="cb7-1243"><a href="#cb7-1243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1244"><a href="#cb7-1244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750395: No data found.</span></span>
<span id="cb7-1245"><a href="#cb7-1245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1246"><a href="#cb7-1246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750395</span></span>
<span id="cb7-1247"><a href="#cb7-1247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1248"><a href="#cb7-1248" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1249"><a href="#cb7-1249" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span>  <span class="dv">99</span>%</span>
<span id="cb7-1250"><a href="#cb7-1250" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1251"><a href="#cb7-1251" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750525: No data found.</span></span>
<span id="cb7-1252"><a href="#cb7-1252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1253"><a href="#cb7-1253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; samples/SAMEA112750525: No data found.</span></span>
<span id="cb7-1254"><a href="#cb7-1254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: </span></span>
<span id="cb7-1255"><a href="#cb7-1255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyses not found for samples SAMEA112750525</span></span>
<span id="cb7-1256"><a href="#cb7-1256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-1257"><a href="#cb7-1257" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb7-1258"><a href="#cb7-1258" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><code>salmon_analysis_ids</code> character vector holds associations of HoloFood metagenomic amplicon accession numbers (SAMEAxxxxxx) to their counterparts in MGnify database (MGYAxxxxxx).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get metagenomic taxonomic data for salmon from MGnify</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> MGnifyR<span class="sc">::</span><span class="fu">getResult</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> mg,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">accession =</span> salmon_analysis_ids,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">get.func =</span> <span class="cn">FALSE</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fetching taxonomy data...</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==</span>                                                                    <span class="er">|</span>   <span class="dv">2</span>%</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===</span>                                                                   <span class="er">|</span>   <span class="dv">4</span>%</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====</span>                                                                 <span class="er">|</span>   <span class="dv">7</span>%</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======</span>                                                                <span class="er">|</span>   <span class="dv">9</span>%</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========</span>                                                              <span class="er">|</span>  <span class="dv">11</span>%</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========</span>                                                             <span class="er">|</span>  <span class="dv">13</span>%</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========</span>                                                           <span class="er">|</span>  <span class="dv">16</span>%</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============</span>                                                          <span class="er">|</span>  <span class="dv">18</span>%</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============</span>                                                        <span class="er">|</span>  <span class="dv">20</span>%</span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================</span>                                                      <span class="er">|</span>  <span class="dv">22</span>%</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================</span>                                                     <span class="er">|</span>  <span class="dv">24</span>%</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================</span>                                                   <span class="er">|</span>  <span class="dv">27</span>%</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================</span>                                                  <span class="er">|</span>  <span class="dv">29</span>%</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================</span>                                                <span class="er">|</span>  <span class="dv">31</span>%</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">33</span>%</span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================</span>                                             <span class="er">|</span>  <span class="dv">36</span>%</span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================</span>                                            <span class="er">|</span>  <span class="dv">38</span>%</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================</span>                                          <span class="er">|</span>  <span class="dv">40</span>%</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================</span>                                        <span class="er">|</span>  <span class="dv">42</span>%</span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================</span>                                       <span class="er">|</span>  <span class="dv">44</span>%</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================</span>                                     <span class="er">|</span>  <span class="dv">47</span>%</span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================</span>                                    <span class="er">|</span>  <span class="dv">49</span>%</span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================</span>                                  <span class="er">|</span>  <span class="dv">51</span>%</span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================</span>                                 <span class="er">|</span>  <span class="dv">53</span>%</span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================</span>                               <span class="er">|</span>  <span class="dv">56</span>%</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================</span>                              <span class="er">|</span>  <span class="dv">58</span>%</span>
<span id="cb8-141"><a href="#cb8-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-142"><a href="#cb8-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================</span>                            <span class="er">|</span>  <span class="dv">60</span>%</span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================</span>                          <span class="er">|</span>  <span class="dv">62</span>%</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================</span>                         <span class="er">|</span>  <span class="dv">64</span>%</span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-160"><a href="#cb8-160" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">67</span>%</span>
<span id="cb8-161"><a href="#cb8-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-162"><a href="#cb8-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-163"><a href="#cb8-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-164"><a href="#cb8-164" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-165"><a href="#cb8-165" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================</span>                      <span class="er">|</span>  <span class="dv">69</span>%</span>
<span id="cb8-166"><a href="#cb8-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-167"><a href="#cb8-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-168"><a href="#cb8-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-169"><a href="#cb8-169" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-170"><a href="#cb8-170" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================</span>                    <span class="er">|</span>  <span class="dv">71</span>%</span>
<span id="cb8-171"><a href="#cb8-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-172"><a href="#cb8-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-173"><a href="#cb8-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-174"><a href="#cb8-174" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-175"><a href="#cb8-175" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================</span>                   <span class="er">|</span>  <span class="dv">73</span>%</span>
<span id="cb8-176"><a href="#cb8-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-177"><a href="#cb8-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-178"><a href="#cb8-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-179"><a href="#cb8-179" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-180"><a href="#cb8-180" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================================</span>                 <span class="er">|</span>  <span class="dv">76</span>%</span>
<span id="cb8-181"><a href="#cb8-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-182"><a href="#cb8-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-183"><a href="#cb8-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-184"><a href="#cb8-184" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-185"><a href="#cb8-185" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================</span>                <span class="er">|</span>  <span class="dv">78</span>%</span>
<span id="cb8-186"><a href="#cb8-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-187"><a href="#cb8-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-188"><a href="#cb8-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-189"><a href="#cb8-189" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-190"><a href="#cb8-190" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================================</span>              <span class="er">|</span>  <span class="dv">80</span>%</span>
<span id="cb8-191"><a href="#cb8-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-192"><a href="#cb8-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-193"><a href="#cb8-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-194"><a href="#cb8-194" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-195"><a href="#cb8-195" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================================</span>            <span class="er">|</span>  <span class="dv">82</span>%</span>
<span id="cb8-196"><a href="#cb8-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-197"><a href="#cb8-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-198"><a href="#cb8-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-199"><a href="#cb8-199" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-200"><a href="#cb8-200" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================================</span>           <span class="er">|</span>  <span class="dv">84</span>%</span>
<span id="cb8-201"><a href="#cb8-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-202"><a href="#cb8-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-203"><a href="#cb8-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-204"><a href="#cb8-204" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-205"><a href="#cb8-205" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================================</span>         <span class="er">|</span>  <span class="dv">87</span>%</span>
<span id="cb8-206"><a href="#cb8-206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-207"><a href="#cb8-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-208"><a href="#cb8-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-209"><a href="#cb8-209" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-210"><a href="#cb8-210" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================================</span>        <span class="er">|</span>  <span class="dv">89</span>%</span>
<span id="cb8-211"><a href="#cb8-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-212"><a href="#cb8-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-213"><a href="#cb8-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-214"><a href="#cb8-214" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-215"><a href="#cb8-215" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================================</span>      <span class="er">|</span>  <span class="dv">91</span>%</span>
<span id="cb8-216"><a href="#cb8-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-217"><a href="#cb8-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-218"><a href="#cb8-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-219"><a href="#cb8-219" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-220"><a href="#cb8-220" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================================================</span>     <span class="er">|</span>  <span class="dv">93</span>%</span>
<span id="cb8-221"><a href="#cb8-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-222"><a href="#cb8-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-223"><a href="#cb8-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-224"><a href="#cb8-224" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-225"><a href="#cb8-225" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================================</span>   <span class="er">|</span>  <span class="dv">96</span>%</span>
<span id="cb8-226"><a href="#cb8-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-227"><a href="#cb8-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-228"><a href="#cb8-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-229"><a href="#cb8-229" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-230"><a href="#cb8-230" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================================</span>  <span class="er">|</span>  <span class="dv">98</span>%</span>
<span id="cb8-231"><a href="#cb8-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span id="cb8-232"><a href="#cb8-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span>
<span id="cb8-233"><a href="#cb8-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-234"><a href="#cb8-234" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-235"><a href="#cb8-235" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb8-236"><a href="#cb8-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Merging with full join...</span></span>
<span id="cb8-237"><a href="#cb8-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1/45</span></span>
<span id="cb8-238"><a href="#cb8-238" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-239"><a href="#cb8-239" aria-hidden="true" tabindex="-1"></a><span class="dv">3</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-240"><a href="#cb8-240" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-241"><a href="#cb8-241" aria-hidden="true" tabindex="-1"></a><span class="dv">5</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-242"><a href="#cb8-242" aria-hidden="true" tabindex="-1"></a><span class="dv">6</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-243"><a href="#cb8-243" aria-hidden="true" tabindex="-1"></a><span class="dv">7</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-244"><a href="#cb8-244" aria-hidden="true" tabindex="-1"></a><span class="dv">8</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-245"><a href="#cb8-245" aria-hidden="true" tabindex="-1"></a><span class="dv">9</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-246"><a href="#cb8-246" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-247"><a href="#cb8-247" aria-hidden="true" tabindex="-1"></a><span class="dv">11</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-248"><a href="#cb8-248" aria-hidden="true" tabindex="-1"></a><span class="dv">12</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-249"><a href="#cb8-249" aria-hidden="true" tabindex="-1"></a><span class="dv">13</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-250"><a href="#cb8-250" aria-hidden="true" tabindex="-1"></a><span class="dv">14</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-251"><a href="#cb8-251" aria-hidden="true" tabindex="-1"></a><span class="dv">15</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-252"><a href="#cb8-252" aria-hidden="true" tabindex="-1"></a><span class="dv">16</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-253"><a href="#cb8-253" aria-hidden="true" tabindex="-1"></a><span class="dv">17</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-254"><a href="#cb8-254" aria-hidden="true" tabindex="-1"></a><span class="dv">18</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-255"><a href="#cb8-255" aria-hidden="true" tabindex="-1"></a><span class="dv">19</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-256"><a href="#cb8-256" aria-hidden="true" tabindex="-1"></a><span class="dv">20</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-257"><a href="#cb8-257" aria-hidden="true" tabindex="-1"></a><span class="dv">21</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-258"><a href="#cb8-258" aria-hidden="true" tabindex="-1"></a><span class="dv">22</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-259"><a href="#cb8-259" aria-hidden="true" tabindex="-1"></a><span class="dv">23</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-260"><a href="#cb8-260" aria-hidden="true" tabindex="-1"></a><span class="dv">24</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-261"><a href="#cb8-261" aria-hidden="true" tabindex="-1"></a><span class="dv">25</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-262"><a href="#cb8-262" aria-hidden="true" tabindex="-1"></a><span class="dv">26</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-263"><a href="#cb8-263" aria-hidden="true" tabindex="-1"></a><span class="dv">27</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-264"><a href="#cb8-264" aria-hidden="true" tabindex="-1"></a><span class="dv">28</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-265"><a href="#cb8-265" aria-hidden="true" tabindex="-1"></a><span class="dv">29</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-266"><a href="#cb8-266" aria-hidden="true" tabindex="-1"></a><span class="dv">30</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-267"><a href="#cb8-267" aria-hidden="true" tabindex="-1"></a><span class="dv">31</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-268"><a href="#cb8-268" aria-hidden="true" tabindex="-1"></a><span class="dv">32</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-269"><a href="#cb8-269" aria-hidden="true" tabindex="-1"></a><span class="dv">33</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-270"><a href="#cb8-270" aria-hidden="true" tabindex="-1"></a><span class="dv">34</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-271"><a href="#cb8-271" aria-hidden="true" tabindex="-1"></a><span class="dv">35</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-272"><a href="#cb8-272" aria-hidden="true" tabindex="-1"></a><span class="dv">36</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-273"><a href="#cb8-273" aria-hidden="true" tabindex="-1"></a><span class="dv">37</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-274"><a href="#cb8-274" aria-hidden="true" tabindex="-1"></a><span class="dv">38</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-275"><a href="#cb8-275" aria-hidden="true" tabindex="-1"></a><span class="dv">39</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-276"><a href="#cb8-276" aria-hidden="true" tabindex="-1"></a><span class="dv">40</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-277"><a href="#cb8-277" aria-hidden="true" tabindex="-1"></a><span class="dv">41</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-278"><a href="#cb8-278" aria-hidden="true" tabindex="-1"></a><span class="dv">42</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-279"><a href="#cb8-279" aria-hidden="true" tabindex="-1"></a><span class="dv">43</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-280"><a href="#cb8-280" aria-hidden="true" tabindex="-1"></a><span class="dv">44</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-281"><a href="#cb8-281" aria-hidden="true" tabindex="-1"></a><span class="dv">45</span><span class="sc">/</span><span class="dv">45</span></span>
<span id="cb8-282"><a href="#cb8-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-283"><a href="#cb8-283" aria-hidden="true" tabindex="-1"></a><span class="co"># Save salmon metagenomic TSE</span></span>
<span id="cb8-284"><a href="#cb8-284" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> tse, <span class="at">file =</span> <span class="st">"../inst/extdata/salmon_metagenomic_tse.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Data fetched from MGnify has MGnify-specific identifiers. We have to first rename samples with HoloFood specific ID and then add the data to MultiAssayExperiment combining all the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in salmon metagenomic TSE object</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">"../inst/extdata/salmon_metagenomic_tse.RDS"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>.add_MGnify <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>        mae, tse, <span class="at">holofood.id =</span> <span class="st">"sample_biosample"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">exp.name =</span> <span class="st">"metagenomic_amplicon"</span>, <span class="at">new.exp =</span> <span class="st">"metagenomic"</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">replace =</span> <span class="cn">TRUE</span>){</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rename columns based on HoloFood ID</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(tse) <span class="ot">&lt;-</span> <span class="fu">colData</span>(tse)[[holofood.id]]</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Merge add MGnify metagenomic data to the existing metagenomic_amplicon</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># experiment. Get the experiment from HoloFood data.</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    tse2 <span class="ot">&lt;-</span> mae[[exp.name]]</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Combine sample metadata</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    cd1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    cd2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse2))</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    cd <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>        cd1, cd2,</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">by.x =</span> holofood.id,</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">by.y =</span> <span class="st">"accession"</span>,</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">all.x =</span> <span class="cn">TRUE</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(cd) <span class="ot">&lt;-</span> cd[[holofood.id]]</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Now order colData to ensure that order is correct</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>    cd <span class="ot">&lt;-</span> cd[<span class="fu">colnames</span>(tse), ]</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add it to TreeSE</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colData</span>(tse) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(cd, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rename to match HoloFood</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(tse) <span class="ot">&lt;-</span> <span class="fu">colData</span>(tse)[[holofood.id]]</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Now order the data based on TreeSE in MAE</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    <span class="co"># First take those columns that can be found</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    mae[[exp.name]] <span class="ot">&lt;-</span> mae[[exp.name]][</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>        , <span class="fu">colnames</span>(mae[[exp.name]]) <span class="sc">%in%</span> <span class="fu">colnames</span>(tse) ]</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Then order</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>    tse <span class="ot">&lt;-</span> tse[, <span class="fu">match</span>(<span class="fu">colnames</span>(mae[[exp.name]]), <span class="fu">colnames</span>(tse))]</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># And add to MAE in place of old experiment</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( replace ){</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>        mae[[exp.name]] <span class="ot">&lt;-</span> tse</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span>{</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Get sample map</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>        sample_map <span class="ot">&lt;-</span> <span class="fu">sampleMap</span>(mae)</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Get samples that are matching with those one that are being added</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>        add_sample_map <span class="ot">&lt;-</span> sample_map[</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>            <span class="fu">match</span>(<span class="fu">colnames</span>(tse), sample_map[[<span class="st">"colname"</span>]]), ]</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Rename the experiment</span></span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>        add_sample_map[[<span class="st">"assay"</span>]] <span class="ot">&lt;-</span> new.exp</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add to sample map'</span></span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>        sample_map <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sample_map, add_sample_map)</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create MAE</span></span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>        tse <span class="ot">&lt;-</span> <span class="fu">ExperimentList</span>(<span class="at">temp =</span> tse)</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(tse) <span class="ot">&lt;-</span> new.exp</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>        exp_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">experiments</span>(mae), tse)</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>        mae <span class="ot">&lt;-</span> <span class="fu">MultiAssayExperiment</span>(</span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>            exp_list, <span class="at">colData =</span> <span class="fu">colData</span>(mae), <span class="at">sampleMap =</span> sample_map)</span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(mae)</span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> <span class="fu">.add_MGnify</span>(mae, tse)</span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; harmonizing input:</span></span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   removing 144 sampleMap rows with 'colname' not in colnames of experiments</span></span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to merged MAE object to RDS</span></span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> mae, <span class="at">file =</span> <span class="st">"../inst/extdata/salmon_mae.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-preprocess" class="level2">
<h2 class="anchored" data-anchor-id="data-preprocess">Data preprocess</h2>
<p>Data cleaning is one of the most time-consuming and most important steps in data cleaning. For instance, we need to handle missing data, transform data assays, and agglomerate the data.</p>
<p>In the next steps, we will:</p>
<ol type="1">
<li>Remove unnecessary data</li>
<li>Convert data from string to numeric format</li>
<li>Transform data with logarithmic methods to prepare it for statistical analysis. In particular, we will use log10-transformation for fatty acid data and sequential transformation from counts to relative abundance to centered log-ratio transformation for compositional metagenomic data</li>
<li>Agglomerate metagenomic data by prevalence. We will keep only the species that are present in more that a specified percentage of samples and are only present at certain percentage of relative abundance in each sample. This step is important to filter out species that are present in only a small number of species and/or at lower abundances.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># path &lt;- system.file("extdata", "salmon_mae.RDS", package = "HoloFoodR")</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">#salmon_mae &lt;- readRDS(path)</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mae &lt;- readRDS(path)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../inst/extdata/salmon_mae.RDS"</span>)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Fetch only experiments that we need</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> mae[, , <span class="fu">c</span>(<span class="st">"fatty_acids_mg"</span>, <span class="st">"metagenomic_amplicon"</span>)]</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: 'experiments' dropped; see 'drops()'</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; harmonizing input:</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   removing 934 sampleMap rows not in names(experiments)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># From metabolomic data, remove organ-fatty acids row because it only contains a</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># string value "muscle"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mae[[<span class="dv">1</span>]]</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> tse[<span class="sc">!</span>(<span class="fu">rowData</span>(tse)[[<span class="st">"marker.name"</span>]] <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Organ-fatty acids"</span>)), ]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> tse</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform matrices to numeric. Some values are "&lt; 0.01"</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># If a number is &lt; 0.01, assume it to be 0</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">"counts"</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>assay[assay <span class="sc">==</span> <span class="st">"&lt;0.01"</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">apply</span>(assay, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">","</span>, <span class="st">"."</span>, x)))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Reassign assay back to MAE</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">"counts"</span>) <span class="ot">&lt;-</span> assay</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># From metagenomic data, remove those rows that do not have taxonomy info</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>not_empty <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">apply</span>(<span class="fu">rowData</span>(mae[[<span class="dv">2</span>]]), <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">all</span>(<span class="fu">is.na</span>(x)))</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> mae[[<span class="dv">2</span>]][not_empty, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For demonstration purposes, we will focus on investigating two experiments (<code>fatty_acids_mg</code> and <code>metagenomic_amplicon</code>) within the trial A performed by the HoloFood consortium. This trial the health effects of fermented seaweed added to the diet of salmons.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter MAE object to include only Trial A</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> mae[ , <span class="fu">colData</span>(mae)[[<span class="st">"Trial code"</span>]] <span class="sc">==</span> <span class="st">"SA"</span>, ]</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>mae</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A MultiAssayExperiment object of 2 listed</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Containing an ExperimentList class object of length 2:</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] fatty_acids_mg: TreeSummarizedExperiment with 60 rows and 45 columns</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [2] metagenomic_amplicon: TreeSummarizedExperiment with 384 rows and 45 columns</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Functionality:</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Before proceeding, we keep the modified version of MultiAssayExperiment object intact and only modify its copy.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>mae_sub <span class="ot">&lt;-</span> mae <span class="co"># Keep original MAE intact</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then we need to filter out all samples that are not found in both experiments since we are interested in multi-omic data analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Take those samples that can be found from both experiments</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>mae_sub <span class="ot">&lt;-</span> <span class="fu">intersectColumns</span>(mae_sub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we transform the metagenomic and metabolomic data to normalize the data, especially compositional data, such as metagenomic counts.</p>
<p>We log-transform fatty acid assay.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform fatty acids in mg with log10</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>mae_sub[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformAssay</span>(</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    mae_sub[[<span class="dv">1</span>]],</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay.type =</span> <span class="st">"counts"</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">MARGIN =</span> <span class="st">"samples"</span>,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"log10"</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pseudocount =</span> <span class="cn">TRUE</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A pseudocount of 0.025 was applied.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we transform metagenomic counts first into relative abundances, and then use centered log-ratio method to tackle the compositional data (see <span class="citation" data-cites="quinnFieldGuideCompositional2019">Quinn et al. (<a href="#ref-quinnFieldGuideCompositional2019" role="doc-biblioref">2019</a>)</span>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform microbiome with centered log-ratio method</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>mae_sub[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">transformAssay</span>(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    mae_sub[[<span class="dv">2</span>]], <span class="at">method =</span> <span class="st">"relabundance"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>mae_sub[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">transformAssay</span>(</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    mae_sub[[<span class="dv">2</span>]],</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"clr"</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">pseudocount =</span> <span class="cn">TRUE</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A pseudocount of 3.86934012273547e-06 was applied.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we can agglomerate features by prevalence to reduce the number of low-abundant species.</p>
<p>First, we visualize prevalence distribution with a histogram to decide the prevalence threshold to use. We use 0.01% detection level to filter out extremely low-abundant species present in each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove taxonomy_unparsed column from rowData of metagenomic_amplicon</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(mae_sub[[<span class="dv">2</span>]]) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mae_sub[[<span class="dv">2</span>]])[, <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(<span class="fu">rowData</span>(mae_sub[[<span class="dv">2</span>]])) <span class="sc">-</span> <span class="dv">1</span>)]</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>prevalence <span class="ot">&lt;-</span> <span class="fu">getPrevalence</span>(mae_sub[[<span class="dv">2</span>]],</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">rank =</span> <span class="st">"Genus"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">detection =</span> <span class="fl">0.001</span>)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: 'clr' includes negative values.</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Agglomeration of it might lead to meaningless values.</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Check the assay, and consider doing transformation againmanually with agglomerated data.</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Exclude microbes with 0 prevalence</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>prevalence <span class="ot">&lt;-</span> prevalence[prevalence <span class="sc">!=</span> <span class="dv">0</span>]</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>prevalence <span class="sc">|&gt;</span> <span class="fu">hist</span>(<span class="at">breaks =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/prevalence_histgoram-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can also look at the raw prevalence numbers</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort prevalence in decreasing order</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(prevalence, <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Mycoplasma Photobacterium     Aliivibrio  Cetobacterium         Vibrio </span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      1.0000000      0.8000000      0.4444444      0.4000000      0.3333333 </span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    Francisella Ammopiptanthus  Mycobacterium       Medicago    Paucibacter </span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      0.2444444      0.2222222      0.1777778      0.1555556      0.1333333</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><em>Mycoplasma</em> is present in all samples, which is not surprising as this genus was found to be one of the most common in salmon intestine (see <span class="citation" data-cites="zarkasiPyrosequencingbasedCharacterizationGastrointestinal2014">Zarkasi et al. (<a href="#ref-zarkasiPyrosequencingbasedCharacterizationGastrointestinal2014" role="doc-biblioref">2014</a>)</span>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare two sets of animal samples in order to add the treatment concentration</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># column to metagenomic data</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">colData</span>(mae_sub)<span class="sc">$</span>animal, <span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]])<span class="sc">$</span>animal) <span class="co"># No difference</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co"># We then can simply add treatment_concentration column to metagenomic data</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]])<span class="sc">$</span>treatment_concentration <span class="ot">&lt;-</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colData</span>(mae_sub)<span class="sc">$</span><span class="st">`</span><span class="at">Treatment concentration</span><span class="st">`</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Add treatment_group (control vs. treatment) to metagenomic data</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>metagenomic_data <span class="ot">&lt;-</span> <span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]])</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>metagenomic_data<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(metagenomic_data<span class="sc">$</span>treatment_concentration <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"control"</span>, <span class="st">"treatment"</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]]) <span class="ot">&lt;-</span> metagenomic_data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="agglomerate-by-prevalence-and-sort-by-the-number-of-reads" class="level3">
<h3 class="anchored" data-anchor-id="agglomerate-by-prevalence-and-sort-by-the-number-of-reads">Agglomerate by prevalence and sort by the number of reads</h3>
<p>With a reasonable prevalence of 20%, and relative abundance detection threshold of 0.2%, only bacterial genera remain.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate by prevalence by genus</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(mae_sub[[<span class="dv">2</span>]], <span class="st">"prev_genus"</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByPrevalence</span>(mae_sub[[<span class="dv">2</span>]],</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">rank =</span> <span class="st">"Genus"</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">agg.na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">prevalence =</span> <span class="dv">20</span> <span class="sc">/</span> <span class="dv">100</span>,</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">detection =</span> <span class="fl">0.002</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: 'clr' includes negative values.</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Agglomeration of it might lead to meaningless values.</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Check the assay, and consider doing transformation againmanually with agglomerated data.</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: 'clr' includes negative values.</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Agglomeration of it might lead to meaningless values.</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Check the assay, and consider doing transformation againmanually with agglomerated data.</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: 'clr' includes negative values.</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Agglomeration of it might lead to meaningless values.</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Check the assay, and consider doing transformation againmanually with agglomerated data.</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(mae_sub[[<span class="dv">2</span>]], <span class="st">"prev_genus"</span>)</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 6 45 </span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(2): agglomerated_by_rank agglomerated_by_rank</span></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(3): counts relabundance clr</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(6): Aliivibrio Cetobacterium ... Vibrio Other</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames(45): SAMEA112642746 SAMEA112642736 ... SAMEA112642770</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   SAMEA112642752</span></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(134): sample_biosample analysis_experiment.type ...</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   treatment_concentration treatment_group</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(0):</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowLinks: NULL</span></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowTree: NULL</span></span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colLinks: NULL</span></span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We now can extract only log10-transformed metabolomic assays because raw concentrations are no longer needed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract only transformed metabolomic assays</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(mae_sub[[<span class="dv">1</span>]]) <span class="ot">&lt;-</span> <span class="fu">assays</span>(mae_sub[[<span class="dv">1</span>]])[<span class="fu">names</span>(<span class="fu">assays</span>(mae_sub[[<span class="dv">1</span>]])) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"log10"</span>)]</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename column names to make them more consistent</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(mae_sub)) <span class="ot">&lt;-</span> <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">names</span>(<span class="fu">colData</span>(mae_sub))))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add salmon host mass to colData</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>merged_coldata <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> <span class="fu">colData</span>(mae_sub), <span class="at">y =</span> <span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]])[<span class="fu">c</span>(<span class="st">"animal"</span>, <span class="st">"host.gutted.mass"</span>)], <span class="at">all.x =</span> <span class="cn">TRUE</span>, <span class="at">by =</span> <span class="st">"animal"</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_coldata) <span class="ot">&lt;-</span> merged_coldata<span class="sc">$</span>animal</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_sub) <span class="ot">&lt;-</span> merged_coldata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We need to add a few columns for the subsequent data analysis steps. We are interested in investigating potential differences between treatment and control. Furthermore, we will only focus on salmons at day 60 because these salmons were sampled after experimental period. In contrast, day-0 salmons that were sampled immediately in the beginning of the trial.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_sub)[<span class="st">"treatment_group"</span>] <span class="ot">&lt;-</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">colData</span>(mae_sub)<span class="sc">$</span>treatment_concentration <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"control"</span>, <span class="st">"treatment"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract metagenomic data</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mae[[<span class="dv">2</span>]]</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add treatment vs. control into TSE object at day 60</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>day60 <span class="ot">&lt;-</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>trial.timepoint <span class="sc">==</span> <span class="dv">60</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> tse[, day60]</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse)<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">colData</span>(tse)<span class="sc">$</span>host.diet.treatment.concentration <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"control"</span>, <span class="st">"treatment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-exploration" class="level2">
<h2 class="anchored" data-anchor-id="data-exploration">Data exploration</h2>
<p>Before proceeding to data analysis, we will employ some basic data exploration methods:</p>
<ol type="1">
<li>Abundance plot of metagenomic species</li>
<li>Alpha-diversity</li>
</ol>
<section id="abundance-plot" class="level3">
<h3 class="anchored" data-anchor-id="abundance-plot">Abundance plot</h3>
<p>And abundance plot of centered-log-ratio transformed metagenomic species is important for general understanding of the data. Additionally, we also indicate “control” and “treatment” groups with colors for initial assessment of difference between the groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundanceDensity</span>(<span class="fu">altExp</span>(mae_sub[[<span class="dv">2</span>]], <span class="st">"prev_genus"</span>), <span class="at">assay.type =</span> <span class="st">"clr"</span>, <span class="at">colour_by =</span> <span class="st">"treatment_group"</span>) <span class="sc">+</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Abundance plot"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in getTopFeatures(object, top = n, assay.type = assay.type):</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'getTopFeatures' is deprecated. Use 'getTop' instead.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/plot_abundance-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="alpha-diversity" class="level3">
<h3 class="anchored" data-anchor-id="alpha-diversity">Alpha-diversity</h3>
<p>We can measure alpha-diversity of the microbiome experiment. This metric measures diversity within a sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate (observed) richness</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">estimateRichness</span>(tse, </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">assay.type =</span> <span class="st">"counts"</span>, </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">index =</span> <span class="st">"observed"</span>, </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">name=</span><span class="st">"observed"</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">"observed"</span>,</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">"treatment_group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/richness-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>To measure the diversity of species in each sample, we will use Shannon diversity and compare diversities in control and treatment groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">estimateDiversity</span>(tse, </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">assay.type =</span> <span class="st">"counts"</span>,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">index =</span> <span class="st">"shannon"</span>, </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">name =</span> <span class="st">"shannon"</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>treatment_group)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>comb <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="fu">t</span>(<span class="fu">combn</span>(<span class="fu">levels</span>(df<span class="sc">$</span>treatment_group), <span class="dv">2</span>)), </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>           <span class="fu">seq</span>(<span class="fu">nrow</span>(<span class="fu">t</span>(<span class="fu">combn</span>(<span class="fu">levels</span>(df<span class="sc">$</span>treatment_group), <span class="dv">2</span>)))))</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> treatment_group, <span class="at">y =</span> shannon)) <span class="sc">+</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_signif</span>(<span class="at">comparisons =</span> comb,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">map_signif_level =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/shannon_diversity-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="perform-cross-association-analysis" class="level2">
<h2 class="anchored" data-anchor-id="perform-cross-association-analysis">Perform cross-association analysis</h2>
<p>Before cross-association analysis, we can filter out all features with low standard deviation to exclude potential false positives from cross-association analysis. To decide the threshold of filtering, we can use a histogram of standard deviations</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>sd <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">apply</span>(<span class="fu">assay</span>(mae_sub[[<span class="dv">1</span>]], <span class="st">"log10"</span>), <span class="dv">1</span>, sd))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sd) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"standard_deviation"</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Histogram of standard deviations</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sd, <span class="fu">aes</span>(<span class="at">x =</span> standard_deviation)) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Standard deviations of log10-transformed fatty acid concentrations"</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/standard_deviation_hist-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>A good threshold would be at 0.1. Thus, we can filter out any fatty acid whose standard deviation of concentration is below 0.1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>tse_filter <span class="ot">&lt;-</span> mae_sub[[<span class="dv">1</span>]][<span class="fu">apply</span>(<span class="fu">assay</span>(mae_sub[[<span class="dv">1</span>]], <span class="st">"log10"</span>), <span class="dv">1</span>, sd) <span class="sc">&gt;</span> <span class="fl">0.1</span>, ]</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(mae_sub[[<span class="dv">1</span>]], <span class="st">"not_sd_0"</span>) <span class="ot">&lt;-</span> tse_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we can perform cross-association analysis using Kendall method and display significantly correlated species and fatty acids.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform cross-association analysis</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">testExperimentCrossAssociation</span>(</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    mae_sub,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">experiment1 =</span> <span class="dv">1</span>,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">experiment2 =</span> <span class="dv">2</span>,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">altexp1 =</span> <span class="st">"not_sd_0"</span>,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">altexp2 =</span> <span class="st">"prev_genus"</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"kendall"</span>,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay.type1 =</span> <span class="st">"log10"</span>,</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay.type2 =</span> <span class="st">"clr"</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">mode =</span> <span class="st">"matrix"</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in testExperimentCrossAssociation(mae_sub, experiment1 = 1, experiment2</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; = 2, : 'testExperimentCrossAssociation' is deprecated. Use</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'getCrossAssociation' instead.</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Calculating correlations...</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; altexp1: not_sd_0, altexp2: prev_genus, assay.type1: log10, col.var1: -, assay.type2: clr, col.var2: -</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; MARGIN: 1, function: stats::cor.test, method: kendall, test.signif: TRUE, p.adj.method: fdr, paired: FALSE, show.warnings: TRUE</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-132"><a href="#cb28-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-133"><a href="#cb28-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-134"><a href="#cb28-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-135"><a href="#cb28-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-136"><a href="#cb28-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-137"><a href="#cb28-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-138"><a href="#cb28-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-139"><a href="#cb28-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-140"><a href="#cb28-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-141"><a href="#cb28-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-142"><a href="#cb28-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-143"><a href="#cb28-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-144"><a href="#cb28-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-145"><a href="#cb28-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-146"><a href="#cb28-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-147"><a href="#cb28-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-148"><a href="#cb28-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-149"><a href="#cb28-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-150"><a href="#cb28-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-151"><a href="#cb28-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-152"><a href="#cb28-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-153"><a href="#cb28-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-154"><a href="#cb28-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-155"><a href="#cb28-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-156"><a href="#cb28-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-157"><a href="#cb28-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-158"><a href="#cb28-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-159"><a href="#cb28-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-160"><a href="#cb28-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-161"><a href="#cb28-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-162"><a href="#cb28-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-163"><a href="#cb28-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-164"><a href="#cb28-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-165"><a href="#cb28-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-166"><a href="#cb28-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-167"><a href="#cb28-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-168"><a href="#cb28-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-169"><a href="#cb28-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-170"><a href="#cb28-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-171"><a href="#cb28-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-172"><a href="#cb28-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-173"><a href="#cb28-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-174"><a href="#cb28-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-175"><a href="#cb28-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-176"><a href="#cb28-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-177"><a href="#cb28-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-178"><a href="#cb28-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-179"><a href="#cb28-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-180"><a href="#cb28-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-181"><a href="#cb28-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-182"><a href="#cb28-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-183"><a href="#cb28-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-184"><a href="#cb28-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-185"><a href="#cb28-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-186"><a href="#cb28-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-187"><a href="#cb28-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-188"><a href="#cb28-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-189"><a href="#cb28-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-190"><a href="#cb28-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-191"><a href="#cb28-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-192"><a href="#cb28-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-193"><a href="#cb28-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-194"><a href="#cb28-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-195"><a href="#cb28-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-196"><a href="#cb28-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-197"><a href="#cb28-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-198"><a href="#cb28-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-199"><a href="#cb28-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-200"><a href="#cb28-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-201"><a href="#cb28-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-202"><a href="#cb28-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-203"><a href="#cb28-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-204"><a href="#cb28-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-205"><a href="#cb28-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-206"><a href="#cb28-206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-207"><a href="#cb28-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-208"><a href="#cb28-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-209"><a href="#cb28-209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-210"><a href="#cb28-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-211"><a href="#cb28-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-212"><a href="#cb28-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-213"><a href="#cb28-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-214"><a href="#cb28-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-215"><a href="#cb28-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-216"><a href="#cb28-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-217"><a href="#cb28-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-218"><a href="#cb28-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-219"><a href="#cb28-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-220"><a href="#cb28-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-221"><a href="#cb28-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-222"><a href="#cb28-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-223"><a href="#cb28-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-224"><a href="#cb28-224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-225"><a href="#cb28-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-226"><a href="#cb28-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-227"><a href="#cb28-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-228"><a href="#cb28-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-229"><a href="#cb28-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-230"><a href="#cb28-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-231"><a href="#cb28-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-232"><a href="#cb28-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-233"><a href="#cb28-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-234"><a href="#cb28-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-235"><a href="#cb28-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-236"><a href="#cb28-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-237"><a href="#cb28-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-238"><a href="#cb28-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-239"><a href="#cb28-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-240"><a href="#cb28-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-241"><a href="#cb28-241" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-242"><a href="#cb28-242" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-243"><a href="#cb28-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-244"><a href="#cb28-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-245"><a href="#cb28-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-246"><a href="#cb28-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-247"><a href="#cb28-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-248"><a href="#cb28-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-249"><a href="#cb28-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-250"><a href="#cb28-250" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-251"><a href="#cb28-251" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-252"><a href="#cb28-252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-253"><a href="#cb28-253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-254"><a href="#cb28-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-255"><a href="#cb28-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-256"><a href="#cb28-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-257"><a href="#cb28-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-258"><a href="#cb28-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-259"><a href="#cb28-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-260"><a href="#cb28-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-261"><a href="#cb28-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-262"><a href="#cb28-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-263"><a href="#cb28-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-264"><a href="#cb28-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-265"><a href="#cb28-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-266"><a href="#cb28-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-267"><a href="#cb28-267" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-268"><a href="#cb28-268" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-269"><a href="#cb28-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-270"><a href="#cb28-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-271"><a href="#cb28-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-272"><a href="#cb28-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-273"><a href="#cb28-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-274"><a href="#cb28-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-275"><a href="#cb28-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-276"><a href="#cb28-276" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-277"><a href="#cb28-277" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-278"><a href="#cb28-278" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-279"><a href="#cb28-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-280"><a href="#cb28-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-281"><a href="#cb28-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-282"><a href="#cb28-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-283"><a href="#cb28-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-284"><a href="#cb28-284" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-285"><a href="#cb28-285" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-286"><a href="#cb28-286" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-287"><a href="#cb28-287" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-288"><a href="#cb28-288" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-289"><a href="#cb28-289" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-290"><a href="#cb28-290" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-291"><a href="#cb28-291" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-292"><a href="#cb28-292" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-293"><a href="#cb28-293" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-294"><a href="#cb28-294" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-295"><a href="#cb28-295" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-296"><a href="#cb28-296" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-297"><a href="#cb28-297" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-298"><a href="#cb28-298" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-299"><a href="#cb28-299" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-300"><a href="#cb28-300" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-301"><a href="#cb28-301" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-302"><a href="#cb28-302" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-303"><a href="#cb28-303" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-304"><a href="#cb28-304" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-305"><a href="#cb28-305" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-306"><a href="#cb28-306" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-307"><a href="#cb28-307" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-308"><a href="#cb28-308" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-309"><a href="#cb28-309" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-310"><a href="#cb28-310" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-311"><a href="#cb28-311" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-312"><a href="#cb28-312" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-313"><a href="#cb28-313" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-314"><a href="#cb28-314" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-315"><a href="#cb28-315" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-316"><a href="#cb28-316" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-317"><a href="#cb28-317" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-318"><a href="#cb28-318" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-319"><a href="#cb28-319" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-320"><a href="#cb28-320" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-321"><a href="#cb28-321" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-322"><a href="#cb28-322" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-323"><a href="#cb28-323" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-324"><a href="#cb28-324" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-325"><a href="#cb28-325" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-326"><a href="#cb28-326" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-327"><a href="#cb28-327" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-328"><a href="#cb28-328" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-329"><a href="#cb28-329" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-330"><a href="#cb28-330" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-331"><a href="#cb28-331" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-332"><a href="#cb28-332" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-333"><a href="#cb28-333" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-334"><a href="#cb28-334" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-335"><a href="#cb28-335" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-336"><a href="#cb28-336" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-337"><a href="#cb28-337" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-338"><a href="#cb28-338" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-339"><a href="#cb28-339" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-340"><a href="#cb28-340" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-341"><a href="#cb28-341" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-342"><a href="#cb28-342" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-343"><a href="#cb28-343" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-344"><a href="#cb28-344" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-345"><a href="#cb28-345" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-346"><a href="#cb28-346" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-347"><a href="#cb28-347" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-348"><a href="#cb28-348" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-349"><a href="#cb28-349" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-350"><a href="#cb28-350" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-351"><a href="#cb28-351" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-352"><a href="#cb28-352" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-353"><a href="#cb28-353" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-354"><a href="#cb28-354" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-355"><a href="#cb28-355" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-356"><a href="#cb28-356" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-357"><a href="#cb28-357" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-358"><a href="#cb28-358" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-359"><a href="#cb28-359" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-360"><a href="#cb28-360" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-361"><a href="#cb28-361" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-362"><a href="#cb28-362" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-363"><a href="#cb28-363" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-364"><a href="#cb28-364" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-365"><a href="#cb28-365" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-366"><a href="#cb28-366" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-367"><a href="#cb28-367" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-368"><a href="#cb28-368" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-369"><a href="#cb28-369" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-370"><a href="#cb28-370" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-371"><a href="#cb28-371" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-372"><a href="#cb28-372" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-373"><a href="#cb28-373" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-374"><a href="#cb28-374" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-375"><a href="#cb28-375" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-376"><a href="#cb28-376" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-377"><a href="#cb28-377" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-378"><a href="#cb28-378" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-379"><a href="#cb28-379" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-380"><a href="#cb28-380" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-381"><a href="#cb28-381" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-382"><a href="#cb28-382" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-383"><a href="#cb28-383" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-384"><a href="#cb28-384" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-385"><a href="#cb28-385" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-386"><a href="#cb28-386" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-387"><a href="#cb28-387" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-388"><a href="#cb28-388" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-389"><a href="#cb28-389" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-390"><a href="#cb28-390" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-391"><a href="#cb28-391" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-392"><a href="#cb28-392" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-393"><a href="#cb28-393" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-394"><a href="#cb28-394" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-395"><a href="#cb28-395" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-396"><a href="#cb28-396" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-397"><a href="#cb28-397" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-398"><a href="#cb28-398" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-399"><a href="#cb28-399" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-400"><a href="#cb28-400" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-401"><a href="#cb28-401" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-402"><a href="#cb28-402" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-403"><a href="#cb28-403" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-404"><a href="#cb28-404" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-405"><a href="#cb28-405" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-406"><a href="#cb28-406" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-407"><a href="#cb28-407" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-408"><a href="#cb28-408" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-409"><a href="#cb28-409" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-410"><a href="#cb28-410" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-411"><a href="#cb28-411" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-412"><a href="#cb28-412" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-413"><a href="#cb28-413" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-414"><a href="#cb28-414" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-415"><a href="#cb28-415" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-416"><a href="#cb28-416" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-417"><a href="#cb28-417" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-418"><a href="#cb28-418" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-419"><a href="#cb28-419" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-420"><a href="#cb28-420" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-421"><a href="#cb28-421" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-422"><a href="#cb28-422" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-423"><a href="#cb28-423" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-424"><a href="#cb28-424" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-425"><a href="#cb28-425" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-426"><a href="#cb28-426" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-427"><a href="#cb28-427" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-428"><a href="#cb28-428" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-429"><a href="#cb28-429" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-430"><a href="#cb28-430" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-431"><a href="#cb28-431" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-432"><a href="#cb28-432" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-433"><a href="#cb28-433" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-434"><a href="#cb28-434" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-435"><a href="#cb28-435" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-436"><a href="#cb28-436" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-437"><a href="#cb28-437" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-438"><a href="#cb28-438" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-439"><a href="#cb28-439" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-440"><a href="#cb28-440" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-441"><a href="#cb28-441" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-442"><a href="#cb28-442" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-443"><a href="#cb28-443" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-444"><a href="#cb28-444" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-445"><a href="#cb28-445" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-446"><a href="#cb28-446" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-447"><a href="#cb28-447" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-448"><a href="#cb28-448" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-449"><a href="#cb28-449" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-450"><a href="#cb28-450" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-451"><a href="#cb28-451" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-452"><a href="#cb28-452" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-453"><a href="#cb28-453" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-454"><a href="#cb28-454" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-455"><a href="#cb28-455" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-456"><a href="#cb28-456" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-457"><a href="#cb28-457" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-458"><a href="#cb28-458" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-459"><a href="#cb28-459" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-460"><a href="#cb28-460" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-461"><a href="#cb28-461" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-462"><a href="#cb28-462" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-463"><a href="#cb28-463" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-464"><a href="#cb28-464" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-465"><a href="#cb28-465" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-466"><a href="#cb28-466" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-467"><a href="#cb28-467" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-468"><a href="#cb28-468" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-469"><a href="#cb28-469" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-470"><a href="#cb28-470" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-471"><a href="#cb28-471" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-472"><a href="#cb28-472" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-473"><a href="#cb28-473" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-474"><a href="#cb28-474" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-475"><a href="#cb28-475" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-476"><a href="#cb28-476" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-477"><a href="#cb28-477" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-478"><a href="#cb28-478" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-479"><a href="#cb28-479" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-480"><a href="#cb28-480" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-481"><a href="#cb28-481" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-482"><a href="#cb28-482" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-483"><a href="#cb28-483" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-484"><a href="#cb28-484" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-485"><a href="#cb28-485" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in cor.test.default(feature1, feature2, method = method, use =</span></span>
<span id="cb28-486"><a href="#cb28-486" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; "pairwise.complete.obs"): Cannot compute exact p-value with ties</span></span>
<span id="cb28-487"><a href="#cb28-487" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Converting table into matrices...</span></span>
<span id="cb28-488"><a href="#cb28-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-489"><a href="#cb28-489" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a heatmap and store it</span></span>
<span id="cb28-490"><a href="#cb28-490" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb28-491"><a href="#cb28-491" aria-hidden="true" tabindex="-1"></a>    res<span class="sc">$</span>cor,</span>
<span id="cb28-492"><a href="#cb28-492" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Print values to cells</span></span>
<span id="cb28-493"><a href="#cb28-493" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb28-494"><a href="#cb28-494" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If the p-value is under threshold</span></span>
<span id="cb28-495"><a href="#cb28-495" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.na</span>(res<span class="sc">$</span>p_adj[i, j]) <span class="sc">&amp;</span> res<span class="sc">$</span>p_adj[i, j] <span class="sc">&lt;</span> <span class="fl">0.05</span> ){</span>
<span id="cb28-496"><a href="#cb28-496" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Print "X"</span></span>
<span id="cb28-497"><a href="#cb28-497" aria-hidden="true" tabindex="-1"></a>          <span class="fu">grid.shadowtext</span>(<span class="fu">sprintf</span>(<span class="st">"%s"</span>, <span class="st">"X"</span>), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>, <span class="at">col =</span> <span class="st">"white"</span>))</span>
<span id="cb28-498"><a href="#cb28-498" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb28-499"><a href="#cb28-499" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb28-500"><a href="#cb28-500" aria-hidden="true" tabindex="-1"></a>    <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">legend_height =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">"cm"</span>))</span>
<span id="cb28-501"><a href="#cb28-501" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-502"><a href="#cb28-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-503"><a href="#cb28-503" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p, <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">20</span>, <span class="dv">2</span>, <span class="dv">25</span>), <span class="st">"mm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/cross_association_analysis-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="principal-coordinate-analysis-pcoa-of-beta-diversity" class="level2">
<h2 class="anchored" data-anchor-id="principal-coordinate-analysis-pcoa-of-beta-diversity">Principal coordinate analysis (PCoA) of beta-diversity</h2>
<p>We can perform principle coordinate analysis with multi-dimensional scaling based on Bray-Curtis index and visualize the results with explained variances on axes.</p>
<p>In the following workflow steps, we will only use salmons at day 60 because, they might be differ from each other while salmons at day 0 are more similar to each other despite being categorized into “control” and “treatment” groups, although they did not have any treatment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Substitute original metagenomic TreeSummarizedExperiment with agglomerated</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># version</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>mae_sub[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">altExp</span>(mae_sub[[<span class="dv">2</span>]], <span class="st">"prev_genus"</span>)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Fetch only salmons at day 60</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>mae_day_60 <span class="ot">&lt;-</span> mae_sub[, <span class="fu">colData</span>(mae_sub[[<span class="dv">2</span>]])<span class="sc">$</span>trial.timepoint <span class="sc">==</span> <span class="dv">60</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run PCoA with multi-dimensional scaling and Bray-Curtis index</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>tse_pcoa <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(mae_day_60[[<span class="st">"metagenomic_amplicon"</span>]],</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">methods =</span> <span class="st">"bray"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"MDS_bray"</span>)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Display dissimilarity on a plot</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotReducedDim</span>(tse_pcoa, <span class="st">"MDS_bray"</span>,</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colour_by =</span> <span class="st">"treatment_group"</span>)</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate explained variance</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>e <span class="ot">&lt;-</span> <span class="fu">attr</span>(<span class="fu">reducedDim</span>(tse_pcoa, <span class="st">"MDS_bray"</span>), <span class="st">"eig"</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>rel_eig <span class="ot">&lt;-</span> e <span class="sc">/</span> <span class="fu">sum</span>(e[e <span class="sc">&gt;</span> <span class="dv">0</span>])</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Add explained variance for each axis</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste</span>(<span class="st">"PCoA 1 ("</span>, <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> rel_eig[[<span class="dv">1</span>]], <span class="dv">1</span>),</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"%"</span>, <span class="st">")"</span>, <span class="at">sep =</span> <span class="st">""</span>),</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">y =</span> <span class="fu">paste</span>(<span class="st">"PCoA 2 ("</span>, <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> rel_eig[[<span class="dv">2</span>]], <span class="dv">1</span>),</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"%"</span>, <span class="st">")"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/pcoa_beta_diversity1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>It seems that control and treatment groups are not dissimilar.</p>
<p>We can also apply different methods on the same data and visualize the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run NMDS on relative abundance assay with Bray-Curtis distances</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>tse_pcoa <span class="ot">&lt;-</span> <span class="fu">runNMDS</span>(tse_pcoa,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">method =</span> <span class="st">"bray"</span>,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">name =</span> <span class="st">"NMDS_bray"</span>)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; initial  value 4.518187 </span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; iter   5 value 2.583910</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; iter  10 value 2.522363</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; iter  10 value 2.520021</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; final  value 2.472995 </span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; converged</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Run MDS on clr assay with Aitchison distances</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>tse_pcoa <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(tse_pcoa,</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">assay.type =</span> <span class="st">"clr"</span>,</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">name =</span> <span class="st">"MDS_aitchison"</span>)</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Run NMDS on clr assay with Euclidean distances</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>tse_pcoa <span class="ot">&lt;-</span> <span class="fu">runNMDS</span>(tse_pcoa,</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">method =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">assay.type =</span> <span class="st">"clr"</span>,</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">name =</span> <span class="st">"NMDS_aitchison"</span>)</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; initial  value 4.254856 </span></span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; iter   5 value 2.858847</span></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; iter  10 value 2.289948</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; final  value 2.254154 </span></span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; converged</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">"MDS_bray"</span>, <span class="st">"MDS_aitchison"</span>,</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"NMDS_bray"</span>, <span class="st">"NMDS_aitchison"</span>),</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>                plotReducedDim,</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>                <span class="at">object =</span> tse_pcoa,</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a>                <span class="at">colour_by =</span> <span class="st">"treatment_group"</span>)</span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate multi-panel plot</span></span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(plots) <span class="sc">+</span></span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">"collect"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>All four methods demonstrate that features from control and treatment are not dissimilar.</p>
</section>
<section id="mofa-analysis" class="level2">
<h2 class="anchored" data-anchor-id="mofa-analysis">MOFA analysis</h2>
<p>Multi-omic factor analysis allows us to discover latent factors that underlie the biological differences by taking in consideration 2 or more omic assays. To cite the original authors, “MOFA can be viewed as a statistically rigorous generalization of (sparse) principal component analysis (PCA) to multi-omics data”.</p>
<p>We will use multi-group functionality and try to find the shared and exclusive variability between two groups of salmons at day 60 (after treatment).</p>
<ol type="1">
<li>Heavy salmons that weigh more than than median</li>
<li>Light salmons that weigh less than median</li>
</ol>
<p>Since we could not find any differentiation between salmons in control and treatment groups, we would like to know if any of metagenomic species or fatty acids drive the difference in salmon weight.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add mass categories to mae_day_60</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This number is different from the media for the entire MAE object</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co"># as this one was filtered to include **only** salmons at day 60</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>mass_median <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">as.numeric</span>(<span class="fu">colData</span>(mae_day_60)[, <span class="st">"host.gutted.mass"</span>]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_day_60)[<span class="st">"mass_category"</span>] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(<span class="fu">colData</span>(mae_day_60)[, <span class="st">"host.gutted.mass"</span>]) <span class="sc">&gt;</span> mass_median, <span class="st">"heavy"</span>, <span class="st">"light"</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the column we require</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="co"># It is a necessary step for MOFA as it will not run if there are more than one</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="co"># column</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae_day_60) <span class="ot">&lt;-</span> <span class="fu">colData</span>(mae_day_60)[ , <span class="fu">c</span>(<span class="st">"mass_category"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract only transformed metagenomic assays for MOFA+ analysis</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(mae_day_60[[<span class="dv">2</span>]]) <span class="ot">&lt;-</span> <span class="fu">assays</span>(mae_day_60[[<span class="dv">2</span>]])[<span class="fu">names</span>(<span class="fu">assays</span>(mae_day_60[[<span class="dv">2</span>]])) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"clr"</span>)]</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform MAE object to MultiAssayExperiment</span></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">create_mofa_from_MultiAssayExperiment</span>(</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>    mae_day_60,</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">"mass_category"</span>,</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">extract_metadata =</span> <span class="cn">TRUE</span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Set model's options</span></span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>model_opts <span class="ot">&lt;-</span> <span class="fu">get_default_model_options</span>(model)</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>model_opts<span class="sc">$</span>num_factors <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>train_opts <span class="ot">&lt;-</span> <span class="fu">get_default_training_options</span>(model)</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Change convergence mode to slightly improve accuracy</span></span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>train_opts<span class="sc">$</span>convergence_mode <span class="ot">&lt;-</span> <span class="st">"medium"</span></span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare MOFA model</span></span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">prepare_mofa</span>(</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> model,</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">model_options =</span> model_opts,</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">training_options =</span> train_opts</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in prepare_mofa(object = model, model_options = model_opts,</span></span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; training_options = train_opts): Some view(s) have less than 15 features, MOFA</span></span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; will have little power to to learn meaningful factors for these view(s)....</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Multi-group mode requested.</span></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; This is an advanced option, if this is the first time that you are running MOFA, we suggest that you try do some exploration first without specifying groups. Two important remarks:</span></span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  - The aim of the multi-group framework is to identify the sources of variability *within* the groups. If your aim is to find a factor that 'separates' the groups, you DO NOT want to use the multi-group framework. Please see the FAQ on the MOFA2 webpage.</span></span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  - It is important to account for the group effect before selecting highly variable features (HVFs). We suggest that either you calculate HVFs per group and then take the union, or regress out the group effect before HVF selection</span></span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Checking data options...</span></span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; No data options specified, using default...</span></span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Checking training options...</span></span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Checking model options...</span></span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Train model</span></span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">run_mofa</span>(model, <span class="at">use_basilisk =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in run_mofa(model, use_basilisk = TRUE): No output filename provided. Using /run/user/1000/RtmpQtOG8E/mofa_20240711-145049.hdf5 to store the trained model.</span></span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Connecting to the mofapy2 package using basilisk. </span></span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Set 'use_basilisk' to FALSE if you prefer to manually set the python binary using 'reticulate'.</span></span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         #########################################################</span></span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">           __  __  ____  ______                    </span><span class="al">###</span><span class="co"> </span></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">          |  \/  |/ __ \|  ____/\    _             </span><span class="al">###</span><span class="co"> </span></span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">          | \  / | |  | | |__ /  \ _| |_           </span><span class="al">###</span><span class="co"> </span></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">          | |\/| | |  | |  __/ /\ \_   _|          </span><span class="al">###</span></span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">          | |  | | |__| | | / ____ \|_|            </span><span class="al">###</span></span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">          |_|  |_|\____/|_|/_/    \_\              </span><span class="al">###</span></span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span><span class="al">###</span><span class="co">                                                   </span><span class="al">###</span><span class="co"> </span></span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         ######################################################### </span></span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        </span></span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         </span></span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; use_float32 set to True: replacing float64 arrays by float32 arrays to speed up computations...</span></span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Successfully loaded view='fatty_acids_mg' group='heavy' with N=15 samples and D=60 features...</span></span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Successfully loaded view='fatty_acids_mg' group='light' with N=15 samples and D=60 features...</span></span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Successfully loaded view='metagenomic_amplicon' group='heavy' with N=15 samples and D=6 features...</span></span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Successfully loaded view='metagenomic_amplicon' group='light' with N=15 samples and D=6 features...</span></span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model options:</span></span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Automatic Relevance Determination prior on the factors: True</span></span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Automatic Relevance Determination prior on the weights: True</span></span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Spike-and-slab prior on the factors: False</span></span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - Spike-and-slab prior on the weights: False</span></span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Likelihoods:</span></span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - View 0 (fatty_acids_mg): gaussian</span></span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; - View 1 (metagenomic_amplicon): gaussian</span></span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: some view(s) have less than 15 features, MOFA won't be able to learn meaningful factors for these view(s)...</span></span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ######################################</span></span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ## Training the model with seed 42 ##</span></span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ######################################</span></span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ELBO before training: -12479.59 </span></span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 1: time=0.00, ELBO=-3111.82, deltaELBO=9367.770 (75.06470999%), Factors=5</span></span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 2: time=0.00, Factors=5</span></span>
<span id="cb32-104"><a href="#cb32-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 3: time=0.00, Factors=5</span></span>
<span id="cb32-105"><a href="#cb32-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 4: time=0.00, Factors=5</span></span>
<span id="cb32-106"><a href="#cb32-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 5: time=0.00, Factors=5</span></span>
<span id="cb32-107"><a href="#cb32-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 6: time=0.00, ELBO=1524.15, deltaELBO=4635.972 (37.14842295%), Factors=5</span></span>
<span id="cb32-108"><a href="#cb32-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 7: time=0.00, Factors=5</span></span>
<span id="cb32-109"><a href="#cb32-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 8: time=0.00, Factors=5</span></span>
<span id="cb32-110"><a href="#cb32-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 9: time=0.00, Factors=5</span></span>
<span id="cb32-111"><a href="#cb32-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 10: time=0.00, Factors=5</span></span>
<span id="cb32-112"><a href="#cb32-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 11: time=0.00, ELBO=1763.48, deltaELBO=239.332 (1.91778902%), Factors=5</span></span>
<span id="cb32-113"><a href="#cb32-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 12: time=0.00, Factors=5</span></span>
<span id="cb32-114"><a href="#cb32-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 13: time=0.00, Factors=5</span></span>
<span id="cb32-115"><a href="#cb32-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 14: time=0.00, Factors=5</span></span>
<span id="cb32-116"><a href="#cb32-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 15: time=0.00, Factors=5</span></span>
<span id="cb32-117"><a href="#cb32-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 16: time=0.00, ELBO=1855.31, deltaELBO=91.826 (0.73580545%), Factors=5</span></span>
<span id="cb32-118"><a href="#cb32-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 17: time=0.00, Factors=5</span></span>
<span id="cb32-119"><a href="#cb32-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 18: time=0.00, Factors=5</span></span>
<span id="cb32-120"><a href="#cb32-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 19: time=0.00, Factors=5</span></span>
<span id="cb32-121"><a href="#cb32-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 20: time=0.00, Factors=5</span></span>
<span id="cb32-122"><a href="#cb32-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 21: time=0.00, ELBO=1871.36, deltaELBO=16.049 (0.12859870%), Factors=5</span></span>
<span id="cb32-123"><a href="#cb32-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 22: time=0.00, Factors=5</span></span>
<span id="cb32-124"><a href="#cb32-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 23: time=0.00, Factors=5</span></span>
<span id="cb32-125"><a href="#cb32-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 24: time=0.00, Factors=5</span></span>
<span id="cb32-126"><a href="#cb32-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 25: time=0.00, Factors=5</span></span>
<span id="cb32-127"><a href="#cb32-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 26: time=0.00, ELBO=1884.59, deltaELBO=13.238 (0.10608117%), Factors=5</span></span>
<span id="cb32-128"><a href="#cb32-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 27: time=0.00, Factors=5</span></span>
<span id="cb32-129"><a href="#cb32-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 28: time=0.00, Factors=5</span></span>
<span id="cb32-130"><a href="#cb32-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 29: time=0.00, Factors=5</span></span>
<span id="cb32-131"><a href="#cb32-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 30: time=0.00, Factors=5</span></span>
<span id="cb32-132"><a href="#cb32-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 31: time=0.00, ELBO=1896.79, deltaELBO=12.192 (0.09769451%), Factors=5</span></span>
<span id="cb32-133"><a href="#cb32-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 32: time=0.00, Factors=5</span></span>
<span id="cb32-134"><a href="#cb32-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 33: time=0.00, Factors=5</span></span>
<span id="cb32-135"><a href="#cb32-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 34: time=0.00, Factors=5</span></span>
<span id="cb32-136"><a href="#cb32-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 35: time=0.00, Factors=5</span></span>
<span id="cb32-137"><a href="#cb32-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 36: time=0.00, ELBO=1904.24, deltaELBO=7.452 (0.05971370%), Factors=5</span></span>
<span id="cb32-138"><a href="#cb32-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 37: time=0.00, Factors=5</span></span>
<span id="cb32-139"><a href="#cb32-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 38: time=0.00, Factors=5</span></span>
<span id="cb32-140"><a href="#cb32-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 39: time=0.00, Factors=5</span></span>
<span id="cb32-141"><a href="#cb32-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 40: time=0.00, Factors=5</span></span>
<span id="cb32-142"><a href="#cb32-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 41: time=0.00, ELBO=1906.98, deltaELBO=2.742 (0.02197514%), Factors=5</span></span>
<span id="cb32-143"><a href="#cb32-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 42: time=0.00, Factors=5</span></span>
<span id="cb32-144"><a href="#cb32-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 43: time=0.00, Factors=5</span></span>
<span id="cb32-145"><a href="#cb32-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 44: time=0.00, Factors=5</span></span>
<span id="cb32-146"><a href="#cb32-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 45: time=0.00, Factors=5</span></span>
<span id="cb32-147"><a href="#cb32-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 46: time=0.00, ELBO=1907.81, deltaELBO=0.827 (0.00662469%), Factors=5</span></span>
<span id="cb32-148"><a href="#cb32-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 47: time=0.00, Factors=5</span></span>
<span id="cb32-149"><a href="#cb32-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 48: time=0.00, Factors=5</span></span>
<span id="cb32-150"><a href="#cb32-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 49: time=0.00, Factors=5</span></span>
<span id="cb32-151"><a href="#cb32-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 50: time=0.00, Factors=5</span></span>
<span id="cb32-152"><a href="#cb32-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 51: time=0.00, ELBO=1908.12, deltaELBO=0.317 (0.00253830%), Factors=5</span></span>
<span id="cb32-153"><a href="#cb32-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 52: time=0.00, Factors=5</span></span>
<span id="cb32-154"><a href="#cb32-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 53: time=0.00, Factors=5</span></span>
<span id="cb32-155"><a href="#cb32-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 54: time=0.00, Factors=5</span></span>
<span id="cb32-156"><a href="#cb32-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 55: time=0.00, Factors=5</span></span>
<span id="cb32-157"><a href="#cb32-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 56: time=0.00, ELBO=1908.30, deltaELBO=0.181 (0.00145008%), Factors=5</span></span>
<span id="cb32-158"><a href="#cb32-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 57: time=0.00, Factors=5</span></span>
<span id="cb32-159"><a href="#cb32-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 58: time=0.00, Factors=5</span></span>
<span id="cb32-160"><a href="#cb32-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 59: time=0.00, Factors=5</span></span>
<span id="cb32-161"><a href="#cb32-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 60: time=0.00, Factors=5</span></span>
<span id="cb32-162"><a href="#cb32-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 61: time=0.00, ELBO=1908.44, deltaELBO=0.133 (0.00106823%), Factors=5</span></span>
<span id="cb32-163"><a href="#cb32-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 62: time=0.00, Factors=5</span></span>
<span id="cb32-164"><a href="#cb32-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 63: time=0.00, Factors=5</span></span>
<span id="cb32-165"><a href="#cb32-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 64: time=0.00, Factors=5</span></span>
<span id="cb32-166"><a href="#cb32-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 65: time=0.00, Factors=5</span></span>
<span id="cb32-167"><a href="#cb32-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 66: time=0.00, ELBO=1908.55, deltaELBO=0.107 (0.00085832%), Factors=5</span></span>
<span id="cb32-168"><a href="#cb32-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 67: time=0.00, Factors=5</span></span>
<span id="cb32-169"><a href="#cb32-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 68: time=0.00, Factors=5</span></span>
<span id="cb32-170"><a href="#cb32-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 69: time=0.00, Factors=5</span></span>
<span id="cb32-171"><a href="#cb32-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 70: time=0.00, Factors=5</span></span>
<span id="cb32-172"><a href="#cb32-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 71: time=0.00, ELBO=1908.63, deltaELBO=0.088 (0.00070809%), Factors=5</span></span>
<span id="cb32-173"><a href="#cb32-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 72: time=0.00, Factors=5</span></span>
<span id="cb32-174"><a href="#cb32-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 73: time=0.00, Factors=5</span></span>
<span id="cb32-175"><a href="#cb32-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 74: time=0.00, Factors=5</span></span>
<span id="cb32-176"><a href="#cb32-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 75: time=0.00, Factors=5</span></span>
<span id="cb32-177"><a href="#cb32-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 76: time=0.00, ELBO=1908.71, deltaELBO=0.074 (0.00059626%), Factors=5</span></span>
<span id="cb32-178"><a href="#cb32-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 77: time=0.00, Factors=5</span></span>
<span id="cb32-179"><a href="#cb32-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 78: time=0.00, Factors=5</span></span>
<span id="cb32-180"><a href="#cb32-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 79: time=0.00, Factors=5</span></span>
<span id="cb32-181"><a href="#cb32-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 80: time=0.00, Factors=5</span></span>
<span id="cb32-182"><a href="#cb32-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 81: time=0.00, ELBO=1908.77, deltaELBO=0.065 (0.00052430%), Factors=5</span></span>
<span id="cb32-183"><a href="#cb32-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 82: time=0.00, Factors=5</span></span>
<span id="cb32-184"><a href="#cb32-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 83: time=0.00, Factors=5</span></span>
<span id="cb32-185"><a href="#cb32-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 84: time=0.00, Factors=5</span></span>
<span id="cb32-186"><a href="#cb32-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 85: time=0.00, Factors=5</span></span>
<span id="cb32-187"><a href="#cb32-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 86: time=0.00, ELBO=1908.83, deltaELBO=0.056 (0.00044477%), Factors=5</span></span>
<span id="cb32-188"><a href="#cb32-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 87: time=0.00, Factors=5</span></span>
<span id="cb32-189"><a href="#cb32-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 88: time=0.00, Factors=5</span></span>
<span id="cb32-190"><a href="#cb32-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 89: time=0.00, Factors=5</span></span>
<span id="cb32-191"><a href="#cb32-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 90: time=0.00, Factors=5</span></span>
<span id="cb32-192"><a href="#cb32-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 91: time=0.00, ELBO=1908.88, deltaELBO=0.048 (0.00038622%), Factors=5</span></span>
<span id="cb32-193"><a href="#cb32-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 92: time=0.00, Factors=5</span></span>
<span id="cb32-194"><a href="#cb32-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 93: time=0.00, Factors=5</span></span>
<span id="cb32-195"><a href="#cb32-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 94: time=0.00, Factors=5</span></span>
<span id="cb32-196"><a href="#cb32-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 95: time=0.00, Factors=5</span></span>
<span id="cb32-197"><a href="#cb32-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 96: time=0.00, ELBO=1908.92, deltaELBO=0.043 (0.00034775%), Factors=5</span></span>
<span id="cb32-198"><a href="#cb32-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 97: time=0.00, Factors=5</span></span>
<span id="cb32-199"><a href="#cb32-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 98: time=0.00, Factors=5</span></span>
<span id="cb32-200"><a href="#cb32-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 99: time=0.00, Factors=5</span></span>
<span id="cb32-201"><a href="#cb32-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 100: time=0.00, Factors=5</span></span>
<span id="cb32-202"><a href="#cb32-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 101: time=0.00, ELBO=1908.96, deltaELBO=0.039 (0.00031026%), Factors=5</span></span>
<span id="cb32-203"><a href="#cb32-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 102: time=0.00, Factors=5</span></span>
<span id="cb32-204"><a href="#cb32-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 103: time=0.00, Factors=5</span></span>
<span id="cb32-205"><a href="#cb32-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 104: time=0.00, Factors=5</span></span>
<span id="cb32-206"><a href="#cb32-206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 105: time=0.00, Factors=5</span></span>
<span id="cb32-207"><a href="#cb32-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 106: time=0.00, ELBO=1908.99, deltaELBO=0.035 (0.00028109%), Factors=5</span></span>
<span id="cb32-208"><a href="#cb32-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 107: time=0.00, Factors=5</span></span>
<span id="cb32-209"><a href="#cb32-209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 108: time=0.00, Factors=5</span></span>
<span id="cb32-210"><a href="#cb32-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 109: time=0.00, Factors=5</span></span>
<span id="cb32-211"><a href="#cb32-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 110: time=0.00, Factors=5</span></span>
<span id="cb32-212"><a href="#cb32-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 111: time=0.00, ELBO=1909.02, deltaELBO=0.030 (0.00023897%), Factors=5</span></span>
<span id="cb32-213"><a href="#cb32-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 112: time=0.00, Factors=5</span></span>
<span id="cb32-214"><a href="#cb32-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 113: time=0.00, Factors=5</span></span>
<span id="cb32-215"><a href="#cb32-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 114: time=0.00, Factors=5</span></span>
<span id="cb32-216"><a href="#cb32-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 115: time=0.00, Factors=5</span></span>
<span id="cb32-217"><a href="#cb32-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 116: time=0.00, ELBO=1909.05, deltaELBO=0.028 (0.00022390%), Factors=5</span></span>
<span id="cb32-218"><a href="#cb32-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 117: time=0.00, Factors=5</span></span>
<span id="cb32-219"><a href="#cb32-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 118: time=0.00, Factors=5</span></span>
<span id="cb32-220"><a href="#cb32-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 119: time=0.00, Factors=5</span></span>
<span id="cb32-221"><a href="#cb32-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 120: time=0.00, Factors=5</span></span>
<span id="cb32-222"><a href="#cb32-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 121: time=0.00, ELBO=1909.08, deltaELBO=0.026 (0.00020470%), Factors=5</span></span>
<span id="cb32-223"><a href="#cb32-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 122: time=0.00, Factors=5</span></span>
<span id="cb32-224"><a href="#cb32-224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 123: time=0.00, Factors=5</span></span>
<span id="cb32-225"><a href="#cb32-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 124: time=0.00, Factors=5</span></span>
<span id="cb32-226"><a href="#cb32-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 125: time=0.00, Factors=5</span></span>
<span id="cb32-227"><a href="#cb32-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 126: time=0.00, ELBO=1909.10, deltaELBO=0.024 (0.00018916%), Factors=5</span></span>
<span id="cb32-228"><a href="#cb32-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 127: time=0.00, Factors=5</span></span>
<span id="cb32-229"><a href="#cb32-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 128: time=0.00, Factors=5</span></span>
<span id="cb32-230"><a href="#cb32-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 129: time=0.00, Factors=5</span></span>
<span id="cb32-231"><a href="#cb32-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 130: time=0.00, Factors=5</span></span>
<span id="cb32-232"><a href="#cb32-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 131: time=0.00, ELBO=1909.12, deltaELBO=0.020 (0.00016087%), Factors=5</span></span>
<span id="cb32-233"><a href="#cb32-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 132: time=0.00, Factors=5</span></span>
<span id="cb32-234"><a href="#cb32-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 133: time=0.00, Factors=5</span></span>
<span id="cb32-235"><a href="#cb32-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 134: time=0.00, Factors=5</span></span>
<span id="cb32-236"><a href="#cb32-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 135: time=0.00, Factors=5</span></span>
<span id="cb32-237"><a href="#cb32-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 136: time=0.00, ELBO=1909.14, deltaELBO=0.021 (0.00016470%), Factors=5</span></span>
<span id="cb32-238"><a href="#cb32-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 137: time=0.00, Factors=5</span></span>
<span id="cb32-239"><a href="#cb32-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 138: time=0.00, Factors=5</span></span>
<span id="cb32-240"><a href="#cb32-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 139: time=0.00, Factors=5</span></span>
<span id="cb32-241"><a href="#cb32-241" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 140: time=0.00, Factors=5</span></span>
<span id="cb32-242"><a href="#cb32-242" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 141: time=0.00, ELBO=1909.16, deltaELBO=0.017 (0.00013885%), Factors=5</span></span>
<span id="cb32-243"><a href="#cb32-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 142: time=0.00, Factors=5</span></span>
<span id="cb32-244"><a href="#cb32-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 143: time=0.00, Factors=5</span></span>
<span id="cb32-245"><a href="#cb32-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 144: time=0.00, Factors=5</span></span>
<span id="cb32-246"><a href="#cb32-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 145: time=0.00, Factors=5</span></span>
<span id="cb32-247"><a href="#cb32-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 146: time=0.00, ELBO=1909.18, deltaELBO=0.017 (0.00013454%), Factors=5</span></span>
<span id="cb32-248"><a href="#cb32-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 147: time=0.00, Factors=5</span></span>
<span id="cb32-249"><a href="#cb32-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 148: time=0.00, Factors=5</span></span>
<span id="cb32-250"><a href="#cb32-250" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 149: time=0.00, Factors=5</span></span>
<span id="cb32-251"><a href="#cb32-251" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 150: time=0.00, Factors=5</span></span>
<span id="cb32-252"><a href="#cb32-252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 151: time=0.00, ELBO=1909.19, deltaELBO=0.015 (0.00012288%), Factors=5</span></span>
<span id="cb32-253"><a href="#cb32-253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 152: time=0.00, Factors=5</span></span>
<span id="cb32-254"><a href="#cb32-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 153: time=0.00, Factors=5</span></span>
<span id="cb32-255"><a href="#cb32-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 154: time=0.00, Factors=5</span></span>
<span id="cb32-256"><a href="#cb32-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 155: time=0.00, Factors=5</span></span>
<span id="cb32-257"><a href="#cb32-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 156: time=0.00, ELBO=1909.21, deltaELBO=0.015 (0.00012041%), Factors=5</span></span>
<span id="cb32-258"><a href="#cb32-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 157: time=0.00, Factors=5</span></span>
<span id="cb32-259"><a href="#cb32-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 158: time=0.00, Factors=5</span></span>
<span id="cb32-260"><a href="#cb32-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 159: time=0.00, Factors=5</span></span>
<span id="cb32-261"><a href="#cb32-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 160: time=0.00, Factors=5</span></span>
<span id="cb32-262"><a href="#cb32-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 161: time=0.00, ELBO=1909.22, deltaELBO=0.013 (0.00010209%), Factors=5</span></span>
<span id="cb32-263"><a href="#cb32-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 162: time=0.00, Factors=5</span></span>
<span id="cb32-264"><a href="#cb32-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 163: time=0.00, Factors=5</span></span>
<span id="cb32-265"><a href="#cb32-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 164: time=0.00, Factors=5</span></span>
<span id="cb32-266"><a href="#cb32-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 165: time=0.00, Factors=5</span></span>
<span id="cb32-267"><a href="#cb32-267" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 166: time=0.00, ELBO=1909.23, deltaELBO=0.013 (0.00010733%), Factors=5</span></span>
<span id="cb32-268"><a href="#cb32-268" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 167: time=0.00, Factors=5</span></span>
<span id="cb32-269"><a href="#cb32-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 168: time=0.00, Factors=5</span></span>
<span id="cb32-270"><a href="#cb32-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 169: time=0.00, Factors=5</span></span>
<span id="cb32-271"><a href="#cb32-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 170: time=0.00, Factors=5</span></span>
<span id="cb32-272"><a href="#cb32-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 171: time=0.00, ELBO=1909.24, deltaELBO=0.011 (0.00008661%), Factors=5</span></span>
<span id="cb32-273"><a href="#cb32-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 172: time=0.00, Factors=5</span></span>
<span id="cb32-274"><a href="#cb32-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 173: time=0.00, Factors=5</span></span>
<span id="cb32-275"><a href="#cb32-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 174: time=0.00, Factors=5</span></span>
<span id="cb32-276"><a href="#cb32-276" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 175: time=0.00, Factors=5</span></span>
<span id="cb32-277"><a href="#cb32-277" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 176: time=0.00, ELBO=1909.25, deltaELBO=0.011 (0.00009123%), Factors=5</span></span>
<span id="cb32-278"><a href="#cb32-278" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 177: time=0.00, Factors=5</span></span>
<span id="cb32-279"><a href="#cb32-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 178: time=0.00, Factors=5</span></span>
<span id="cb32-280"><a href="#cb32-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 179: time=0.00, Factors=5</span></span>
<span id="cb32-281"><a href="#cb32-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 180: time=0.00, Factors=5</span></span>
<span id="cb32-282"><a href="#cb32-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 181: time=0.00, ELBO=1909.27, deltaELBO=0.010 (0.00008178%), Factors=5</span></span>
<span id="cb32-283"><a href="#cb32-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 182: time=0.00, Factors=5</span></span>
<span id="cb32-284"><a href="#cb32-284" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 183: time=0.00, Factors=5</span></span>
<span id="cb32-285"><a href="#cb32-285" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 184: time=0.00, Factors=5</span></span>
<span id="cb32-286"><a href="#cb32-286" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 185: time=0.00, Factors=5</span></span>
<span id="cb32-287"><a href="#cb32-287" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 186: time=0.00, ELBO=1909.28, deltaELBO=0.010 (0.00008337%), Factors=5</span></span>
<span id="cb32-288"><a href="#cb32-288" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 187: time=0.00, Factors=5</span></span>
<span id="cb32-289"><a href="#cb32-289" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 188: time=0.00, Factors=5</span></span>
<span id="cb32-290"><a href="#cb32-290" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 189: time=0.00, Factors=5</span></span>
<span id="cb32-291"><a href="#cb32-291" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 190: time=0.00, Factors=5</span></span>
<span id="cb32-292"><a href="#cb32-292" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 191: time=0.00, ELBO=1909.28, deltaELBO=0.008 (0.00006779%), Factors=5</span></span>
<span id="cb32-293"><a href="#cb32-293" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 192: time=0.00, Factors=5</span></span>
<span id="cb32-294"><a href="#cb32-294" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 193: time=0.00, Factors=5</span></span>
<span id="cb32-295"><a href="#cb32-295" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 194: time=0.00, Factors=5</span></span>
<span id="cb32-296"><a href="#cb32-296" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 195: time=0.00, Factors=5</span></span>
<span id="cb32-297"><a href="#cb32-297" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 196: time=0.00, ELBO=1909.29, deltaELBO=0.010 (0.00007724%), Factors=5</span></span>
<span id="cb32-298"><a href="#cb32-298" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 197: time=0.00, Factors=5</span></span>
<span id="cb32-299"><a href="#cb32-299" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 198: time=0.00, Factors=5</span></span>
<span id="cb32-300"><a href="#cb32-300" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 199: time=0.00, Factors=5</span></span>
<span id="cb32-301"><a href="#cb32-301" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 200: time=0.00, Factors=5</span></span>
<span id="cb32-302"><a href="#cb32-302" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 201: time=0.00, ELBO=1909.30, deltaELBO=0.008 (0.00006097%), Factors=5</span></span>
<span id="cb32-303"><a href="#cb32-303" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 202: time=0.00, Factors=5</span></span>
<span id="cb32-304"><a href="#cb32-304" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 203: time=0.00, Factors=5</span></span>
<span id="cb32-305"><a href="#cb32-305" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 204: time=0.00, Factors=5</span></span>
<span id="cb32-306"><a href="#cb32-306" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 205: time=0.00, Factors=5</span></span>
<span id="cb32-307"><a href="#cb32-307" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 206: time=0.00, ELBO=1909.31, deltaELBO=0.008 (0.00006172%), Factors=5</span></span>
<span id="cb32-308"><a href="#cb32-308" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 207: time=0.00, Factors=5</span></span>
<span id="cb32-309"><a href="#cb32-309" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 208: time=0.00, Factors=5</span></span>
<span id="cb32-310"><a href="#cb32-310" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 209: time=0.00, Factors=5</span></span>
<span id="cb32-311"><a href="#cb32-311" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 210: time=0.00, Factors=5</span></span>
<span id="cb32-312"><a href="#cb32-312" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 211: time=0.00, ELBO=1909.32, deltaELBO=0.008 (0.00006143%), Factors=5</span></span>
<span id="cb32-313"><a href="#cb32-313" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 212: time=0.00, Factors=5</span></span>
<span id="cb32-314"><a href="#cb32-314" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 213: time=0.00, Factors=5</span></span>
<span id="cb32-315"><a href="#cb32-315" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 214: time=0.00, Factors=5</span></span>
<span id="cb32-316"><a href="#cb32-316" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 215: time=0.00, Factors=5</span></span>
<span id="cb32-317"><a href="#cb32-317" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 216: time=0.00, ELBO=1909.32, deltaELBO=0.007 (0.00005439%), Factors=5</span></span>
<span id="cb32-318"><a href="#cb32-318" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 217: time=0.00, Factors=5</span></span>
<span id="cb32-319"><a href="#cb32-319" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 218: time=0.00, Factors=5</span></span>
<span id="cb32-320"><a href="#cb32-320" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 219: time=0.00, Factors=5</span></span>
<span id="cb32-321"><a href="#cb32-321" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 220: time=0.00, Factors=5</span></span>
<span id="cb32-322"><a href="#cb32-322" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 221: time=0.00, ELBO=1909.33, deltaELBO=0.007 (0.00005505%), Factors=5</span></span>
<span id="cb32-323"><a href="#cb32-323" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 222: time=0.00, Factors=5</span></span>
<span id="cb32-324"><a href="#cb32-324" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 223: time=0.00, Factors=5</span></span>
<span id="cb32-325"><a href="#cb32-325" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 224: time=0.00, Factors=5</span></span>
<span id="cb32-326"><a href="#cb32-326" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 225: time=0.00, Factors=5</span></span>
<span id="cb32-327"><a href="#cb32-327" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 226: time=0.00, ELBO=1909.34, deltaELBO=0.007 (0.00005751%), Factors=5</span></span>
<span id="cb32-328"><a href="#cb32-328" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 227: time=0.00, Factors=5</span></span>
<span id="cb32-329"><a href="#cb32-329" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 228: time=0.00, Factors=5</span></span>
<span id="cb32-330"><a href="#cb32-330" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 229: time=0.00, Factors=5</span></span>
<span id="cb32-331"><a href="#cb32-331" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 230: time=0.00, Factors=5</span></span>
<span id="cb32-332"><a href="#cb32-332" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 231: time=0.00, ELBO=1909.34, deltaELBO=0.006 (0.00004540%), Factors=5</span></span>
<span id="cb32-333"><a href="#cb32-333" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 232: time=0.00, Factors=5</span></span>
<span id="cb32-334"><a href="#cb32-334" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 233: time=0.00, Factors=5</span></span>
<span id="cb32-335"><a href="#cb32-335" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 234: time=0.00, Factors=5</span></span>
<span id="cb32-336"><a href="#cb32-336" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 235: time=0.00, Factors=5</span></span>
<span id="cb32-337"><a href="#cb32-337" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Iteration 236: time=0.00, ELBO=1909.35, deltaELBO=0.005 (0.00004361%), Factors=5</span></span>
<span id="cb32-338"><a href="#cb32-338" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-339"><a href="#cb32-339" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Converged!</span></span>
<span id="cb32-340"><a href="#cb32-340" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-341"><a href="#cb32-341" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-342"><a href="#cb32-342" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-343"><a href="#cb32-343" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #######################</span></span>
<span id="cb32-344"><a href="#cb32-344" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ## Training finished ##</span></span>
<span id="cb32-345"><a href="#cb32-345" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #######################</span></span>
<span id="cb32-346"><a href="#cb32-346" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-347"><a href="#cb32-347" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-348"><a href="#cb32-348" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Saving model in /run/user/1000/RtmpQtOG8E/mofa_20240711-145049.hdf5...</span></span>
<span id="cb32-349"><a href="#cb32-349" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 factors were found to explain no variance and they were removed for downstream analysis. You can disable this option by setting load_model(..., remove_inactive_factors = FALSE)</span></span>
<span id="cb32-350"><a href="#cb32-350" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in .quality_control(object, verbose = verbose): Factor(s) 2 are strongly correlated with the total number of expressed features for at least one of your omics. Such factors appear when there are differences in the total 'levels' between your samples, *sometimes* because of poor normalisation in the preprocessing steps.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pearson-correlation-between-factors" class="level2">
<h2 class="anchored" data-anchor-id="pearson-correlation-between-factors">Pearson correlation between factors</h2>
<p>For sanity check, we expect factors to be uncorrelated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_factor_cor</span>(</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  model,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"pearson"</span>,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cl.ratio =</span> <span class="fl">0.2</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">tl.srt =</span> <span class="dv">0</span>,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">"Pearson correlation between factors"</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">0</span>),</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/check_ft-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>Overall, we do not observe highly correlated factors which is the a good sign.</p>
</section>
<section id="visualization-of-mofa-analysis" class="level2">
<h2 class="anchored" data-anchor-id="visualization-of-mofa-analysis">Visualization of MOFA analysis</h2>
<section id="variance-explained" class="level3">
<h3 class="anchored" data-anchor-id="variance-explained">Variance explained</h3>
<p>Next, we will plot the variances explained by each factor.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x =</span> <span class="st">"group"</span>, <span class="at">y =</span> <span class="st">"view"</span>, <span class="at">factor =</span> <span class="dv">1</span>, <span class="at">legend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/var_factor1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Factor 1 captures the variances explained in metagenomic view across both mass categories.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x =</span> <span class="st">"group"</span>, <span class="at">y =</span> <span class="st">"view"</span>, <span class="at">factor =</span> <span class="dv">2</span>, <span class="at">legend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/var_factor2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>On the other hand, factor 2 captures variance explained in fatty acid view, also across both mass categories.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x =</span> <span class="st">"group"</span>, <span class="at">y =</span> <span class="st">"view"</span>, <span class="at">factor =</span> <span class="dv">3</span>, <span class="at">legend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/var_factor3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Factor 3 is similar to factor 2 but the variance explained numbers are lower.</p>
<p>We can also summarize the total variance explained in one plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="plot-factor" class="level2">
<h2 class="anchored" data-anchor-id="plot-factor">Plot factor</h2>
<p>We can now plot single factors. For demonstration purposes, we will focus on the first three factors and color the samples by treatment group (control and treatment).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_factor</span>(model,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">factors =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>),</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">color_by =</span> <span class="st">"mass_category"</span>,</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="cn">TRUE</span>,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">add_violin =</span> <span class="cn">TRUE</span>,</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">color_violin =</span> <span class="cn">TRUE</span>,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Mass category"</span>) <span class="sc">+</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_hue</span>(</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Heavy"</span>, <span class="st">"Light"</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/plot_factor-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-weights" class="level2">
<h2 class="anchored" data-anchor-id="plot-weights">Plot weights</h2>
<p>The next step is to have a deeper look inside the observations we found above. We can plot factor weight that should show us what fatty acids and bacterial genera have the highest positive or negative impact on the selected factors.</p>
<section id="plot-top-weights" class="level3">
<h3 class="anchored" data-anchor-id="plot-top-weights">Plot top weights</h3>
<p>It is possible to plot top weight per each view per each factor.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>custom_plotter <span class="ot">&lt;-</span> <span class="cf">function</span>(name, factor) {</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">plot_top_weights</span>(</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>        model,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">view =</span> name,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">factors =</span> factor,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">nfeatures =</span> <span class="dv">6</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Top weights of the "</span>, name, <span class="st">" assay for factor "</span>, factor))</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(p)</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="fu">custom_plotter</span>(<span class="st">"metagenomic_amplicon"</span>, <span class="at">factor =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/top_weight_visualization-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">custom_plotter</span>(<span class="st">"fatty_acids_mg"</span>, <span class="at">factor =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/top_weight_visualization-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">custom_plotter</span>(<span class="st">"fatty_acids_mg"</span>, <span class="at">factor =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/top_weight_visualization-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-weights-per-factor" class="level3">
<h3 class="anchored" data-anchor-id="plot-weights-per-factor">Plot weights per factor</h3>
<p>We can also plot top weights per view for a selection of factors to have a better overview of the impacts of different features on factors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_weights</span>(</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    model,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">view =</span> <span class="dv">2</span>,</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">nfeatures =</span> <span class="dv">10</span>,</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">factors =</span> <span class="dv">1</span>,</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">text_size =</span> <span class="dv">4</span>,</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top weights for the first factor in metagenomic view"</span>) <span class="sc">+</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/weight_visualization2-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_weights</span>(</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>    model,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">view =</span> <span class="dv">1</span>,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">nfeatures =</span> <span class="dv">10</span>,</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">factors =</span> <span class="dv">2</span>,</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">text_size =</span> <span class="dv">4</span>,</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 10 weights for the second factor in fatty acid view"</span>) <span class="sc">+</span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)</span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_weights</span>(</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>    model,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">view =</span> <span class="dv">1</span>,</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">nfeatures =</span> <span class="dv">10</span>,</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">factors =</span> <span class="dv">3</span>,</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">text_size =</span> <span class="dv">4</span>,</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 10 weights for the third factor in fatty acid view"</span>) <span class="sc">+</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="integratedlearner" class="level2">
<h2 class="anchored" data-anchor-id="integratedlearner">IntegratedLearner</h2>
<p>IntegratedLearner R package is build on SuperLearner package which uses Bayesian ensemble method to predict desired variables by utilizing the information across multi omic layers. The advantage of such method compared to traditional frequentist approaches is the option to account for quantifiable uncertainty.</p>
<p>IntegratedLearner allow us to test multiple machine learning algorithms. Thus, we will test two: Random Forest and Bayesian Additive Regression Trees machines (BART machines). Both methods use trees as their basis and might sound very similar since they are both ensemble machine learning algorithms.</p>
<p>Random Forest employs multiple decisions trees, where each tree performs a prediction to, for example, classify an object. In the final stage, the most “voted” class will be assigned to the object.</p>
<p>BART machines algorithms employ regression trees, where each tree makes one <em>weak</em> prediction, i.e., explains a tiny component of an outcome. BART then combines these <em>weak</em> predictions to create a stronger prediction, while accounting for uncertainty.</p>
<p>Random Forest algorithm is used for classification. We also attempt to classify salmons into “treatment” and “control” groups. Previously, we did not see any difference between the groups when using Bray-Curtis distances for metagenomic data. Thus, we are interested in understanding whether we are able to classify them into two different groups if we combine the omic data.</p>
<p>Furthermore, we apply it to investigate whether we can reliable classify salmons into “heavy” (above mass median) or “light” (below or equal to mass) salmons. We hypothesize that “heavy” salmons are distinct from “light” salmons that might be due to differences in treatment, genomics, or starting weight baseline.</p>
<p>BART machines algorithm can be used for continuous data, such as host gutted mass. We are not directly interested in predicting salmon masses but rather in investigating what species and fatty acids have the greatest impact on the mass. With BART machines, we can use variable inclusion proportion to estimate the weight each feature holds in prediction results. It allows us to indirectly list the most important features from metagenomic and fatty acid datasets. Combined with the relative importance of each layer (metagenomics or fatty acids), we can output the most promising features for the subsquent analyses.</p>
<section id="data-preparation" class="level3">
<h3 class="anchored" data-anchor-id="data-preparation">Data preparation</h3>
<p>We will be only using data at day 60 after salmons have grown and been subjected (treatment) or not (control) to the algae meal.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get colData from MAE object at day 60</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>mae_day_60_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(mae_day_60)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add treatment_group to colData(mae_day_60) that was removed for MOFA analysis</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add treatment_group (control vs. treatment) to colData of MAE at day 60</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>mae_day_60_coldata <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="fu">colData</span>(mae_sub)[<span class="st">"treatment_group"</span>], <span class="fu">colData</span>(mae_day_60), <span class="at">by =</span> <span class="dv">0</span>)</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Add host.gutted.weight to colData(mae_day_60)</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>mae_day_60_coldata <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="fu">colData</span>(mae_sub)[<span class="fu">c</span>(<span class="st">"host.gutted.mass"</span>, <span class="st">"animal"</span>)],</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>                            mae_day_60_coldata,</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by =</span> <span class="st">"animal"</span>, <span class="at">by.y =</span> <span class="st">"Row.names"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also need to prepare the data for the IntegratedLearner package. We are using MultiAssayExperiment container, so extracting the data is trivial and can be performed with basic R transformations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create feature dataframe for IntegratedLearner</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>metabolomics_df <span class="ot">&lt;-</span> mae_day_60[[<span class="st">"fatty_acids_mg"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assay</span>(<span class="st">"log10"</span>) <span class="sc">|&gt;</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>metagenomics_df <span class="ot">&lt;-</span> mae_day_60[[<span class="st">"metagenomic_amplicon"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assay</span>(<span class="st">"clr"</span>) <span class="sc">|&gt;</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace "sample" names to animal accession numbers</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(metabolomics_df) <span class="ot">&lt;-</span> <span class="fu">colData</span>(mae_day_60[[<span class="st">"fatty_acids_mg"</span>]])[, <span class="st">"animal"</span>]</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(metagenomics_df) <span class="ot">&lt;-</span> <span class="fu">colData</span>(mae_day_60[[<span class="st">"metagenomic_amplicon"</span>]])[, <span class="st">"animal"</span>]</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Join feature tables together</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>feature_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(metagenomics_df, metabolomics_df)</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check: are columns from colData the same as rownames of feature table</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(metagenomics_df) <span class="sc">==</span> <span class="fu">rownames</span>(<span class="fu">colData</span>(mae_day_60))) <span class="co"># TRUE</span></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(metabolomics_df) <span class="sc">==</span> <span class="fu">rownames</span>(<span class="fu">colData</span>(mae_day_60))) <span class="co"># TRUE</span></span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also need to retain only features with non-zero variance as these the most useful for the algorithm.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose feature table for subsequent filtering</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>feature_table_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(feature_table))</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out out near-zero-variance features</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>nzv <span class="ot">&lt;-</span> <span class="fu">nearZeroVar</span>(feature_table_t)</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>features_filtered <span class="ot">&lt;-</span> feature_table_t[, <span class="sc">-</span>nzv]</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>features_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(features_filtered))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next we need to encode mass category string into 0s and 1s.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>create_sample_metadata <span class="ot">&lt;-</span> <span class="cf">function</span>(df, Y, subjectID, factor) {</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (factor <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    Y <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(df[[Y]]))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Change all 1s to 0 and all 2s to 1</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    Y <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(Y <span class="sc">==</span> <span class="dv">1</span>, <span class="dv">0</span>, <span class="fu">ifelse</span>(Y <span class="sc">==</span> <span class="dv">2</span>, <span class="dv">1</span>, Y))</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>    sample_metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">Y =</span> Y,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">subjectID =</span> subjectID</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>    Y <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df[[Y]])</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    sample_metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">Y =</span> Y,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">subjectID =</span> subjectID</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(sample_metadata) <span class="ot">&lt;-</span> subjectID</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sample_metadata)</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">create_sample_metadata</span>(<span class="at">df =</span> mae_day_60_coldata,</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">Y =</span> <span class="st">"mass_category"</span>,</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">subjectID =</span> <span class="fu">colnames</span>(metabolomics_df),</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">factor =</span> <span class="cn">TRUE</span>)</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create feature metadata dataframe for metagenomics</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>feature_metadata_metagenomics <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">featureID =</span> <span class="fu">rownames</span>(metagenomics_df),</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">featureType =</span> <span class="st">"species"</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(feature_metadata_metagenomics) <span class="ot">&lt;-</span> feature_metadata_metagenomics<span class="sc">$</span>featureID</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Create feature metadata dataframe for metabolomics</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>feature_metadata_metabolomics <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">featureID =</span> <span class="fu">rownames</span>(metabolomics_df),</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">featureType =</span> <span class="st">"metabolites"</span></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(feature_metadata_metabolomics) <span class="ot">&lt;-</span> feature_metadata_metabolomics<span class="sc">$</span>featureID</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Create feature_metadata dataframe for IntegratedLearner</span></span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>feature_metadata <span class="ot">&lt;-</span> <span class="fu">rbind</span>(feature_metadata_metagenomics, feature_metadata_metabolomics)</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform featureID into factors</span></span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>feature_metadata<span class="sc">$</span>featureID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(feature_metadata<span class="sc">$</span>featureID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mass category distribution (0: heavy, 1: light)</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(sample_metadata<span class="sc">$</span>Y)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0  1 </span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 15</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(feature_table) <span class="sc">==</span> <span class="fu">rownames</span>(feature_metadata)) <span class="co"># TRUE</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(feature_table) <span class="sc">==</span> <span class="fu">rownames</span>(sample_metadata)) <span class="co"># TRUE</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-forest-algorithm" class="level3">
<h3 class="anchored" data-anchor-id="random-forest-algorithm">Random forest algorithm</h3>
<p>We will split data into train and validation datasets to train the data and then perform independent validation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set seed for reproduciblity</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">423</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Split dataframes into train and validation dataframes</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>feature_table_sample <span class="ot">&lt;-</span> <span class="fu">sample</span>(</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>), <span class="fu">ncol</span>(feature_table), <span class="at">replace =</span> <span class="cn">TRUE</span>, <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.67</span>, <span class="fl">0.33</span>))</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>feature_table_train <span class="ot">&lt;-</span> feature_table[, feature_table_sample]</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>feature_table_valid <span class="ot">&lt;-</span> feature_table[, <span class="sc">!</span>feature_table_sample]</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_metadata</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>sample_metadata_train <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_train), ]</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>sample_metadata_valid <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_valid), ]</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_train) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_train)) <span class="co"># TRUE</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_valid) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_valid)) <span class="co"># TRUE</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We apply Random Forest as our base model for per-layer predictions, and non-negative least squares method to make final predictions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Random forest fit</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>rf_fit <span class="ot">&lt;-</span> <span class="fu">IntegratedLearner</span>(<span class="at">feature_table =</span> feature_table_train,</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata =</span> sample_metadata_train,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_metadata =</span> feature_metadata,</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_table_valid =</span> feature_table_valid,</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata_valid =</span> sample_metadata_valid,</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">folds =</span> <span class="dv">10</span>,</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">base_learner =</span> <span class="st">"SL.randomForest"</span>,</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">meta_learner =</span> <span class="st">"SL.nnls.auc"</span>,</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  1 ...</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-33"><a href="#cb53-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb53-34"><a href="#cb53-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb53-35"><a href="#cb53-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  2 ...</span></span>
<span id="cb53-36"><a href="#cb53-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-37"><a href="#cb53-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-38"><a href="#cb53-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-39"><a href="#cb53-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-40"><a href="#cb53-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-41"><a href="#cb53-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-42"><a href="#cb53-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-43"><a href="#cb53-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-44"><a href="#cb53-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-45"><a href="#cb53-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-46"><a href="#cb53-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-47"><a href="#cb53-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-48"><a href="#cb53-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-49"><a href="#cb53-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-50"><a href="#cb53-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-51"><a href="#cb53-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-52"><a href="#cb53-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-53"><a href="#cb53-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-54"><a href="#cb53-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb53-55"><a href="#cb53-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-56"><a href="#cb53-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb53-57"><a href="#cb53-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb53-58"><a href="#cb53-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running stacked model...</span></span>
<span id="cb53-59"><a href="#cb53-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-60"><a href="#cb53-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-61"><a href="#cb53-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-62"><a href="#cb53-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-63"><a href="#cb53-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-64"><a href="#cb53-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-65"><a href="#cb53-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-66"><a href="#cb53-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-67"><a href="#cb53-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-68"><a href="#cb53-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-69"><a href="#cb53-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-70"><a href="#cb53-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-71"><a href="#cb53-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-72"><a href="#cb53-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-73"><a href="#cb53-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-74"><a href="#cb53-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-75"><a href="#cb53-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-76"><a href="#cb53-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-77"><a href="#cb53-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb53-78"><a href="#cb53-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb53-79"><a href="#cb53-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb53-80"><a href="#cb53-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.nnls.auc_All</span></span>
<span id="cb53-81"><a href="#cb53-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running concatenated model...</span></span>
<span id="cb53-82"><a href="#cb53-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-83"><a href="#cb53-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-84"><a href="#cb53-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-85"><a href="#cb53-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-86"><a href="#cb53-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-87"><a href="#cb53-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-88"><a href="#cb53-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-89"><a href="#cb53-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-90"><a href="#cb53-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-91"><a href="#cb53-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-92"><a href="#cb53-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-93"><a href="#cb53-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-94"><a href="#cb53-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-95"><a href="#cb53-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-96"><a href="#cb53-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-97"><a href="#cb53-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-98"><a href="#cb53-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-99"><a href="#cb53-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-100"><a href="#cb53-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb53-101"><a href="#cb53-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb53-102"><a href="#cb53-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb53-103"><a href="#cb53-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb53-104"><a href="#cb53-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Time for model fit : 0.015 minutes </span></span>
<span id="cb53-105"><a href="#cb53-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb53-106"><a href="#cb53-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for individual layers: SL.randomForest </span></span>
<span id="cb53-107"><a href="#cb53-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for stacked layer: SL.nnls.auc </span></span>
<span id="cb53-108"><a href="#cb53-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for concatenated layer: SL.randomForest </span></span>
<span id="cb53-109"><a href="#cb53-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb53-110"><a href="#cb53-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for training data: </span></span>
<span id="cb53-111"><a href="#cb53-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb53-112"><a href="#cb53-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb53-113"><a href="#cb53-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.731       0.259 </span></span>
<span id="cb53-114"><a href="#cb53-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-115"><a href="#cb53-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.713 </span></span>
<span id="cb53-116"><a href="#cb53-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-117"><a href="#cb53-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.722 </span></span>
<span id="cb53-118"><a href="#cb53-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-119"><a href="#cb53-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb53-120"><a href="#cb53-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for test data: </span></span>
<span id="cb53-121"><a href="#cb53-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb53-122"><a href="#cb53-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb53-123"><a href="#cb53-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.611       0.500 </span></span>
<span id="cb53-124"><a href="#cb53-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-125"><a href="#cb53-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.667 </span></span>
<span id="cb53-126"><a href="#cb53-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-127"><a href="#cb53-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.611 </span></span>
<span id="cb53-128"><a href="#cb53-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb53-129"><a href="#cb53-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb53-130"><a href="#cb53-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Weights for individual layers predictions in IntegratedLearner: </span></span>
<span id="cb53-131"><a href="#cb53-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb53-132"><a href="#cb53-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.682       0.318 </span></span>
<span id="cb53-133"><a href="#cb53-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization of AUC curves to predict heavy vs. light mass categories</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>plot.obj <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">plot.learner</span>(rf_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="treatment-group" class="level4">
<h4 class="anchored" data-anchor-id="treatment-group">Treatment group</h4>
<p>We can test whether predictions of treatment versus control can be made.</p>
<p>After encoding, control is 0, and treatment is 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>metagenomic_data<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(metagenomic_data<span class="sc">$</span>treatment_concentration <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"control"</span>, <span class="st">"treatment"</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Control: 0</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Treatment: 1</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">create_sample_metadata</span>(<span class="at">df =</span> mae_day_60_coldata,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">Y =</span> <span class="st">"treatment_group"</span>,</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">subjectID =</span> <span class="fu">colnames</span>(metabolomics_df),</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">factor =</span> <span class="cn">TRUE</span>)</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Split sample_metadata into train and validation dataframes</span></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="co"># feature_table and feature_metadata remain the same</span></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>sample_metadata_train <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_train), ]</span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>sample_metadata_valid <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_valid), ]</span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check</span></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_train) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_train)) <span class="co"># TRUE</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_valid) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_valid)) <span class="co"># TRUE</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We apply th same non-negative least square meta-learner as before.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Random forest fit for treatment vs. control</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>rf_fit <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">IntegratedLearner</span>(<span class="at">feature_table =</span> feature_table_train,</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata =</span> sample_metadata_train,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_metadata =</span> feature_metadata,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_table_valid =</span> feature_table_valid,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata_valid =</span> sample_metadata_valid,</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">folds =</span> <span class="dv">10</span>,</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">base_learner =</span> <span class="st">"SL.randomForest"</span>,</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">meta_learner =</span> <span class="st">"SL.nnls.auc"</span>,</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  1 ...</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  2 ...</span></span>
<span id="cb56-36"><a href="#cb56-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-37"><a href="#cb56-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-40"><a href="#cb56-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-41"><a href="#cb56-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-42"><a href="#cb56-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-43"><a href="#cb56-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-44"><a href="#cb56-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-45"><a href="#cb56-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-46"><a href="#cb56-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-47"><a href="#cb56-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-48"><a href="#cb56-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-49"><a href="#cb56-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-50"><a href="#cb56-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-51"><a href="#cb56-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-52"><a href="#cb56-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-53"><a href="#cb56-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-54"><a href="#cb56-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb56-55"><a href="#cb56-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-56"><a href="#cb56-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb56-57"><a href="#cb56-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb56-58"><a href="#cb56-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running stacked model...</span></span>
<span id="cb56-59"><a href="#cb56-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-60"><a href="#cb56-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-61"><a href="#cb56-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-62"><a href="#cb56-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-63"><a href="#cb56-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-64"><a href="#cb56-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-65"><a href="#cb56-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-66"><a href="#cb56-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-67"><a href="#cb56-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-68"><a href="#cb56-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-69"><a href="#cb56-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-70"><a href="#cb56-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-71"><a href="#cb56-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-72"><a href="#cb56-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-73"><a href="#cb56-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-74"><a href="#cb56-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-75"><a href="#cb56-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-76"><a href="#cb56-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-77"><a href="#cb56-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb56-78"><a href="#cb56-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb56-79"><a href="#cb56-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb56-80"><a href="#cb56-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.nnls.auc_All</span></span>
<span id="cb56-81"><a href="#cb56-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running concatenated model...</span></span>
<span id="cb56-82"><a href="#cb56-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-83"><a href="#cb56-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-84"><a href="#cb56-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-85"><a href="#cb56-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-86"><a href="#cb56-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-87"><a href="#cb56-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-88"><a href="#cb56-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-89"><a href="#cb56-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-90"><a href="#cb56-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-91"><a href="#cb56-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-92"><a href="#cb56-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-93"><a href="#cb56-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-94"><a href="#cb56-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-95"><a href="#cb56-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-96"><a href="#cb56-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-97"><a href="#cb56-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-98"><a href="#cb56-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-99"><a href="#cb56-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-100"><a href="#cb56-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb56-101"><a href="#cb56-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.randomForest_All</span></span>
<span id="cb56-102"><a href="#cb56-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb56-103"><a href="#cb56-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.randomForest_All</span></span>
<span id="cb56-104"><a href="#cb56-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Time for model fit : 0.016 minutes </span></span>
<span id="cb56-105"><a href="#cb56-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb56-106"><a href="#cb56-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for individual layers: SL.randomForest </span></span>
<span id="cb56-107"><a href="#cb56-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for stacked layer: SL.nnls.auc </span></span>
<span id="cb56-108"><a href="#cb56-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for concatenated layer: SL.randomForest </span></span>
<span id="cb56-109"><a href="#cb56-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb56-110"><a href="#cb56-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for training data: </span></span>
<span id="cb56-111"><a href="#cb56-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb56-112"><a href="#cb56-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb56-113"><a href="#cb56-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        1.00        0.38 </span></span>
<span id="cb56-114"><a href="#cb56-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-115"><a href="#cb56-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:1 </span></span>
<span id="cb56-116"><a href="#cb56-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-117"><a href="#cb56-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.991 </span></span>
<span id="cb56-118"><a href="#cb56-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-119"><a href="#cb56-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb56-120"><a href="#cb56-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for test data: </span></span>
<span id="cb56-121"><a href="#cb56-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb56-122"><a href="#cb56-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb56-123"><a href="#cb56-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       1.000       0.667 </span></span>
<span id="cb56-124"><a href="#cb56-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-125"><a href="#cb56-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:1 </span></span>
<span id="cb56-126"><a href="#cb56-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-127"><a href="#cb56-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:1 </span></span>
<span id="cb56-128"><a href="#cb56-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb56-129"><a href="#cb56-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb56-130"><a href="#cb56-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Weights for individual layers predictions in IntegratedLearner: </span></span>
<span id="cb56-131"><a href="#cb56-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb56-132"><a href="#cb56-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.839       0.161 </span></span>
<span id="cb56-133"><a href="#cb56-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization (treatment vs. control)</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>plot.obj <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">plot.learner</span>(rf_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Predictions for training data of random forest algorithm.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>rf_fit<span class="sc">$</span>yhat.train</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    metabolites species   stacked concatenated</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1        0.328   0.598 0.3540190        0.393</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2        0.260   0.376 0.2719732        0.241</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        0.351   0.312 0.3472417        0.306</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4        0.296   0.461 0.3095540        0.284</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5        0.639   0.383 0.6043199        0.603</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6        0.391   0.333 0.3786452        0.395</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7        0.327   0.187 0.3133342        0.295</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8        0.768   0.386 0.7285709        0.757</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9        0.557   0.275 0.5570000        0.545</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10       0.067   0.463 0.1513536        0.049</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11       0.286   0.630 0.3326014        0.396</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12       0.200   0.209 0.2012192        0.156</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13       0.303   0.544 0.3227971        0.285</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14       0.287   0.849 0.3653376        0.377</span></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15       0.255   0.496 0.2550000        0.240</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16       0.652   0.419 0.6332847        0.679</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17       0.542   0.605 0.5420000        0.599</span></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18       0.646   0.271 0.6460000        0.628</span></span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19       0.650   0.094 0.5724988        0.606</span></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 20       0.748   0.698 0.7431194        0.800</span></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 21       0.290   0.390 0.2980323        0.282</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="bayesian-additive-regression-trees-machine-algorithm" class="level3">
<h3 class="anchored" data-anchor-id="bayesian-additive-regression-trees-machine-algorithm">Bayesian Additive Regression Trees machine algorithm</h3>
<section id="host.gutted.mass" class="level4">
<h4 class="anchored" data-anchor-id="host.gutted.mass">host.gutted.mass</h4>
<p>For BART machine algorithm we are using the continuous response variable: <code>host.gutted.mass</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create sample_metadata dataframe for IntegratedLearner</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>sample_metadata_gutted_weight <span class="ot">&lt;-</span> <span class="fu">create_sample_metadata</span>(<span class="at">df =</span> mae_day_60_coldata,</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">Y =</span> <span class="st">"host.gutted.mass"</span>,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">subjectID =</span> <span class="fu">colnames</span>(metabolomics_df),</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">factor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add rownames to animal HoloFood IDs</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_metadata_gutted_weight) <span class="ot">&lt;-</span> sample_metadata_gutted_weight<span class="sc">$</span>subjectID</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>sample_metadata_gutted_weight_train <span class="ot">&lt;-</span> sample_metadata_gutted_weight[sample_metadata_train<span class="sc">$</span>subjectID, ]</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>sample_metadata_gutted_weight_valid <span class="ot">&lt;-</span> sample_metadata_gutted_weight[sample_metadata_valid<span class="sc">$</span>subjectID, ]</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_gutted_weight_train) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_train)) <span class="co"># TRUE</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_gutted_weight_valid) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_valid)) <span class="co"># TRUE</span></span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_metadata_gutted_weight) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(metabolomics_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># predict() function is conflicting with one of bartMachine algorithms</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="co"># so we need to unload it from namespace</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">unloadNamespace</span>(<span class="st">"MOFA2"</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>bart_fit <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">IntegratedLearner</span>(</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">feature_table =</span> feature_table_train,</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sample_metadata =</span> sample_metadata_gutted_weight_train,</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">feature_metadata =</span> feature_metadata,</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">feature_table_valid =</span> feature_table_valid,</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sample_metadata_valid =</span> sample_metadata_gutted_weight_valid,</span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">folds =</span> <span class="dv">10</span>,</span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">base_learner =</span> <span class="st">"SL.BART"</span>,</span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">meta_learner =</span> <span class="st">"SL.nnls.auc"</span>,</span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">family =</span> <span class="fu">gaussian</span>())</span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  1 ... </span></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  2 ... </span></span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-37"><a href="#cb60-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-38"><a href="#cb60-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-39"><a href="#cb60-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running stacked model...</span></span>
<span id="cb60-40"><a href="#cb60-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running concatenated model...</span></span>
<span id="cb60-41"><a href="#cb60-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-42"><a href="#cb60-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-43"><a href="#cb60-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-44"><a href="#cb60-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-45"><a href="#cb60-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-46"><a href="#cb60-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-47"><a href="#cb60-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-48"><a href="#cb60-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-49"><a href="#cb60-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-50"><a href="#cb60-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-51"><a href="#cb60-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb60-52"><a href="#cb60-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Time for model fit : 0.391 minutes </span></span>
<span id="cb60-53"><a href="#cb60-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb60-54"><a href="#cb60-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for individual layers: SL.BART </span></span>
<span id="cb60-55"><a href="#cb60-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for stacked layer: SL.nnls.auc </span></span>
<span id="cb60-56"><a href="#cb60-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for concatenated layer: SL.BART </span></span>
<span id="cb60-57"><a href="#cb60-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb60-58"><a href="#cb60-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R^2 for training data: </span></span>
<span id="cb60-59"><a href="#cb60-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb60-60"><a href="#cb60-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb60-61"><a href="#cb60-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.01616814  0.01426359 </span></span>
<span id="cb60-62"><a href="#cb60-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-63"><a href="#cb60-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.002307995 </span></span>
<span id="cb60-64"><a href="#cb60-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-65"><a href="#cb60-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.02072589 </span></span>
<span id="cb60-66"><a href="#cb60-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-67"><a href="#cb60-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb60-68"><a href="#cb60-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R^2 for test data: </span></span>
<span id="cb60-69"><a href="#cb60-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb60-70"><a href="#cb60-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb60-71"><a href="#cb60-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   0.5776534   0.2193864 </span></span>
<span id="cb60-72"><a href="#cb60-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-73"><a href="#cb60-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.2908147 </span></span>
<span id="cb60-74"><a href="#cb60-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-75"><a href="#cb60-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.6712902 </span></span>
<span id="cb60-76"><a href="#cb60-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb60-77"><a href="#cb60-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb60-78"><a href="#cb60-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Weights for individual layers predictions in IntegratedLearner: </span></span>
<span id="cb60-79"><a href="#cb60-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb60-80"><a href="#cb60-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.155       0.845 </span></span>
<span id="cb60-81"><a href="#cb60-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb60-82"><a href="#cb60-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb60-83"><a href="#cb60-83" aria-hidden="true" tabindex="-1"></a>plot.obj <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">plot.learner</span>(bart_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Since we are employing Bayestin statistics we can generate credible intervals</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate credible intervals</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Requires longitudinal data for extracting very important features</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>bart_weights <span class="ot">&lt;-</span> bart_fit<span class="sc">$</span>weights</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>bart_dataX <span class="ot">&lt;-</span> bart_fit<span class="sc">$</span>X_train_layers</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>bart_dataY <span class="ot">&lt;-</span> bart_fit<span class="sc">$</span>Y_train</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>bart_post.samples <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(bart_weights))</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bart_post.samples) <span class="ot">&lt;-</span> <span class="fu">names</span>(bart_dataX)</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Is not serialized when using binomial family</span></span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(bart_post.samples)) {</span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>  bart_post.samples[[i]] <span class="ot">&lt;-</span> <span class="fu">bart_machine_get_posterior</span>(bart_fit<span class="sc">$</span>model_fits<span class="sc">$</span>model_layers[[i]],</span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>                                                       bart_dataX[[i]])<span class="sc">$</span>y_hat_posterior_samples</span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a>bart_weighted.post.samples <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="st">"+"</span>, <span class="fu">Map</span>(<span class="st">"*"</span>, bart_post.samples, bart_weights))</span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bart_weighted.post.samples) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(bart_dataX[[<span class="dv">1</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show credible intervals for hosted.gut.weight</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">caterplot</span>(<span class="fu">t</span>(bart_weighted.post.samples),</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">horizontal =</span> <span class="cn">FALSE</span>, <span class="at">add =</span> <span class="cn">FALSE</span>)</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(bart_dataY[temp])</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span><span class="st">""</span>, <span class="at">xlab =</span> <span class="st">"Observations"</span>, <span class="at">ylab =</span> <span class="st">"Host gutted mass (in gramms)"</span>,</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">line =</span> <span class="cn">NA</span>, <span class="at">outer =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>Finally, as stated in the introduction to IntegratedLearner section, we are able to output the relative importance of each layer and the most important features in each layer.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Find most important features </span><span class="al">###</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>omicsEye_theme <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="co"># set default text format based on categorical and length</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  angle <span class="ot">=</span> <span class="dv">45</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  hjust <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>  size <span class="ot">=</span> <span class="dv">6</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> hjust, <span class="at">angle =</span> angle),</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">face =</span> <span class="st">'bold'</span>),</span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">'black'</span>, <span class="at">size =</span> .<span class="dv">25</span>),</span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">'black'</span>, <span class="at">size =</span> .<span class="dv">25</span>),</span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line.x =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">'black'</span>, <span class="at">size =</span> .<span class="dv">25</span>),</span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line.y =</span> ggplot2<span class="sc">::</span><span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">'black'</span>, <span class="at">size =</span> .<span class="dv">25</span>),</span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> ggplot2<span class="sc">::</span><span class="fu">element_blank</span>(),</span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> ggplot2<span class="sc">::</span><span class="fu">element_blank</span>(),</span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> ggplot2<span class="sc">::</span><span class="fu">element_blank</span>())</span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>myColtmp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cornflowerblue"</span>,<span class="st">"darkcyan"</span>,<span class="st">"orchid4"</span>,</span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>            <span class="st">"brown"</span>,<span class="st">"goldenrod4"</span>,<span class="st">"mistyrose4"</span>,<span class="st">"darkgreen"</span>,<span class="st">"purple"</span>)</span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-28"><a href="#cb63-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-29"><a href="#cb63-29" aria-hidden="true" tabindex="-1"></a>VIMP_stack <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(bart_fit<span class="sc">$</span>weights)</span>
<span id="cb63-30"><a href="#cb63-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-31"><a href="#cb63-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(VIMP_stack) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>)</span>
<span id="cb63-32"><a href="#cb63-32" aria-hidden="true" tabindex="-1"></a>VIMP_stack<span class="sc">$</span>sd <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb63-33"><a href="#cb63-33" aria-hidden="true" tabindex="-1"></a>VIMP_stack<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"stack"</span></span>
<span id="cb63-34"><a href="#cb63-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-35"><a href="#cb63-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-36"><a href="#cb63-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Microbiome</span></span>
<span id="cb63-37"><a href="#cb63-37" aria-hidden="true" tabindex="-1"></a>qq <span class="ot">&lt;-</span> bartMachine<span class="sc">::</span><span class="fu">investigate_var_importance</span>(bart_fit<span class="sc">$</span>model_fits<span class="sc">$</span>model_layers<span class="sc">$</span>species, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb63-38"><a href="#cb63-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; .....</span></span>
<span id="cb63-39"><a href="#cb63-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-40"><a href="#cb63-40" aria-hidden="true" tabindex="-1"></a>VIMP_microbiome <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(qq<span class="sc">$</span>avg_var_props, qq<span class="sc">$</span>sd_var_props)</span>
<span id="cb63-41"><a href="#cb63-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(VIMP_microbiome) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>, <span class="st">"sd"</span>)</span>
<span id="cb63-42"><a href="#cb63-42" aria-hidden="true" tabindex="-1"></a>VIMP_microbiome<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"species"</span></span>
<span id="cb63-43"><a href="#cb63-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-44"><a href="#cb63-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Metabolomics</span></span>
<span id="cb63-45"><a href="#cb63-45" aria-hidden="true" tabindex="-1"></a>qq <span class="ot">&lt;-</span> bartMachine<span class="sc">::</span><span class="fu">investigate_var_importance</span>(bart_fit<span class="sc">$</span>model_fits<span class="sc">$</span>model_layers<span class="sc">$</span>metabolites, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb63-46"><a href="#cb63-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; .....</span></span>
<span id="cb63-47"><a href="#cb63-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-48"><a href="#cb63-48" aria-hidden="true" tabindex="-1"></a>VIMP_metabolites <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(qq<span class="sc">$</span>avg_var_props, qq<span class="sc">$</span>sd_var_props)</span>
<span id="cb63-49"><a href="#cb63-49" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(VIMP_metabolites )<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>, <span class="st">"sd"</span>)</span>
<span id="cb63-50"><a href="#cb63-50" aria-hidden="true" tabindex="-1"></a>VIMP_metabolites<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"metabolites"</span></span>
<span id="cb63-51"><a href="#cb63-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-52"><a href="#cb63-52" aria-hidden="true" tabindex="-1"></a>VIMP <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rbind.data.frame</span>(VIMP_stack,</span>
<span id="cb63-53"><a href="#cb63-53" aria-hidden="true" tabindex="-1"></a>                                     VIMP_microbiome[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,],</span>
<span id="cb63-54"><a href="#cb63-54" aria-hidden="true" tabindex="-1"></a>                                     VIMP_metabolites[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,]))</span>
<span id="cb63-55"><a href="#cb63-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-56"><a href="#cb63-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-57"><a href="#cb63-57" aria-hidden="true" tabindex="-1"></a>VIMP <span class="ot">&lt;-</span> <span class="fu">rownames_to_column</span>(VIMP, <span class="st">"ID"</span>)</span>
<span id="cb63-58"><a href="#cb63-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-59"><a href="#cb63-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-60"><a href="#cb63-60" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> VIMP <span class="sc">%&gt;%</span></span>
<span id="cb63-61"><a href="#cb63-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">"stack"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-62"><a href="#cb63-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mean))  <span class="sc">%&gt;%</span></span>
<span id="cb63-63"><a href="#cb63-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> mean, <span class="at">x =</span> <span class="fu">reorder</span>(ID,<span class="sc">-</span>mean))) <span class="sc">+</span></span>
<span id="cb63-64"><a href="#cb63-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">'darkseagreen'</span>) <span class="sc">+</span></span>
<span id="cb63-65"><a href="#cb63-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb63-66"><a href="#cb63-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">omicsEye_theme</span>() <span class="sc">+</span></span>
<span id="cb63-67"><a href="#cb63-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Layer Weights"</span>) <span class="sc">+</span></span>
<span id="cb63-68"><a href="#cb63-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)</span>
<span id="cb63-69"><a href="#cb63-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span id="cb63-70"><a href="#cb63-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Please use the `linewidth` argument instead.</span></span>
<span id="cb63-71"><a href="#cb63-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-72"><a href="#cb63-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-73"><a href="#cb63-73" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> VIMP <span class="sc">%&gt;%</span></span>
<span id="cb63-74"><a href="#cb63-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'species'</span>, <span class="st">'metabolites'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb63-75"><a href="#cb63-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(mean) <span class="sc">%&gt;%</span></span>
<span id="cb63-76"><a href="#cb63-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">str_replace_all</span>(ID, <span class="fu">fixed</span>(<span class="st">"_"</span>), <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb63-77"><a href="#cb63-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">factor</span>(type,</span>
<span id="cb63-78"><a href="#cb63-78" aria-hidden="true" tabindex="-1"></a>                       <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'species'</span>, <span class="st">'metabolites'</span>),</span>
<span id="cb63-79"><a href="#cb63-79" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'species'</span>, <span class="st">'metabolites'</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb63-80"><a href="#cb63-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">reorder</span>(ID, <span class="sc">-</span>mean), mean, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb63-81"><a href="#cb63-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span> type, <span class="at">scale =</span> <span class="st">'free'</span>) <span class="sc">+</span></span>
<span id="cb63-82"><a href="#cb63-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"lightsalmon"</span>) <span class="sc">+</span></span>
<span id="cb63-83"><a href="#cb63-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span><span class="fu">ifelse</span>(mean<span class="sc">-</span>sd<span class="sc">&gt;</span><span class="dv">0</span>,mean<span class="sc">-</span>sd,<span class="dv">0</span>), <span class="at">ymax=</span>mean<span class="sc">+</span>sd), <span class="at">width=</span>.<span class="dv">2</span>, <span class="at">position=</span><span class="fu">position_dodge</span>(.<span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb63-84"><a href="#cb63-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb63-85"><a href="#cb63-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb63-86"><a href="#cb63-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">omicsEye_theme</span>() <span class="sc">+</span></span>
<span id="cb63-87"><a href="#cb63-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span> (<span class="at">strip.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb63-88"><a href="#cb63-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Inclusion proportion'</span>) <span class="sc">+</span></span>
<span id="cb63-89"><a href="#cb63-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>)</span>
<span id="cb63-90"><a href="#cb63-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-91"><a href="#cb63-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-92"><a href="#cb63-92" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p4,</span>
<span id="cb63-93"><a href="#cb63-93" aria-hidden="true" tabindex="-1"></a>             <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb63-94"><a href="#cb63-94" aria-hidden="true" tabindex="-1"></a>             <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'Estimated IntegratedLearner layer weights'</span>),</span>
<span id="cb63-95"><a href="#cb63-95" aria-hidden="true" tabindex="-1"></a>             <span class="at">label_size =</span> <span class="dv">8</span>, <span class="at">vjust =</span> <span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb63-96"><a href="#cb63-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>,<span class="fl">0.5</span>,<span class="fl">0.5</span>), <span class="st">"cm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p5,</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Top features of microbiome and metabolites layers"</span>),</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">label_size =</span> <span class="dv">8</span>, <span class="at">vjust =</span> <span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>,<span class="fl">0.5</span>,<span class="fl">0.5</span>), <span class="st">"cm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="heavy-vs.-light-classification" class="level3">
<h3 class="anchored" data-anchor-id="heavy-vs.-light-classification">Heavy vs.&nbsp;light classification</h3>
<p>With BART machines, we can also perform classification.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># predict() function is conflicting with one of bartMachine algorithms</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="co"># so we need to unload it from namespace</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="fu">unloadNamespace</span>(<span class="st">"MOFA2"</span>)</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">create_sample_metadata</span>(<span class="at">df =</span> mae_day_60_coldata,</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">Y =</span> <span class="st">"mass_category"</span>,</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">subjectID =</span> <span class="fu">colnames</span>(metabolomics_df),</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">factor =</span> <span class="cn">TRUE</span>)</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_metadata</span></span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>sample_metadata_train <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_train), ]</span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>sample_metadata_valid <span class="ot">&lt;-</span> sample_metadata[<span class="fu">colnames</span>(feature_table_valid), ]</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check</span></span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_train) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_train)) <span class="co"># TRUE</span></span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(sample_metadata_valid) <span class="sc">==</span> <span class="fu">colnames</span>(feature_table_valid)) <span class="co"># TRUE</span></span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb65-18"><a href="#cb65-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-19"><a href="#cb65-19" aria-hidden="true" tabindex="-1"></a>bart_fit <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">IntegratedLearner</span>(<span class="at">feature_table =</span> feature_table_train,</span>
<span id="cb65-20"><a href="#cb65-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata =</span> sample_metadata_train,</span>
<span id="cb65-21"><a href="#cb65-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_metadata =</span> feature_metadata,</span>
<span id="cb65-22"><a href="#cb65-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">feature_table_valid =</span> feature_table_valid,</span>
<span id="cb65-23"><a href="#cb65-23" aria-hidden="true" tabindex="-1"></a>                               <span class="at">sample_metadata_valid =</span> sample_metadata_valid,</span>
<span id="cb65-24"><a href="#cb65-24" aria-hidden="true" tabindex="-1"></a>                               <span class="at">folds =</span> <span class="dv">10</span>,</span>
<span id="cb65-25"><a href="#cb65-25" aria-hidden="true" tabindex="-1"></a>                               <span class="at">base_learner =</span> <span class="st">"SL.BART"</span>,</span>
<span id="cb65-26"><a href="#cb65-26" aria-hidden="true" tabindex="-1"></a>                               <span class="at">meta_learner =</span> <span class="st">"SL.nnls.auc"</span>,</span>
<span id="cb65-27"><a href="#cb65-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb65-28"><a href="#cb65-28" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb65-29"><a href="#cb65-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  1 ...</span></span>
<span id="cb65-30"><a href="#cb65-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-31"><a href="#cb65-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-32"><a href="#cb65-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-33"><a href="#cb65-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-34"><a href="#cb65-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-35"><a href="#cb65-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-36"><a href="#cb65-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-37"><a href="#cb65-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-38"><a href="#cb65-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-39"><a href="#cb65-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-40"><a href="#cb65-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-41"><a href="#cb65-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-42"><a href="#cb65-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-43"><a href="#cb65-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-44"><a href="#cb65-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-45"><a href="#cb65-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-46"><a href="#cb65-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-47"><a href="#cb65-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-48"><a href="#cb65-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-49"><a href="#cb65-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-50"><a href="#cb65-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-51"><a href="#cb65-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-52"><a href="#cb65-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-53"><a href="#cb65-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-54"><a href="#cb65-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-55"><a href="#cb65-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-56"><a href="#cb65-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-57"><a href="#cb65-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-58"><a href="#cb65-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-59"><a href="#cb65-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-60"><a href="#cb65-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-61"><a href="#cb65-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-62"><a href="#cb65-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-63"><a href="#cb65-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-64"><a href="#cb65-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-65"><a href="#cb65-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-66"><a href="#cb65-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 60</span></span>
<span id="cb65-67"><a href="#cb65-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-68"><a href="#cb65-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-69"><a href="#cb65-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-70"><a href="#cb65-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb65-71"><a href="#cb65-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-72"><a href="#cb65-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-73"><a href="#cb65-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.BART_All</span></span>
<span id="cb65-74"><a href="#cb65-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-75"><a href="#cb65-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running base model for layer  2 ...</span></span>
<span id="cb65-76"><a href="#cb65-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-77"><a href="#cb65-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-78"><a href="#cb65-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-79"><a href="#cb65-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-80"><a href="#cb65-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-81"><a href="#cb65-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-82"><a href="#cb65-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-83"><a href="#cb65-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-84"><a href="#cb65-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-85"><a href="#cb65-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-86"><a href="#cb65-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-87"><a href="#cb65-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-88"><a href="#cb65-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-89"><a href="#cb65-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-90"><a href="#cb65-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-91"><a href="#cb65-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-92"><a href="#cb65-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-93"><a href="#cb65-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-94"><a href="#cb65-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-95"><a href="#cb65-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-96"><a href="#cb65-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-97"><a href="#cb65-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-98"><a href="#cb65-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-99"><a href="#cb65-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-100"><a href="#cb65-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-101"><a href="#cb65-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-102"><a href="#cb65-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-103"><a href="#cb65-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-104"><a href="#cb65-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-105"><a href="#cb65-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-106"><a href="#cb65-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-107"><a href="#cb65-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-108"><a href="#cb65-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-109"><a href="#cb65-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-110"><a href="#cb65-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-111"><a href="#cb65-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-112"><a href="#cb65-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 6</span></span>
<span id="cb65-113"><a href="#cb65-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-114"><a href="#cb65-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-115"><a href="#cb65-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-116"><a href="#cb65-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb65-117"><a href="#cb65-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-118"><a href="#cb65-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-119"><a href="#cb65-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.BART_All</span></span>
<span id="cb65-120"><a href="#cb65-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-121"><a href="#cb65-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running stacked model...</span></span>
<span id="cb65-122"><a href="#cb65-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-123"><a href="#cb65-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-124"><a href="#cb65-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-125"><a href="#cb65-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-126"><a href="#cb65-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-127"><a href="#cb65-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-128"><a href="#cb65-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-129"><a href="#cb65-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-130"><a href="#cb65-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-131"><a href="#cb65-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-132"><a href="#cb65-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-133"><a href="#cb65-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-134"><a href="#cb65-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-135"><a href="#cb65-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-136"><a href="#cb65-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-137"><a href="#cb65-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-138"><a href="#cb65-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-139"><a href="#cb65-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-140"><a href="#cb65-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 2</span></span>
<span id="cb65-141"><a href="#cb65-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.nnls.auc_All</span></span>
<span id="cb65-142"><a href="#cb65-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb65-143"><a href="#cb65-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.nnls.auc_All</span></span>
<span id="cb65-144"><a href="#cb65-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running concatenated model...</span></span>
<span id="cb65-145"><a href="#cb65-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-146"><a href="#cb65-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-147"><a href="#cb65-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-148"><a href="#cb65-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-149"><a href="#cb65-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-150"><a href="#cb65-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-151"><a href="#cb65-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-152"><a href="#cb65-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-153"><a href="#cb65-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-154"><a href="#cb65-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-155"><a href="#cb65-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-156"><a href="#cb65-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-157"><a href="#cb65-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-158"><a href="#cb65-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-159"><a href="#cb65-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-160"><a href="#cb65-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-161"><a href="#cb65-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-162"><a href="#cb65-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-163"><a href="#cb65-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-164"><a href="#cb65-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-165"><a href="#cb65-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-166"><a href="#cb65-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-167"><a href="#cb65-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-168"><a href="#cb65-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-169"><a href="#cb65-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-170"><a href="#cb65-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-171"><a href="#cb65-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-172"><a href="#cb65-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-173"><a href="#cb65-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-174"><a href="#cb65-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-175"><a href="#cb65-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-176"><a href="#cb65-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-177"><a href="#cb65-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-178"><a href="#cb65-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-179"><a href="#cb65-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-180"><a href="#cb65-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-181"><a href="#cb65-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of covariates in All is: 66</span></span>
<span id="cb65-182"><a href="#cb65-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-183"><a href="#cb65-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-184"><a href="#cb65-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CV SL.BART_All</span></span>
<span id="cb65-185"><a href="#cb65-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Non-Negative least squares convergence: TRUE</span></span>
<span id="cb65-186"><a href="#cb65-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; serializing in order to be saved for future R sessions...done</span></span>
<span id="cb65-187"><a href="#cb65-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-188"><a href="#cb65-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; full SL.BART_All</span></span>
<span id="cb65-189"><a href="#cb65-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; predicting probabilities where "1" is considered the target level...</span></span>
<span id="cb65-190"><a href="#cb65-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Time for model fit : 0.39 minutes </span></span>
<span id="cb65-191"><a href="#cb65-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb65-192"><a href="#cb65-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for individual layers: SL.BART </span></span>
<span id="cb65-193"><a href="#cb65-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for stacked layer: SL.nnls.auc </span></span>
<span id="cb65-194"><a href="#cb65-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Model fit for concatenated layer: SL.BART </span></span>
<span id="cb65-195"><a href="#cb65-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb65-196"><a href="#cb65-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for training data: </span></span>
<span id="cb65-197"><a href="#cb65-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb65-198"><a href="#cb65-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb65-199"><a href="#cb65-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.787       0.139 </span></span>
<span id="cb65-200"><a href="#cb65-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-201"><a href="#cb65-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.796 </span></span>
<span id="cb65-202"><a href="#cb65-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-203"><a href="#cb65-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.787 </span></span>
<span id="cb65-204"><a href="#cb65-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-205"><a href="#cb65-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb65-206"><a href="#cb65-206" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AUC metric for test data: </span></span>
<span id="cb65-207"><a href="#cb65-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Individual layers: </span></span>
<span id="cb65-208"><a href="#cb65-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb65-209"><a href="#cb65-209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       0.667       0.667 </span></span>
<span id="cb65-210"><a href="#cb65-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-211"><a href="#cb65-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Stacked model:0.667 </span></span>
<span id="cb65-212"><a href="#cb65-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-213"><a href="#cb65-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Concatenated model:0.667 </span></span>
<span id="cb65-214"><a href="#cb65-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ======================</span></span>
<span id="cb65-215"><a href="#cb65-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb65-216"><a href="#cb65-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Weights for individual layers predictions in IntegratedLearner: </span></span>
<span id="cb65-217"><a href="#cb65-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metabolites     species </span></span>
<span id="cb65-218"><a href="#cb65-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           1           0 </span></span>
<span id="cb65-219"><a href="#cb65-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ========================================</span></span>
<span id="cb65-220"><a href="#cb65-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-221"><a href="#cb65-221" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb65-222"><a href="#cb65-222" aria-hidden="true" tabindex="-1"></a>plot.obj <span class="ot">&lt;-</span> IntegratedLearner<span class="sc">:::</span><span class="fu">plot.learner</span>(bart_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="case_study_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="conclusions" class="level2">
<h2 class="anchored" data-anchor-id="conclusions">Conclusions</h2>
<p>The present case study has demonstrated how easy and fast it is to download large dataset and transform the data into a MultiAssayExperiment, which in turn gives the researchers access to an extensive plethora of downstream tools, such mia and MOFA2 that can be used to pre-process and visualize the multi-omics data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running under: Fedora Linux 40.20240708.0 (Kinoite)</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; BLAS/LAPACK: /nix/store/01wg6k72kwqxa3mvvfak9kdcwciihx72-blas-3/lib/libblas.so.3;  LAPACK version 3.12.0</span></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                 </span></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8           LC_COLLATE=en_US.UTF-8       </span></span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8       LC_MESSAGES=en_US.UTF-8      </span></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8          LC_NAME=fi_FI.UTF-8          </span></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] LC_ADDRESS=fi_FI.UTF-8        LC_TELEPHONE=fi_FI.UTF-8     </span></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8    LC_IDENTIFICATION=en_US.UTF-8</span></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; time zone: Europe/Helsinki</span></span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] grid      splines   stats4    stats     graphics  grDevices utils    </span></span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [8] datasets  methods   base     </span></span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] MOFA2_1.14.0                    ggsignif_0.6.4                 </span></span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] scater_1.32.0                   scuttle_1.14.0                 </span></span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] caret_6.0-94                    lattice_0.22-6                 </span></span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] cowplot_1.1.3                   ComplexHeatmap_2.20.0          </span></span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] shadowtext_0.1.3                lubridate_1.9.3                </span></span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] forcats_1.0.0                   stringr_1.5.1                  </span></span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] purrr_1.0.2                     readr_2.1.5                    </span></span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] tidyr_1.3.1                     tibble_3.2.1                   </span></span>
<span id="cb66-33"><a href="#cb66-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] tidyverse_2.0.0                 mcmcplots_0.4.3                </span></span>
<span id="cb66-34"><a href="#cb66-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] coda_0.19-4.1                   bartMachine_1.3.4.1            </span></span>
<span id="cb66-35"><a href="#cb66-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] missForest_1.5                  randomForest_4.7-1.1           </span></span>
<span id="cb66-36"><a href="#cb66-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [23] bartMachineJARs_1.2.1           rJava_1.0-11                   </span></span>
<span id="cb66-37"><a href="#cb66-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] SuperLearner_2.0-29             gam_1.22-3                     </span></span>
<span id="cb66-38"><a href="#cb66-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [27] foreach_1.5.2                   nnls_1.5                       </span></span>
<span id="cb66-39"><a href="#cb66-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [29] IntegratedLearner_0.0.1         MGnifyR_1.0.0                  </span></span>
<span id="cb66-40"><a href="#cb66-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] reticulate_1.38.0               patchwork_1.2.0                </span></span>
<span id="cb66-41"><a href="#cb66-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] miaViz_1.12.0                   ggraph_2.2.1                   </span></span>
<span id="cb66-42"><a href="#cb66-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [35] dplyr_1.1.4                     mia_1.13.23                    </span></span>
<span id="cb66-43"><a href="#cb66-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [37] MultiAssayExperiment_1.30.2     TreeSummarizedExperiment_2.12.0</span></span>
<span id="cb66-44"><a href="#cb66-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [39] Biostrings_2.72.1               XVector_0.44.0                 </span></span>
<span id="cb66-45"><a href="#cb66-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] SingleCellExperiment_1.26.0     SummarizedExperiment_1.34.0    </span></span>
<span id="cb66-46"><a href="#cb66-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [43] Biobase_2.64.0                  GenomicRanges_1.56.1           </span></span>
<span id="cb66-47"><a href="#cb66-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [45] GenomeInfoDb_1.40.1             IRanges_2.38.0                 </span></span>
<span id="cb66-48"><a href="#cb66-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [47] S4Vectors_0.42.0                BiocGenerics_0.50.0            </span></span>
<span id="cb66-49"><a href="#cb66-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [49] MatrixGenerics_1.16.0           matrixStats_1.3.0              </span></span>
<span id="cb66-50"><a href="#cb66-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [51] ggplot2_3.5.1                   BiocManager_1.30.23            </span></span>
<span id="cb66-51"><a href="#cb66-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [53] knitr_1.47                     </span></span>
<span id="cb66-52"><a href="#cb66-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb66-53"><a href="#cb66-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb66-54"><a href="#cb66-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] fs_1.6.4                    tidyjson_0.3.2             </span></span>
<span id="cb66-55"><a href="#cb66-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [3] DirichletMultinomial_1.46.0 doParallel_1.0.17          </span></span>
<span id="cb66-56"><a href="#cb66-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [5] httr_1.4.7                  RColorBrewer_1.1-3         </span></span>
<span id="cb66-57"><a href="#cb66-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [7] tools_4.4.1                 doRNG_1.8.6                </span></span>
<span id="cb66-58"><a href="#cb66-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [9] backports_1.5.0             utf8_1.2.4                 </span></span>
<span id="cb66-59"><a href="#cb66-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [11] R6_2.5.1                    vegan_2.6-6.1              </span></span>
<span id="cb66-60"><a href="#cb66-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [13] HDF5Array_1.32.0            lazyeval_0.2.2             </span></span>
<span id="cb66-61"><a href="#cb66-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [15] uwot_0.2.2                  mgcv_1.9-1                 </span></span>
<span id="cb66-62"><a href="#cb66-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [17] GetoptLong_1.0.5            rhdf5filters_1.16.0        </span></span>
<span id="cb66-63"><a href="#cb66-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [19] permute_0.9-7               withr_3.0.0                </span></span>
<span id="cb66-64"><a href="#cb66-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [21] gridExtra_2.3               cli_3.6.3                  </span></span>
<span id="cb66-65"><a href="#cb66-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [23] Cairo_1.6-2                 denstrip_1.5.4             </span></span>
<span id="cb66-66"><a href="#cb66-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [25] sandwich_3.1-0              labeling_0.4.3             </span></span>
<span id="cb66-67"><a href="#cb66-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [27] slam_0.1-50                 mvtnorm_1.2-5              </span></span>
<span id="cb66-68"><a href="#cb66-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [29] yulab.utils_0.1.4           foreign_0.8-86             </span></span>
<span id="cb66-69"><a href="#cb66-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [31] parallelly_1.37.1           decontam_1.24.0            </span></span>
<span id="cb66-70"><a href="#cb66-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [33] itertools_0.1-3             rstudioapi_0.16.0          </span></span>
<span id="cb66-71"><a href="#cb66-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [35] shape_1.4.6.1               generics_0.1.3             </span></span>
<span id="cb66-72"><a href="#cb66-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [37] gridGraphics_0.5-1          rbiom_1.0.3                </span></span>
<span id="cb66-73"><a href="#cb66-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [39] biomformat_1.32.0           Matrix_1.7-0               </span></span>
<span id="cb66-74"><a href="#cb66-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [41] ggbeeswarm_0.7.2            fansi_1.0.6                </span></span>
<span id="cb66-75"><a href="#cb66-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [43] DECIPHER_3.0.0              abind_1.4-5                </span></span>
<span id="cb66-76"><a href="#cb66-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [45] lifecycle_1.0.4             yaml_2.3.8                 </span></span>
<span id="cb66-77"><a href="#cb66-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [47] recipes_1.0.10              rhdf5_2.48.0               </span></span>
<span id="cb66-78"><a href="#cb66-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [49] SparseArray_1.4.8           Rtsne_0.17                 </span></span>
<span id="cb66-79"><a href="#cb66-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [51] crayon_1.5.3                dir.expiry_1.12.0          </span></span>
<span id="cb66-80"><a href="#cb66-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [53] beachmat_2.20.0             pillar_1.9.0               </span></span>
<span id="cb66-81"><a href="#cb66-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [55] rjson_0.2.21                boot_1.3-30                </span></span>
<span id="cb66-82"><a href="#cb66-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [57] future.apply_1.11.2         lpSolve_5.6.20             </span></span>
<span id="cb66-83"><a href="#cb66-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [59] codetools_0.2-20            glue_1.7.0                 </span></span>
<span id="cb66-84"><a href="#cb66-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [61] ggfun_0.1.5                 data.table_1.15.4          </span></span>
<span id="cb66-85"><a href="#cb66-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [63] vctrs_0.6.5                 png_0.1-8                  </span></span>
<span id="cb66-86"><a href="#cb66-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [65] treeio_1.28.0               urltools_1.7.3             </span></span>
<span id="cb66-87"><a href="#cb66-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [67] gtable_0.3.5                assertthat_0.2.1           </span></span>
<span id="cb66-88"><a href="#cb66-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [69] cachem_1.1.0                gower_1.0.1                </span></span>
<span id="cb66-89"><a href="#cb66-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [71] xfun_0.45                   prodlim_2024.06.25         </span></span>
<span id="cb66-90"><a href="#cb66-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] S4Arrays_1.4.1              tidygraph_1.3.1            </span></span>
<span id="cb66-91"><a href="#cb66-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [75] survival_3.7-0              timeDate_4032.109          </span></span>
<span id="cb66-92"><a href="#cb66-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [77] pheatmap_1.0.12             iterators_1.0.14           </span></span>
<span id="cb66-93"><a href="#cb66-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [79] hardhat_1.4.0               lava_1.8.0                 </span></span>
<span id="cb66-94"><a href="#cb66-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [81] bluster_1.14.0              ROCR_1.0-11                </span></span>
<span id="cb66-95"><a href="#cb66-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [83] ipred_0.9-14                nlme_3.1-165               </span></span>
<span id="cb66-96"><a href="#cb66-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [85] ggtree_3.12.0               filelock_1.0.3             </span></span>
<span id="cb66-97"><a href="#cb66-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [87] irlba_2.3.5.1               vipor_0.4.7                </span></span>
<span id="cb66-98"><a href="#cb66-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [89] rpart_4.1.23                colorspace_2.1-0           </span></span>
<span id="cb66-99"><a href="#cb66-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [91] DBI_1.2.3                   Hmisc_5.1-3                </span></span>
<span id="cb66-100"><a href="#cb66-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [93] nnet_7.3-19                 tidyselect_1.2.1           </span></span>
<span id="cb66-101"><a href="#cb66-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [95] curl_5.2.1                  compiler_4.4.1             </span></span>
<span id="cb66-102"><a href="#cb66-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [97] httr2_1.0.1                 htmlTable_2.4.2            </span></span>
<span id="cb66-103"><a href="#cb66-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [99] BiocNeighbors_1.22.0        basilisk.utils_1.16.0      </span></span>
<span id="cb66-104"><a href="#cb66-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [101] DelayedArray_0.30.1         triebeard_0.4.1            </span></span>
<span id="cb66-105"><a href="#cb66-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [103] checkmate_2.3.1             scales_1.3.0               </span></span>
<span id="cb66-106"><a href="#cb66-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [105] sfsmisc_1.1-18              quadprog_1.5-8             </span></span>
<span id="cb66-107"><a href="#cb66-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [107] rappdirs_0.3.3              digest_0.6.36              </span></span>
<span id="cb66-108"><a href="#cb66-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [109] minqa_1.2.7                 rmarkdown_2.27             </span></span>
<span id="cb66-109"><a href="#cb66-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [111] basilisk_1.16.0             htmltools_0.5.8.1          </span></span>
<span id="cb66-110"><a href="#cb66-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [113] pkgconfig_2.0.3             base64enc_0.1-3            </span></span>
<span id="cb66-111"><a href="#cb66-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [115] lme4_1.1-35.4               sparseMatrixStats_1.16.0   </span></span>
<span id="cb66-112"><a href="#cb66-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [117] fastmap_1.2.0               GlobalOptions_0.1.2        </span></span>
<span id="cb66-113"><a href="#cb66-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [119] rlang_1.1.4                 htmlwidgets_1.6.4          </span></span>
<span id="cb66-114"><a href="#cb66-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [121] UCSC.utils_1.0.0            DelayedMatrixStats_1.26.0  </span></span>
<span id="cb66-115"><a href="#cb66-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [123] farver_2.1.2                zoo_1.8-12                 </span></span>
<span id="cb66-116"><a href="#cb66-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [125] jsonlite_1.8.8              BiocParallel_1.38.0        </span></span>
<span id="cb66-117"><a href="#cb66-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [127] ModelMetrics_1.2.2.2        BiocSingular_1.20.0        </span></span>
<span id="cb66-118"><a href="#cb66-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [129] magrittr_2.0.3              Formula_1.2-5              </span></span>
<span id="cb66-119"><a href="#cb66-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [131] GenomeInfoDbData_1.2.12     ggplotify_0.1.2            </span></span>
<span id="cb66-120"><a href="#cb66-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [133] Rhdf5lib_1.26.0             munsell_0.5.1              </span></span>
<span id="cb66-121"><a href="#cb66-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [135] Rcpp_1.0.12                 ape_5.8                    </span></span>
<span id="cb66-122"><a href="#cb66-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [137] ggnewscale_0.4.10           viridis_0.6.5              </span></span>
<span id="cb66-123"><a href="#cb66-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [139] pROC_1.18.5                 stringi_1.8.4              </span></span>
<span id="cb66-124"><a href="#cb66-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [141] zlibbioc_1.50.0             MASS_7.3-61                </span></span>
<span id="cb66-125"><a href="#cb66-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [143] plyr_1.8.9                  mediation_4.5.0            </span></span>
<span id="cb66-126"><a href="#cb66-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [145] listenv_0.9.1               parallel_4.4.1             </span></span>
<span id="cb66-127"><a href="#cb66-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [147] ggrepel_0.9.5               graphlayouts_1.1.1         </span></span>
<span id="cb66-128"><a href="#cb66-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [149] circlize_0.4.16             hms_1.1.3                  </span></span>
<span id="cb66-129"><a href="#cb66-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [151] HoloFoodR_0.99.7            igraph_2.0.3               </span></span>
<span id="cb66-130"><a href="#cb66-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [153] rngtools_1.5.2              reshape2_1.4.4             </span></span>
<span id="cb66-131"><a href="#cb66-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [155] ScaledMatrix_1.12.0         evaluate_0.24.0            </span></span>
<span id="cb66-132"><a href="#cb66-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [157] RcppParallel_5.1.7          nloptr_2.1.0               </span></span>
<span id="cb66-133"><a href="#cb66-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [159] tzdb_0.4.0                  tweenr_2.0.3               </span></span>
<span id="cb66-134"><a href="#cb66-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [161] polyclip_1.10-6             future_1.33.2              </span></span>
<span id="cb66-135"><a href="#cb66-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [163] clue_0.3-65                 BiocBaseUtils_1.6.0        </span></span>
<span id="cb66-136"><a href="#cb66-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [165] ggforce_0.4.2               rsvd_1.0.5                 </span></span>
<span id="cb66-137"><a href="#cb66-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [167] tidytree_0.4.6              class_7.3-22               </span></span>
<span id="cb66-138"><a href="#cb66-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [169] viridisLite_0.4.2           aplot_0.2.3                </span></span>
<span id="cb66-139"><a href="#cb66-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [171] memoise_2.0.1               beeswarm_0.4.0             </span></span>
<span id="cb66-140"><a href="#cb66-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [173] cluster_2.1.6               corrplot_0.92              </span></span>
<span id="cb66-141"><a href="#cb66-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [175] globals_0.16.3              timechange_0.3.0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-quinnFieldGuideCompositional2019" class="csl-entry" role="listitem">
Quinn, Thomas P., Ionas Erb, Greg Gloor, Cedric Notredame, Mark F. Richardson, and Tamsyn M. Crowley. 2019. <span>“A Field Guide for the Compositional Analysis of Any-Omics Data.”</span> <em>GigaScience</em> 8 (9): giz107. <a href="https://doi.org/10.1093/gigascience/giz107">https://doi.org/10.1093/gigascience/giz107</a>.
</div>
<div id="ref-zarkasiPyrosequencingbasedCharacterizationGastrointestinal2014" class="csl-entry" role="listitem">
Zarkasi, K. Z., G. C. J. Abell, R. S. Taylor, C. Neuman, E. Hatje, M. L. Tamplin, M. Katouli, and J. P. Bowman. 2014. <span>“Pyrosequencing-Based Characterization of Gastrointestinal Bacteria of <span>Atlantic</span> Salmon (<span>Salmo</span> Salar <span>L</span>.) Within a Commercial Mariculture System.”</span> <em>Journal of Applied Microbiology</em> 117 (1): 18–27. <a href="https://doi.org/10.1111/jam.12514">https://doi.org/10.1111/jam.12514</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>