---
title: "ECCB Paper HoloFood Salmon - MOFA2 Analysis"
format: html
---

```{r}
library(mia)
library(MOFA2)
library(reticulate)
library(knitr)
# path <- system.file("extdata", "salmon_merged_mae.RDS", package = "HoloFoodR")
# salmon_merged_mae <- readRDS(path)
salmon_merged_mae <- readRDS("../inst/extdata/salmon_merged_mae.RDS")
```

## Data exploration

```{r}
colData(salmon_merged_mae) |>
  head() |>
  kable()
```
```{r}
colData(salmon_merged_mae) |> names()
```

```{r}
salmon_merged_mae[["FATTY ACIDS MG"]] |> colData() |> head()
```

```{r}
# Rename assays in each of experimentList() for more clarity

assayNames(salmon_merged_mae[["FATTY ACIDS MG"]]) <- "fatty_acid_mg"
assayNames(salmon_merged_mae[["FATTY ACIDS PERCENTAGE"]]) <- "fatty_acid_percentage"
assayNames(salmon_merged_mae[["IODINE"]]) <- "iodine"
```

```{r}
# Transform matrices to numeric
# If a number is < 0.01, I assume it's a 0
fatty_acid_mg_assay <- assay(salmon_merged_mae[["FATTY ACIDS MG"]], "fatty_acid_mg")
fatty_acid_mg_assay[fatty_acid_mg_assay == "<0.01"] <- 0
fatty_acid_mg_assay_numeric <- apply(fatty_acid_mg_assay, 2, as.numeric) 

fatty_acid_mg_assay_numeric <- matrix(fatty_acid_mg_assay_numeric,
                                      nrow = nrow(fatty_acid_mg_assay),
                                      byrow = TRUE,
                                      dimnames = dimnames(fatty_acid_mg_assay))

assay(salmon_merged_mae[["FATTY ACIDS MG"]], "fatty_acid_mg") <- fatty_acid_mg_assay_numeric
```


```{r}
# Transform fatty acids in mg with log10
salmon_merged_mae[["FATTY ACIDS MG"]] <- transformAssay(salmon_merged_mae[["FATTY ACIDS MG"]],
                                                        assay.type = "fatty_acid_mg",
                                                        method = "log10",
                                                        pseudocount = 0.01 / 2)
# Transform microbiome with rclr
salmon_merged_mae[["METAGENOMIC COUNTS"]] <- transformAssay(salmon_merged_mae[["METAGENOMIC COUNTS"]], method = "relabundance")
salmon_merged_mae[["METAGENOMIC COUNTS"]] <- transformAssay(salmon_merged_mae[["METAGENOMIC COUNTS"]], assay.type = "relabundance", method = "rclr")
```

## MOFA analysis

```{r}
# Remove entries from MAE that do not have animal metadata
colData(salmon_merged_mae) <- colData(salmon_merged_mae)[!is.na(colData(salmon_merged_mae)[["animal"]]), ]

# We need to add treatment description to colData of the MAE object
# to use it as a grouping variable in MOFA
fatty_acid_cols_to_merge <- colData(salmon_merged_mae[["FATTY ACIDS MG"]])[c("Treatment description", "animal", "accession")]

colData(salmon_merged_mae) <- merge(
  as.data.frame(colData(salmon_merged_mae),
  as.data.frame(fatty_acid_cols_to_merge), by = c("animal")
))

colData(salmon_merged_mae)[["treatment"]] <- fatty_acid_cols_to_merge[["Treatment description"]]



colData(salmon_merged_mae[["FATTY ACIDS MG"]])["Treatment description"]

colnames(salmon_merged_mae[["FATTY ACIDS MG"]])
```

```{r}
colData(salmon_merged_mae)
```


