<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get data from HoloFood database — getData • HoloFoodR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get data from HoloFood database — getData"><meta name="description" content="Get data from HoloFood database"><meta property="og:description" content="Get data from HoloFood database"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HoloFoodR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/HoloFoodR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/HoloFoodR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get data from HoloFood database</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/HoloFoodR/blob/updates/R/getData.R" class="external-link"><code>R/getData.R</code></a></small>
      <div class="d-none name"><code>getData.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Get data from HoloFood database</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getData</span><span class="op">(</span></span>
<span>  type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  accession.type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  accession <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  flatten <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p><code>NULL</code> or <code>character scalar</code> specifying the type of
data to query. Must be one of the following options:
<code>"analysis-summaries"</code>, <code>"animals"</code>, <code>"genome-catalogues"</code>,
<code>"samples"</code>, <code>"sample_metadata_markers"</code> or
<code>"viral-catalogues"</code>. When genome or viral catalogues is fetched by
their accession ID, the type can also be <code>"genomes"</code> or
<code>"fragments"</code>. (Default: <code>NULL</code>)</p></dd>


<dt id="arg-accession-type">accession.type<a class="anchor" aria-label="anchor" href="#arg-accession-type"></a></dt>
<dd><p><code>NULL</code> or <code>character scalar</code> specifying the
type of accession IDs. Must be one of the following options:
<code>"animals"</code>, <code>"genome-catalogues"</code>, <code>"samples"</code> or
<code>"viral-catalogues"</code>. (Default: <code>NULL</code>)</p></dd>


<dt id="arg-accession">accession<a class="anchor" aria-label="anchor" href="#arg-accession"></a></dt>
<dd><p><code>NULL</code> or <code>character vector</code> specifying the
accession IDs of type <code>accession.type</code>. (Default: <code>NULL</code>)</p></dd>


<dt id="arg-flatten">flatten<a class="anchor" aria-label="anchor" href="#arg-flatten"></a></dt>
<dd><p><code>Logical scalar</code> specifying whether to flatten the
resulting <code>data.frame</code>. This means that columns with multiple values
are separated to multiple columns. (Default: <code>FALSE</code>)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>optional arguments:</p><ul><li><p><strong>max.hits</strong> <code>NULL</code> or <code>integer scalar</code> specifying the
  maximum number of results to fetch. When NULL, all results are fetched.
  (Default: <code>NULL</code>)</p></li>
<li><p><strong>use.cache</strong> <code>Logical scalar</code> specifying whether to
  use cache (Default: <code>FALSE</code>)</p></li>
<li><p><strong>cache.dir</strong> <code>Character scalar</code> specifying cache
  directory. (Default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></li>
<li><p><strong>clear.cache</strong> <code>Logical scalar</code> specifying whether to
  remove and clear cache (Default: <code>FALSE</code>)</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>list</code> or <code>data.frame</code></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>With <code>getData</code>, you can fetch data from the database. Compared to
<code>getResult</code>, this function is more flexible since it can fetch any kind
of data from the database. However, this function returns the data
without further wrangling as <code>list</code> or <code>data.frame</code> which are not
optimized format for fetching data on samples.</p>
<p>Search results can be filtered; for example, animals can be filtered based on
available samples. See [Api browser](https://www.holofooddata.org/api/docs)
for information on filters. You can find help on customizing queries from
[here](https://emg-docs.readthedocs.io/en/latest/api.html#customising-queries).</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="getResult.html">getResult</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find genome catalogues</span></span></span>
<span class="r-in"><span><span class="va">catalogues</span> <span class="op">&lt;-</span> <span class="fu">getData</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"genome-catalogues"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">catalogues</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 id                   title</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  salmon-gut-v2-0  HoloFood Salmon Gut v2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 chicken-gut-v2-0 HoloFood Chicken Gut v2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         biome related_mag_catalogue_id  system</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  root:Host-associated:Fish:Digestive system  non-model-fish-gut-v2-0  salmon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 root:Host-associated:Birds:Digestive system       chicken-gut-v1-0-1 chicken</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   analysis_summaries</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       c("HoloF....</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       c("HoloF....</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find genomes based on certain genome catalogue iD</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getData</span><span class="op">(</span></span></span>
<span class="r-in"><span>    type <span class="op">=</span> <span class="st">"genomes"</span>, accession.type <span class="op">=</span> <span class="st">"genome-catalogues"</span>,</span></span>
<span class="r-in"><span>    accession <span class="op">=</span> <span class="va">catalogues</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"id"</span><span class="op">]</span>, max.hits <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># See the data.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       accession cluster_representative</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 MGYG000307500          MGYG000307500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 MGYG000307501          MGYG000307501</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 MGYG000307502          MGYG000307501</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 MGYG000307503          MGYG000299622</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 MGYG000307504          MGYG000299579</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 MGYG000307505          MGYG000299579</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                      taxonomy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                    Bacteria &gt; Firmicutes_A &gt; Clostridia &gt; Oscillospirales &gt; Ruminococcaceae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 Bacteria &gt; Proteobacteria &gt; Gammaproteobacteria &gt; Pseudomonadales &gt; Pseudomonadaceae &gt; Pseudomonas &gt; Pseudomonas aeruginosa</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 Bacteria &gt; Proteobacteria &gt; Gammaproteobacteria &gt; Pseudomonadales &gt; Pseudomonadaceae &gt; Pseudomonas &gt; Pseudomonas aeruginosa</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4          Bacteria &gt; Proteobacteria &gt; Gammaproteobacteria &gt; Enterobacterales &gt; Aeromonadaceae &gt; Aeromonas &gt; Aeromonas sobria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                                                       Bacteria &gt; Firmicutes &gt; Bacilli &gt; Mycoplasmatales &gt; Mycoplasmoidaceae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                                                       Bacteria &gt; Firmicutes &gt; Bacilli &gt; Mycoplasmatales &gt; Mycoplasmoidaceae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                representative_url metadata1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000307500       196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000307501       197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000307501       198</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000299622       199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000299579       200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000299579       201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Genome_type metadata.Length metadata.N_contigs metadata.N50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                  MAG         1319783                 19       153761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  MAG         6676386                136        83355</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                  MAG         4766025               4741         1104</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                  MAG         2381842                367         8519</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                  MAG          681326                 38        37281</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                  MAG          627706                 26        43119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.GC_content metadata.Completeness metadata.Contamination</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                27.3                 85.79                    0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2               66.27                 98.37                  0.192</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3               64.82                 68.36                  2.053</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4               59.04                 58.95                   0.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5               25.14                 96.02                  0.384</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6               25.04                 95.25                  0.384</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.rRNA_5S metadata.rRNA_16S metadata.rRNA_23S metadata.tRNAs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1              0.0               0.0               0.0             18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2              0.0               0.0               0.0             17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            93.28              36.2             99.04             15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4             91.6               0.0               0.0             14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5              0.0             99.74             99.59             19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6              0.0             99.74             99.59             19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Genome_accession metadata.Sample_accession metadata.Study_accession</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1               ERZ15182294            SAMEA112246717                ERP136460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2               ERZ15182298            SAMEA112246686                ERP136460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3               ERZ15182309            SAMEA112246683                ERP136460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4               ERZ15182314            SAMEA112246713                ERP136460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5               ERZ15233654            SAMEA112264417                ERP125469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6               ERZ15233655            SAMEA112264472                ERP125469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Country metadata.Continent</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6           Norway             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                       metadata.FTP_download</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0003075/MGYG000307500/genomes1/MGYG000307500.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0003075/MGYG000307501/genomes1/MGYG000307501.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0003075/MGYG000307501/genomes1/MGYG000307502.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0002996/MGYG000299622/genomes1/MGYG000307503.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0002995/MGYG000299579/genomes1/MGYG000307504.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/non-model-fish-gut/v2.0/all_genomes/MGYG0002995/MGYG000299579/genomes1/MGYG000307505.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Host_species annotations.cazy.GH annotations.cazy.PL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1           Salmo salar                   7                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2           Salmo salar                  24                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3           Salmo salar                  24                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4           Salmo salar                  27                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5           Salmo salar                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6           Salmo salar                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   annotations.cazy.CE annotations.cazy.AA annotations.cazy.CB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                   0                   0                   1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                   3                   1                   4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                   3                   1                   4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                   1                   1                   6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                   0                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                   0                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   annotations.cazy.GT annotations.cazy.CL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                   7                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  34                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                  34                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                  20                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                   2                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                   2                   0</span>
<span class="r-in"><span><span class="co"># It includes for instance summary of the CAZy</span></span></span>
<span class="r-in"><span><span class="co"># (Carbohydrate-Active enZymes) annotations as a counts per category</span></span></span>
<span class="r-in"><span><span class="va">cazy</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span> , <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"annotations.cazy"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cazy</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   annotations.cazy.GH annotations.cazy.PL annotations.cazy.CE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                   7                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  24                   0                   3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                  24                   0                   3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                  27                   0                   1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                   0                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                   0                   0                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   annotations.cazy.AA annotations.cazy.CB annotations.cazy.GT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                   0                   1                   7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                   1                   4                  34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                   1                   4                  34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                   1                   6                  20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                   0                   0                   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                   0                   0                   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   annotations.cazy.CL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                   0</span>
<span class="r-in"><span><span class="co"># Moreover, it includes a sample list. This sample list represents a </span></span></span>
<span class="r-in"><span><span class="co"># collection of samples where the MAG was identified. Thr data has also the</span></span></span>
<span class="r-in"><span><span class="co"># completeness of MAG in a sample.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"metadata.Sample_accession"</span>, <span class="st">"metadata.Completeness"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Sample_accession metadata.Completeness</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            SAMEA112246717                 85.79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            SAMEA112246686                 98.37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            SAMEA112246683                 68.36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            SAMEA112246713                 58.95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            SAMEA112264417                 96.02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            SAMEA112264472                 95.25</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

