<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Get data from HoloFood database"><title>Get data from HoloFood database — getData • HoloFoodR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get data from HoloFood database — getData"><meta property="og:description" content="Get data from HoloFood database"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HoloFoodR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/HoloFoodR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/HoloFoodR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Get data from HoloFood database</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/HoloFoodR/blob/HEAD/R/getData.R" class="external-link"><code>R/getData.R</code></a></small>
      <div class="d-none name"><code>getData.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Get data from HoloFood database</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getData</span><span class="op">(</span></span>
<span>  type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  accession.type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  accession <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  flatten <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>type</dt>
<dd><p><code>NULL</code> or <code>character scalar</code> specifying the type of
data to query. Must be one of the following options:
<code>"analysis-summaries"</code>, <code>"animals"</code>, <code>"genome-catalogues"</code>,
<code>"samples"</code>, <code>"sample_metadata_markers"</code> or
<code>"viral-catalogues"</code>. When genome or viral catalogues is fetched by
their accession ID, the type can also be <code>"genomes"</code> or
<code>"fragments"</code>. (Default: <code>NULL</code>)</p></dd>


<dt>accession.type</dt>
<dd><p><code>NULL</code> or <code>character scalar</code> specifying the
type of accession IDs. Must be one of the following options:
<code>"animals"</code>, <code>"genome-catalogues"</code>, <code>"samples"</code> or
<code>"viral-catalogues"</code>. (Default: <code>NULL</code>)</p></dd>


<dt>accession</dt>
<dd><p><code>NULL</code> or <code>character vector</code> specifying the
accession IDs of type <code>accession.type</code>. (Default: <code>NULL</code>)</p></dd>


<dt>flatten</dt>
<dd><p><code>Logical scalar</code> specifying whether to flatten the
resulting <code>data.frame</code>. This means that columns with multiple values
are separated to multiple columns. (Default: <code>FALSE</code>)</p></dd>


<dt>...</dt>
<dd><p>optional arguments:</p><ul><li><p><strong>max.hits</strong> <code>NULL</code> or <code>integer scalar</code> specifying the
  maximum number of results to fetch. When NULL, all results are fetched.
  (Default: <code>NULL</code>)</p></li>
<li><p><strong>use.cache</strong> <code>Logical scalar</code> specifying whether to
  use cache (Default: <code>FALSE</code>)</p></li>
<li><p><strong>cache.dir</strong> <code>Character scalar</code> specifying cache
  directory. (Default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></li>
<li><p><strong>clear.cache</strong> <code>Logical scalar</code> specifying whether to
  remove and clear cache (Default: <code>FALSE</code>)</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>list</code> or <code>data.frame</code></p>


    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>With <code>getData</code>, you can fetch data from the database. Compared to
<code>getResult</code>, this function is more flexible since it can fetch any kind
of data from the database. However, this function returns the data
without further wrangling as <code>list</code> or <code>data.frame</code> which are not
optimized format for fetching data on samples.</p>
<p>Search results can be filtered; for example, animals can be filtered based on
available samples. See [Api browser](https://www.holofooddata.org/api/docs)
for information on filters. You can find help on customizing queries from
[here](https://emg-docs.readthedocs.io/en/latest/api.html#customising-queries).</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="getResult.html">getResult</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find genome catalogues</span></span></span>
<span class="r-in"><span><span class="va">catalogues</span> <span class="op">&lt;-</span> <span class="fu">getData</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"genome-catalogues"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">catalogues</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 id                   title</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 chicken-gut-v1-0 HoloFood Chicken Gut v1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  salmon-gut-v1-0  HoloFood Salmon Gut v1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         biome    related_mag_catalogue_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 root:Host-associated:Birds:Digestive system            chicken-gut-v1-0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  root:Host-associated:Fish:Digestive system non-model-fish-gut-gut-v2-0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    system analysis_summaries</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 chicken       c("HoloF....</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  salmon       c("HoloF....</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find genomes based on certain genome catalogue iD</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getData</span><span class="op">(</span></span></span>
<span class="r-in"><span>    type <span class="op">=</span> <span class="st">"genomes"</span>, accession.type <span class="op">=</span> <span class="st">"genome-catalogues"</span>,</span></span>
<span class="r-in"><span>    accession <span class="op">=</span> <span class="va">catalogues</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"id"</span><span class="op">]</span>, max.hits <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       accession cluster_representative</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 MGYG000308381          MGYG000310807</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 MGYG000308382          MGYG000320406</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 MGYG000308383          MGYG000319700</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 MGYG000308384          MGYG000316616</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 MGYG000308385          MGYG000318258</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 MGYG000308386          MGYG000312616</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                             taxonomy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          Bacteria &gt; Firmicutes_A &gt; Clostridia &gt; Oscillospirales &gt; Acutalibacteraceae &gt; Acutalibacter &gt; Acutalibacter ornithocaccae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  Bacteria &gt; Firmicutes_A &gt; Clostridia &gt; Lachnospirales &gt; Lachnospiraceae &gt; Scybalocola &gt; Scybalocola faecipullorum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 Bacteria &gt; Cyanobacteria &gt; Vampirovibrionia &gt; Gastranaerophilales &gt; Gastranaerophilaceae &gt; Stercorousia &gt; Stercorousia sp000437435</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  Bacteria &gt; Actinobacteriota &gt; Actinomycetia &gt; Actinomycetales &gt; Bifidobacteriaceae &gt; Bifidobacterium &gt; Bifidobacterium pullorum_B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                            Bacteria &gt; Firmicutes_A &gt; Clostridia &gt; Lachnospirales &gt; CAG-274 &gt; Gallispira &gt; Gallispira edinburgensis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6           Bacteria &gt; Proteobacteria &gt; Gammaproteobacteria &gt; Enterobacterales &gt; Enterobacteriaceae &gt; Escherichia &gt; Escherichia coli</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                representative_url metadata1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000310807       781</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000320406       782</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000319700       783</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000316616       784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000318258       785</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 https://www.ebi.ac.uk/metagenomics/api/v1/genomes/MGYG000312616       786</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Genome_type metadata.Length metadata.N_contigs metadata.N50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                  MAG         1890035                189        12501</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  MAG         2145586                258         9928</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                  MAG         2206567                 44       140695</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                  MAG         1441532                255         6193</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                  MAG         2550799                 74        51548</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                  MAG         3876255                351        15239</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.GC_content metadata.Completeness metadata.Contamination</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1               62.27                  86.2                    0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2               44.49                 67.14                    0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3               30.68                 83.76                   1.28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4               64.96                 79.39                   2.35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5               37.11                 89.82                    0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6               51.07                 87.75                   1.01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.rRNA_5S metadata.rRNA_16S metadata.rRNA_23S metadata.tRNAs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1              0.0               0.0               0.0             15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2              0.0               0.0               0.0             13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3              0.0               0.0               0.0             20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4              0.0             10.18               0.0             19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5              0.0             16.96               0.0             16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6             95.8               0.0              8.34             19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Genome_accession metadata.Sample_accession metadata.Study_accession</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1               ERZ15233365            SAMEA112264197                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2               ERZ15233366            SAMEA112264327                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3               ERZ15233367            SAMEA112264363                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4               ERZ15233368            SAMEA112264171                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5               ERZ15233369            SAMEA112264145                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6               ERZ15233370            SAMEA112264292                ERP122587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   metadata.Country metadata.Continent</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            Spain             Europe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                metadata.FTP_download</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003108/MGYG000310807/genomes1/MGYG000308381.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003204/MGYG000320406/genomes1/MGYG000308382.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003197/MGYG000319700/genomes1/MGYG000308383.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003166/MGYG000316616/genomes1/MGYG000308384.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003182/MGYG000318258/genomes1/MGYG000308385.gff.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/chicken-gut/v1.0/all_genomes/MGYG0003126/MGYG000312616/genomes1/MGYG000308386.gff.gz</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

