<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get data on samples from HoloFood database — getResult • HoloFoodR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get data on samples from HoloFood database — getResult"><meta name="description" content="Get data on samples from HoloFood database"><meta property="og:description" content="Get data on samples from HoloFood database"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HoloFoodR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/HoloFoodR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/HoloFoodR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get data on samples from HoloFood database</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/HoloFoodR/blob/fix_metabolights/R/getResult.R" class="external-link"><code>R/getResult.R</code></a></small>
      <div class="d-none name"><code>getResult.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Get data on samples from HoloFood database</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getResult</span><span class="op">(</span><span class="va">accession</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-accession">accession<a class="anchor" aria-label="anchor" href="#arg-accession"></a></dt>
<dd><p><code>Character vector</code> specifying the
accession IDs of type samples.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>optional arguments:</p><ul><li><p><strong>use.cache</strong> <code>Logical scalar</code> specifying whether to
  use cache. Note that when <code>get.metabolomic = TRUE</code> is specified, the
  file from the MetaboLights is stored in the local system to the location
  specified by <code>cache.dir</code> despite of the value of <code>use.cache</code>.
  (Default: <code>FALSE</code>)</p></li>
<li><p><strong>cache.dir</strong> <code>Character scalar</code> specifying cache
  directory. (Default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></li>
<li><p><strong>clear.cache</strong> <code>Logical scalar</code> specifying whether to
  use.cache (Default: <code>FALSE</code>)</p></li>
<li><p><strong>assay.type</strong> <code>Character scalar</code> specifying the name of
  assay in resulting <code>TreeSummarizedExperiment</code> object.
  (Default: <code>"counts"</code>)</p></li>
<li><p><strong>get.metabolomic</strong> <code>Logical scalar</code> specifying whether to
  retrieve processed metabolomic data from MetaboLights database. For
  retrieving spectra data, refer to
  <code><a href="getMetaboLights.html">getMetaboLights</a></code> documentation.
  (Default: <code>FALSE</code>)</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>MultiAssayExperiment</code></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>With <code>getResult</code>, you can fetch data on samples from the HoloFood
database. Compared to <code>getData</code>, this function is more convenient for
fetching the samples data because it converts the data to
<code>MultiAssayExperiment</code> where different omics are stored as
<code>TreeSummarizedExperiment</code> objects which are optimized for downstream
analytics. Columns of returned <code>MultiAssayExperiment</code> are individual
animals. These columns are linked with individual samples that are stored in
<code>TreeSummarizedExperiment</code> objects.</p>
<p>The HoloFood database lacks non-targeted metabolomic data but they can be
fetched from MetaboLights resource. Certain datasets include processed
features. Those datasets can be retrieved with the function
<code>getResult</code> which integrates metabolomic data with other datasets from
HoloFood.</p>
<p>Furthermore, while the HoloFoodR database does not include metagenomic
assembly data, users can access such data from the MGnify database. The
MGnifyR package provides a convenient interface for accessing this database.
By employing <code>MGnifyR::getResult()</code>, users can obtain data formatted as
a <code>MultiAssayExperiment</code> object, containing multiple
<code>TreeSummarizedExperiment</code> objects. Consequently, data from both
HoloFood and MGnify databases are inherently compatible for subsequent
downstream analysis.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="getData.html">getData</a></code>
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
<code><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment-class.html" class="external-link">MultiAssayExperiment</a></code>
<code>MGnifyR:getResult</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find samples on certain animal</span></span></span>
<span class="r-in"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="doQuery.html">doQuery</a></span><span class="op">(</span><span class="st">"samples"</span>, animal_accession <span class="op">=</span> <span class="st">"SAMEA112904746"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get the data</span></span></span>
<span class="r-in"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu">getResult</span><span class="op">(</span><span class="va">samples</span><span class="op">[[</span><span class="st">"accession"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Data for the following samples cannot be found. The sample types are metagenomic_assembly, host_genomic, transcriptomic and metatranscriptomic. (Note that metagenomic assemblies can be found from the MGnify database. See MGnifyR package.):</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> 'SAMEA10130039', 'SAMEA13389496', 'SAMEA13389497', 'SAMEA13901618', 'SAMEA13901619', 'SAMEA13929785', 'SAMEA7571815'</span>
<span class="r-in"><span><span class="va">mae</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 8 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 8:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] BIOGENIC AMINES: TreeSummarizedExperiment with 7 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] FATTY ACIDS: TreeSummarizedExperiment with 19 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] HISTOLOGY: TreeSummarizedExperiment with 20 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] INFLAMMATORY MARKERS: TreeSummarizedExperiment with 14 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] metagenomic_assembly: TreeSummarizedExperiment with 0 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] host_genomic: TreeSummarizedExperiment with 0 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] transcriptomic: TreeSummarizedExperiment with 0 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] metatranscriptomic: TreeSummarizedExperiment with 0 rows and 1 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

