<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="HoloFoodR">
<title>HoloFoodR: interface to HoloFoodR database • HoloFoodR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HoloFoodR: interface to HoloFoodR database">
<meta property="og:description" content="HoloFoodR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HoloFoodR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/HoloFoodR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/HoloFoodR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>HoloFoodR: interface to HoloFoodR database</h1>
            
            <h4 data-toc-skip class="date">2024-06-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/HoloFoodR/blob/HEAD/vignettes/HoloFoodR.Rmd" class="external-link"><code>vignettes/HoloFoodR.Rmd</code></a></small>
      <div class="d-none name"><code>HoloFoodR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>HoloFoodR</code> is a package designed to ease access to the
EBI’s <a href="https://www.holofooddata.org/" class="external-link">HoloFoodR</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis.</p>
<p>The HoloFood database does not encompass metagenomics data; however,
such data is stored within the <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> database. Both
packages offer analogous functionalities, streamlining the integration
of data and enhancing accessibility.</p>
<p><em><a href="https://bioconductor.org/packages/3.20/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>HoloFoodR</code> is hosted on Bioconductor, and can be
installed using via <code>BiocManager</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"HoloFoodR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-package">Load the package<a class="anchor" aria-label="anchor" href="#load-the-package"></a>
</h2>
<p>Once installed, <code>HoloFoodR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/HoloFoodR" class="external-link">HoloFoodR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: TreeSummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span><span class="co">#&gt; Loading required package: XVector</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     strsplit</span></span>
<span><span class="co">#&gt; Loading required package: MultiAssayExperiment</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functionalities">Functionalities<a class="anchor" aria-label="anchor" href="#functionalities"></a>
</h2>
<p><code>HoloFoodR</code> offers three functions <code><a href="../reference/doQuery.html">doQuery()</a></code>,
<code><a href="../reference/getResult.html">getResult()</a></code> and <code><a href="../reference/getData.html">getData()</a></code> which can be
utilized to search and fetch data from HoloFood database.</p>
<p>In this tutorial, we demonstrate how to search animals, subset
animals based on whether they have specific sample type, and finally
fetch the data on samples. Note that this same can be done with
<code><a href="../reference/doQuery.html">doQuery()</a></code> and <code><a href="../reference/getResult.html">getResult()</a></code> (or
<code><a href="../reference/getData.html">getData()</a></code> and <code><a href="../reference/getResult.html">getResult()</a></code>) only by utilizing
query filters. This tutorial is for demonstrating the functionality of
the package.</p>
<p>Additionally, the package includes <code><a href="../reference/getMetaboLights.html">getMetaboLights()</a></code>
function which can be utilized to retrieve metabolomic data from
MetaboLights database.</p>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p>To search animals, genome catalogues, samples or viral catalogues,
you can use <code><a href="../reference/doQuery.html">doQuery()</a></code> function. You can also use
<code><a href="../reference/getData.html">getData()</a></code> but <code><a href="../reference/doQuery.html">doQuery()</a></code> is optimized for
searching these datatypes. For example, instead of nested list of sample
types <code><a href="../reference/doQuery.html">doQuery()</a></code> returns sample types as presence/absence
table which is more convenient.</p>
<p>Here we search animals, and subset them based on whether they include
histological samples. Note that this same can be done also by using
query filters.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">animals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span><span class="st">"animals"</span>, max.hits <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">animals</span> <span class="op">&lt;-</span> <span class="va">animals</span><span class="op">[</span> <span class="va">animals</span><span class="op">[[</span><span class="st">"histological"</span><span class="op">]</span><span class="op">]</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">animals</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "accession"            "system"               "canonical_url"       </span></span>
<span><span class="co">#&gt; [4] "histological"         "host_genomic"         "inflammatory_markers"</span></span></code></pre></div>
<p><code><a href="../reference/doQuery.html">doQuery()</a></code> returns a <code>data.frame</code> including
information on type of data being searched.</p>
</div>
<div class="section level3">
<h3 id="get-data">Get data<a class="anchor" aria-label="anchor" href="#get-data"></a>
</h3>
<p>Now we have information on which animal has histological samples.
Let’s get data on those animals to know the sample IDs to fetch.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">animal_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getData.html">getData</a></span><span class="op">(</span></span>
<span>    accession.type <span class="op">=</span> <span class="st">"animals"</span>, accession <span class="op">=</span> <span class="va">animals</span><span class="op">[[</span><span class="st">"accession"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>The returned value of <code><a href="../reference/getData.html">getData()</a></code> function is a
<code>list</code>. We can have the data also as a
<code>data.frame</code> when we specify <code>flatten = TRUE</code>. The
data has information on animals including samples that have been drawn
from them.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">animal_data</span><span class="op">[[</span><span class="st">"samples"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "accession"        "title"            "sample_type"      "animal"          </span></span>
<span><span class="co">#&gt; [5] "canonical_url"    "metagenomics_url"</span></span></code></pre></div>
<p>The elements of the <code>list</code> are <code>data.frames</code>.
For example, “samples” table contains information on samples drawn from
animals that were specified in input.</p>
<p>Now we can collect sample IDs.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">samples</span><span class="op">[[</span><span class="st">"accession"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-data-on-samples">Get data on samples<a class="anchor" aria-label="anchor" href="#get-data-on-samples"></a>
</h3>
<p>To get data on samples, we can utilize <code><a href="../reference/getResult.html">getResult()</a></code>
function. It returns the data in <em><a href="https://bioconductor.org/packages/3.20/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
(<code>MAE</code>) format. By specifying
<code>get.metabolomic = TRUE</code>, we can fetch also non-targeted
metabolomic data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">sample_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Data for the following samples cannot be found. The sample types are metagenomic_assembly, host_genomic, transcriptomic and metatranscriptomic. (Note that metagenomic assemblies can be found from the MGnify database. See MGnifyR package.):</span></span>
<span><span class="co">#&gt; 'SAMEA10130025', 'SAMEA13389405', 'SAMEA13389406', 'SAMEA13901590', 'SAMEA13901591', 'SAMEA13929779', 'SAMEA7697591', 'SAMEA10130091', 'SAMEA13389692', 'SAMEA13389693', 'SAMEA13901708', 'SAMEA7571845', 'SAMEA10158030', 'SAMEA13389419', 'SAMEA13389420', 'SAMEA13901594', 'SAMEA13901595', 'SAMEA13929781', 'SAMEA7697592', 'SAMEA10130039', 'SAMEA13389496', 'SAMEA13389497', 'SAMEA13901618', 'SAMEA13901619', 'SAMEA13929785', 'SAMEA7571815', 'SAMEA10130112', 'SAMEA13389794', 'SAMEA13389795', 'SAMEA13901758', 'SAMEA13901759', 'SAMEA13929811', 'SAMEA7571864', 'SAMEA10158022', 'SAMEA13389146', 'SAMEA13389147', 'SAMEA13901511', 'SAMEA13901512', 'SAMEA13929767', 'SAMEA7571777', 'SAMEA10130019', 'SAMEA13389353', 'SAMEA13389354', 'SAMEA13389355', 'SAMEA13901574', 'SAMEA13901575', 'SAMEA14095991', 'SAMEA7722475', 'SAMEA10130101', 'SAMEA13389738', 'SAMEA13389739', 'SAMEA13901730', 'SAMEA13901731', 'SAMEA13929802', 'SAMEA7571856', 'SAMEA10455480', 'SAMEA13389220', 'SAMEA10129993', 'SAMEA13389183', 'SAMEA13389184', 'SAMEA13901520', 'SAMEA13901521', 'SAMEA7697579', 'SAMEA10130017', 'SAMEA13389345', 'SAMEA13389346', 'SAMEA13901571', 'SAMEA13901572', 'SAMEA13929772', 'SAMEA7571801', 'SAMEA10130113', 'SAMEA13389807', 'SAMEA13389808', 'SAMEA13901762', 'SAMEA13901763', 'SAMEA13929813', 'SAMEA7571866', 'SAMEA10455481', 'SAMEA13389227', 'SAMEA10455479', 'SAMEA13389169', 'SAMEA10130020', 'SAMEA13389357', 'SAMEA13389358', 'SAMEA13901576', 'SAMEA13901577', 'SAMEA13929773', 'SAMEA7697587', 'SAMEA10130100', 'SAMEA13389734', 'SAMEA13389735', 'SAMEA13901728', 'SAMEA13901729', 'SAMEA13929801', 'SAMEA7697622', 'SAMEA10130016', 'SAMEA13389342', 'SAMEA13389343', 'SAMEA13901569', 'SAMEA13901570', 'SAMEA13929771', 'SAMEA7571800', 'SAMEA10130040', 'SAMEA13389503', 'SAMEA13389504', 'SAMEA13901620', 'SAMEA13901621', 'SAMEA13929786', 'SAMEA7571816', 'SAMEA10129979', 'SAMEA13389081', 'SAMEA13389082', 'SAMEA13901489', 'SAMEA13901490', 'SAMEA7571769', 'SAMEA10130002', 'SAMEA13389243', 'SAMEA13389244', 'SAMEA13901540', 'SAMEA13901541', 'SAMEA7697582', 'SAMEA10129985', 'SAMEA13389133', 'SAMEA13389134', 'SAMEA13901505', 'SAMEA13901506', 'SAMEA13929764', 'SAMEA7571775', 'SAMEA10455476', 'SAMEA13389110', 'SAMEA10130031', 'SAMEA13389443', 'SAMEA13389444', 'SAMEA13901602', 'SAMEA13901603', 'SAMEA7571811', 'SAMEA10130023', 'SAMEA13389395', 'SAMEA13389396', 'SAMEA13901586', 'SAMEA13901587', 'SAMEA13929777', 'SAMEA7571806', 'SAMEA10130090', 'SAMEA13389687', 'SAMEA13389688', 'SAMEA13901707', 'SAMEA7571844', 'SAMEA10130119', 'SAMEA13389832', 'SAMEA13389833', 'SAMEA13901773', 'SAMEA13929818', 'SAMEA7697633', 'SAMEA10129996', 'SAMEA13389204', 'SAMEA13389205', 'SAMEA13901529', 'SAMEA13901530', 'SAMEA7697580', 'SAMEA10130088', 'SAMEA13389677', 'SAMEA13389678', 'SAMEA13901704', 'SAMEA13929799', 'SAMEA7571843'</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 8 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 8:</span></span>
<span><span class="co">#&gt;  [1] BIOGENIC AMINES: TreeSummarizedExperiment with 7 rows and 35 columns</span></span>
<span><span class="co">#&gt;  [2] FATTY ACIDS: TreeSummarizedExperiment with 19 rows and 57 columns</span></span>
<span><span class="co">#&gt;  [3] HISTOLOGY: TreeSummarizedExperiment with 20 rows and 57 columns</span></span>
<span><span class="co">#&gt;  [4] INFLAMMATORY MARKERS: TreeSummarizedExperiment with 14 rows and 58 columns</span></span>
<span><span class="co">#&gt;  [5] metagenomic_assembly: TreeSummarizedExperiment with 0 rows and 53 columns</span></span>
<span><span class="co">#&gt;  [6] host_genomic: TreeSummarizedExperiment with 0 rows and 53 columns</span></span>
<span><span class="co">#&gt;  [7] transcriptomic: TreeSummarizedExperiment with 0 rows and 44 columns</span></span>
<span><span class="co">#&gt;  [8] metatranscriptomic: TreeSummarizedExperiment with 0 rows and 16 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p><code>MAE</code> object stores individual omics as <em><a href="https://bioconductor.org/packages/3.20/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>
(<code>TreeSE</code>) objects.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 7 35 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(7): Cadaverin Gesamtamine (Total biogenic amines) ... Spermin</span></span>
<span><span class="co">#&gt;   Tyramin</span></span>
<span><span class="co">#&gt; rowData names(4): marker.name marker.type marker.canonical_url units</span></span>
<span><span class="co">#&gt; colnames(35): SAMEA112906114 SAMEA112906592 ... SAMEA112906002</span></span>
<span><span class="co">#&gt;   SAMEA112906785</span></span>
<span><span class="co">#&gt; colData names(13): accession sample_type ... Project Sample code</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p>In <code>TreeSE</code>, each column represents a sample and rows
represent features.</p>
</div>
<div class="section level3">
<h3 id="incorporate-with-mgnify-data">Incorporate with MGnify data<a class="anchor" aria-label="anchor" href="#incorporate-with-mgnify-data"></a>
</h3>
<p><code>MGnifyR</code> is a package that can be utilized to fetch
metagenomics data from MGnify database. From the <code>MGnifyR</code>
package, we can use <code>MGnifyR::searchAnalysis()</code> function to
search analyses based on sample IDs that we have.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MGnifyR</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu">MgnifyClient</span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get those samples that are metagenomic samples</span></span>
<span><span class="va">metagenomic_samples</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">[</span></span>
<span>    <span class="va">samples</span><span class="op">[[</span><span class="st">"sample_type"</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"metagenomic_assembly"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Get analysis IDs based on sample IDs</span></span>
<span><span class="va">analysis_ids</span> <span class="op">&lt;-</span> <span class="fu">searchAnalysis</span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, type <span class="op">=</span> <span class="st">"samples"</span>, <span class="va">metagenomic_samples</span><span class="op">[[</span><span class="st">"accession"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analysis_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt;  SAMEA10130025   SAMEA7697591  SAMEA10130091   SAMEA7571845  SAMEA10158030 </span></span>
<span><span class="co">#&gt; "MGYA00606535" "MGYA00616692" "MGYA00606528" "MGYA00616689" "MGYA00606518" </span></span>
<span><span class="co">#&gt;   SAMEA7697592 </span></span>
<span><span class="co">#&gt; "MGYA00615947"</span></span></code></pre></div>
<p>Then we can fetch data based on accession IDs.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae_metagenomic</span> <span class="op">&lt;-</span> <span class="fu">MGnifyR</span><span class="fu">::</span><span class="fu">getResult</span><span class="op">(</span><span class="va">mg</span>, <span class="va">analysis_ids</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae_metagenomic</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 6 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 6:</span></span>
<span><span class="co">#&gt;  [1] microbiota: TreeSummarizedExperiment with 675 rows and 52 columns</span></span>
<span><span class="co">#&gt;  [2] go-slim: TreeSummarizedExperiment with 116 rows and 52 columns</span></span>
<span><span class="co">#&gt;  [3] go-terms: TreeSummarizedExperiment with 3264 rows and 52 columns</span></span>
<span><span class="co">#&gt;  [4] interpro-identifiers: TreeSummarizedExperiment with 19681 rows and 52 columns</span></span>
<span><span class="co">#&gt;  [5] taxonomy: TreeSummarizedExperiment with 1438 rows and 52 columns</span></span>
<span><span class="co">#&gt;  [6] taxonomy-lsu: TreeSummarizedExperiment with 1856 rows and 52 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p><code>MGnifyR::getResult()</code> returns <code>MAE</code> object
just like <code>HoloFoodR</code>. However, metagenomic data points to
individual analyses instead of samples. We can harmonize the data by
replacing analysis IDs with sample IDs, and then we can combine the data
to single <code>MAE</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get experiments from metagenomic data</span></span>
<span><span class="va">exps</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">mae_metagenomic</span><span class="op">)</span></span>
<span><span class="co"># Convert analysis names to sample names</span></span>
<span><span class="va">exps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">exps</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># Get corresponding sample ID</span></span>
<span>    <span class="va">sample_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">analysis_ids</span><span class="op">)</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">analysis_ids</span><span class="op">)</span> <span class="op">]</span></span>
<span>    <span class="co"># Replace analysis ID with sample ID</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_id</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add to original MultiAssayExperiment</span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span>, <span class="va">exps</span><span class="op">)</span></span>
<span><span class="va">mae</span></span></code></pre></div>
<p>Now, with the <code>MAE</code> object linking samples from various
omics together, compatibility is ensured as the single omics datasets
are in <code>(Tree)SummarizedExperiment</code> format. This
compatibility allows us to harness cutting-edge downstream analytics
tools like <a href="https://microbiome.github.io/" class="external-link">miaverse
framework</a> that support these data containers seamlessly.</p>
</div>
<div class="section level3">
<h3 id="extra-get-data-from-metabolights-database">Extra: Get data from MetaboLights database<a class="anchor" aria-label="anchor" href="#extra-get-data-from-metabolights-database"></a>
</h3>
<p>The HoloFood database exclusively contains targeted metabolomic data.
However, it provides URL addresses linking to the MetaboLights database,
where untargeted metabolomics data can be accessed. To retrieve this
data, you can utilize the getMetaboLights() function or preferably
getResult(). Below, we retrieve all the metabolomic data associated with
HoloFood.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get untargeted metabolomic samples</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span><span class="st">"samples"</span>, sample_type <span class="op">=</span> <span class="st">"metabolomic"</span><span class="op">)</span></span>
<span><span class="co"># Get the data</span></span>
<span><span class="va">metabolomic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetaboLights.html">getMetaboLights</a></span><span class="op">(</span><span class="va">samples</span><span class="op">[[</span><span class="st">"metabolomics_url"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show names of data.frames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">metabolomic</span><span class="op">)</span></span></code></pre></div>
<p>The result is a list that includes three data.frames:</p>
<pre><code>- study metadata
- assay metadata
- assay that includes abundance table and feature metadata</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] HoloFoodR_0.99.8                MultiAssayExperiment_1.31.3    </span></span>
<span><span class="co">#&gt;  [3] TreeSummarizedExperiment_2.13.0 Biostrings_2.73.1              </span></span>
<span><span class="co">#&gt;  [5] XVector_0.45.0                  SingleCellExperiment_1.27.2    </span></span>
<span><span class="co">#&gt;  [7] SummarizedExperiment_1.35.0     Biobase_2.65.0                 </span></span>
<span><span class="co">#&gt;  [9] GenomicRanges_1.57.1            GenomeInfoDb_1.41.1            </span></span>
<span><span class="co">#&gt; [11] IRanges_2.39.0                  S4Vectors_0.43.0               </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.51.0             MatrixGenerics_1.17.0          </span></span>
<span><span class="co">#&gt; [15] matrixStats_1.3.0               knitr_1.47                     </span></span>
<span><span class="co">#&gt; [17] BiocStyle_2.33.1               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] httr2_1.0.1             xfun_0.45               bslib_0.7.0            </span></span>
<span><span class="co">#&gt;  [4] htmlwidgets_1.6.4       lattice_0.22-6          yulab.utils_0.1.4      </span></span>
<span><span class="co">#&gt;  [7] vctrs_0.6.5             tools_4.4.1             generics_0.1.3         </span></span>
<span><span class="co">#&gt; [10] curl_5.2.1              parallel_4.4.1          tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [13] fansi_1.0.6             BiocBaseUtils_1.7.0     pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt; [16] Matrix_1.7-0            desc_1.4.3              lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt; [19] GenomeInfoDbData_1.2.12 compiler_4.4.1          treeio_1.29.0          </span></span>
<span><span class="co">#&gt; [22] textshaping_0.4.0       codetools_0.2-20        htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [25] sass_0.4.9              lazyeval_0.2.2          yaml_2.3.8             </span></span>
<span><span class="co">#&gt; [28] tidyr_1.3.1             pkgdown_2.0.9           pillar_1.9.0           </span></span>
<span><span class="co">#&gt; [31] crayon_1.5.3            jquerylib_0.1.4         BiocParallel_1.39.0    </span></span>
<span><span class="co">#&gt; [34] DelayedArray_0.31.3     cachem_1.1.0            abind_1.4-5            </span></span>
<span><span class="co">#&gt; [37] nlme_3.1-165            tidyselect_1.2.1        digest_0.6.36          </span></span>
<span><span class="co">#&gt; [40] dplyr_1.1.4             purrr_1.0.2             bookdown_0.39          </span></span>
<span><span class="co">#&gt; [43] fastmap_1.2.0           grid_4.4.1              cli_3.6.3              </span></span>
<span><span class="co">#&gt; [46] SparseArray_1.5.10      magrittr_2.0.3          S4Arrays_1.5.1         </span></span>
<span><span class="co">#&gt; [49] utf8_1.2.4              ape_5.8                 rappdirs_0.3.3         </span></span>
<span><span class="co">#&gt; [52] UCSC.utils_1.1.0        rmarkdown_2.27          httr_1.4.7             </span></span>
<span><span class="co">#&gt; [55] ragg_1.3.2              memoise_2.0.1           evaluate_0.24.0        </span></span>
<span><span class="co">#&gt; [58] rlang_1.1.4             Rcpp_1.0.12             tidytree_0.4.6         </span></span>
<span><span class="co">#&gt; [61] glue_1.7.0              BiocManager_1.30.23     jsonlite_1.8.8         </span></span>
<span><span class="co">#&gt; [64] R6_2.5.1                systemfonts_1.1.0       fs_1.6.4               </span></span>
<span><span class="co">#&gt; [67] zlibbioc_1.51.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
